diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 77bd78c7b..67ef80950 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -116,6 +116,7 @@ import javax.crypto.SecretKey;
 
 import X.c1_01f;
 import X.c1_2Oc;
+import X.c1_2Ot;
 import X.c1_3Ul;
 import X.c1_2P6;
 import X.c1_2ua;
@@ -648,7 +649,11 @@ public class Coocoo {
         Log.e(TAG, "-----------------3---------------- // value : " + value);
     }
 
+    // invoke-static {vx}, Lcom/coocoo/coocoo/Coocoo;->traceLog4(Ljava/lang/Object;)V
     public static void traceLog4(Object object) {
+        if (object instanceof c1_2Ot) {
+            Log.e(TAG, "-----------------4---------------- // object : " + object);
+        }
         Log.e(TAG, "-----------------4---------------- // object : " + object);
     }
 
@@ -656,6 +661,21 @@ public class Coocoo {
         Log.e(TAG, "-----------------5---------------- // value : " + value);
     }
 
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog6(Z)V
+    public static void traceLog6(boolean value) {
+        Log.e(TAG, "-----------------6---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog7(Ljava/lang/CharSequence;)V
+    public static void traceLog7(CharSequence value) {
+        Log.e(TAG, "-----------------7---------------- // value : " + value);
+    }
+
+    // invoke-static {}, Lcom/coocoo/coocoo/Coocoo;->stackTrace()V
+    public static void stackTrace() throws Exception {
+        throw new Exception("Debug");
+    }
+
     public static void traceLogCodeStack() {
         try {
             throw new IllegalArgumentException("traceLogCodeStack");
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
index a42ff3bbe..6d0a22336 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
@@ -2,6 +2,7 @@ package com.coocoo.whatsappdelegate
 
 import com.coocoo.annotation.SmaliInject
 import com.coocoo.report.Report
+import com.coocoo.utils.LogUtil
 import com.coocoowhatsapp.contact.picker.ContactPickerFragment
 
 object ContactPickerSendBtnDelegate {
@@ -12,6 +13,7 @@ object ContactPickerSendBtnDelegate {
     @SmaliInject
     @JvmStatic
     fun onSendBtnClick(fragment: ContactPickerFragment) {
+        LogUtil.d("onSendBtnClick")
         fragment.A09()?.intent?.getStringExtra(BUNDLE_KEY_STATUS_FEED_ID)?.let { statusId ->
             if (statusId.isNotEmpty()) {
                 Report.reportStatusFeedPlaybackShareDone(statusId)
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
index 2b8c43fa2..c058420b6 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
@@ -946,4 +946,10 @@ public class ConversationsFragmentDelegate extends FragmentDelegate<Conversation
         }
         notifyConversationsAdapter();
     }
+
+    @SmaliInject
+    public void resolveIntent(Intent intent) {
+        List<String> data = intent.getStringArrayListExtra("jids");
+        LogUtil.d("resolveIntent:" + data);
+    }
 }
