diff --git a/app/build.gradle b/app/build.gradle
index 50e40530d..cbf792786 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -42,8 +42,8 @@ android {
 //                || cur_flavor == "YoUnclone"
 //                || cur_flavor == "GbUnclone"
 //                || cur_flavor == "WhatsappPlus") {
-//            rootProject.ext.version_name = '"11.60.0"' // HeyMods
-//            rootProject.ext.version_code = "1160000" // HeyMods
+            rootProject.ext.version_name = '"9.0.0"' // HeyMods
+            rootProject.ext.version_code = "9000000" // HeyMods
 //        } else if (cur_flavor == "FM") {
 //            rootProject.ext.version_name = '"10.1.0"'
 //            rootProject.ext.version_code = "1001000"
@@ -84,8 +84,8 @@ android {
 //            rootProject.ext.version_name = '"0.20.197"'
 //            rootProject.ext.version_code = "20197"
 //        }
-//            rootProject.ext.version_name = '"11.60.0"' // HeyMods
-//            rootProject.ext.version_code = "1160000" // HeyMods
+            rootProject.ext.version_name = '"9.0.0"' // HeyMods
+            rootProject.ext.version_code = "9000000" // HeyMods
     }
 
     flavorDimensions "version"
diff --git a/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt b/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt
index da729f62e..a4726e634 100644
--- a/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt
+++ b/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt
@@ -1,6 +1,7 @@
 package com.coocoo.manager
 
 import com.coocoo.CooCooApp
+import com.coocoo.exoplayer2.util.Log
 import com.coocoo.firebase.remoteConfig.RemoteConfig
 import com.coocoo.remote.simple.model.WhatstallerInfo
 import com.coocoo.report.ReportOKDownload
@@ -139,7 +140,9 @@ object WhatstallerPreDownloader {
         withContext(Dispatchers.IO) {
             try {
                 val whatsTallerInfo = info ?: return@withContext false
+                Log.d("AAA", "apkChannel: $apkChannel, whatsTallerInfo: $info")
                 val lastFile = getFile(apkChannel, whatsTallerInfo) ?: return@withContext false
+                Log.d("AAA", "lastFile: ${lastFile.absolutePath}")
                 return@withContext lastFile.exists() && MD5Checker.compare(
                     whatsTallerInfo.md5Pair,
                     lastFile
