diff --git a/app/src/main/java/com/coocoo/fm/h.java b/app/src/main/java/com/coocoo/fm/h.java
index 0f23f1ddc..1fae8ac7a 100644
--- a/app/src/main/java/com/coocoo/fm/h.java
+++ b/app/src/main/java/com/coocoo/fm/h.java
@@ -29,7 +29,7 @@ final class h extends AsyncTask<String, String, String>
     @Override
     protected String doInBackground(String... strings) {
         final SQLiteDatabase readableDatabase = yo.sql.getReadableDatabase();
-        final StringBuilder sb = new StringBuilder("SELECT data,timestamp,_id,key_id,media_size FROM messages WHERE key_remote_jid='");
+        final StringBuilder sb = new StringBuilder("SELECT _id FROM message WHERE sender_jid_row_id='");
         sb.append(this.c);
         sb.append("'");
         final Cursor rawQuery = readableDatabase.rawQuery(sb.toString(), (String[])null);
@@ -41,7 +41,7 @@ final class h extends AsyncTask<String, String, String>
                 }
             } while (rawQuery.moveToNext());
             rawQuery.close();
-            final StringBuilder sb2 = new StringBuilder("SELECT key_id FROM messages WHERE key_remote_jid='");
+            final StringBuilder sb2 = new StringBuilder("SELECT key_id FROM message WHERE sender_jid_row_id='");
             sb2.append(this.c);
             sb2.append("' AND _id=");
             sb2.append(this.b);
