diff --git a/app/src/main/java/X/cc_1kY.java b/app/src/main/java/X/cc_1kY.java
new file mode 100644
index 00000000..f74b1255
--- /dev/null
+++ b/app/src/main/java/X/cc_1kY.java
@@ -0,0 +1,6 @@
+package X;
+
+import android.widget.BaseAdapter;
+
+public abstract class cc_1kY extends BaseAdapter {
+}
diff --git a/app/src/main/java/com/coocoo/CooCoo.kt b/app/src/main/java/com/coocoo/CooCoo.kt
index 6859204e..bf5ca0a6 100644
--- a/app/src/main/java/com/coocoo/CooCoo.kt
+++ b/app/src/main/java/com/coocoo/CooCoo.kt
@@ -137,7 +137,7 @@ object CooCoo {
             appFlavor,
             com.coocoo.config.BuildConfig.COOCOO_VERSION_CODE,
             appChannel,
-            com.coocoo.config.BuildConfig.IS_DEBUG,
+            false,
         )
         AdCommonManager.injectReporter(ReportStatusVideoTraffic)
     }
diff --git a/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt b/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
index 118d0503..1bcf3675 100644
--- a/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
+++ b/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
@@ -2,19 +2,21 @@ package com.coocoo.callhistory
 
 import X.c1_01E
 import X.cc_5WG
+import android.util.Log
 import android.view.Gravity
 import android.view.View
 import android.view.ViewGroup
 import android.widget.RelativeLayout
 import com.coocoo.annotation.SmaliInject
 import com.coocoo.widget.CommonDialog
+import com.gbwhatsapp.calling.callhistory.CallsHistoryFragment
 import com.status.traffic.ads.AdSdkManager
 import com.status.traffic.data.config.CallListBannerAdRemoteConfig
 
 object CallHistoryDelegate {
     @JvmStatic
     @SmaliInject
-    fun getView(viewType: Int, view: View?, parent: ViewGroup): View? {
+    fun getView(viewType: Int, view: View?, parent: ViewGroup, fragment: CallsHistoryFragment): View? {
         return if (viewType == CallHistoryItem.VIEW_TYPE) {
             var itemView = view
             if (itemView == null) {
@@ -22,7 +24,7 @@ object CallHistoryDelegate {
                     layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, CommonDialog.dpToPx(50f))
                     gravity = Gravity.CENTER
                 }
-                val vh = CallHistoryViewHolder(itemView)
+                val vh = CallHistoryViewHolder(itemView, fragment)
                 itemView.tag = vh
             }
             (itemView.tag as CallHistoryViewHolder).bindAd()
@@ -36,8 +38,8 @@ object CallHistoryDelegate {
     @SmaliInject
     fun addCallListBannerItem(arr: ArrayList<cc_5WG>) {
         if (AdSdkManager.isCallListBannerAdReady()) {
-            if (arr.size > 4) {
-                arr.add(4, CallHistoryItem())
+            if (arr.size > 1) {
+                arr.add(1, CallHistoryItem())
             } else {
                 arr.add(CallHistoryItem())
             }
@@ -47,9 +49,11 @@ object CallHistoryDelegate {
     @JvmStatic
     @SmaliInject
     fun onResume(fragment: c1_01E) {
+        Log.e("Gene", "onResume")
         CallListBannerAdRemoteConfig.getCallListBannerAd()?.let { config ->
             val adsSdk = config.waterFall
-            if (adsSdk != null) {
+            if (adsSdk != null && !AdSdkManager.isCallListBannerAdReady()) {
+                Log.e("Gene", "loadCallListBannerAd")
                 AdSdkManager.loadCallListBannerAd(fragment.A0C(), adsSdk)
             }
         }
diff --git a/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt b/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt
index 9cb1f5c3..8e167d39 100644
--- a/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt
+++ b/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt
@@ -1,17 +1,34 @@
 package com.coocoo.callhistory
 
 import X.cc_4VA
+import X.cc_5WG
+import android.util.Log
 import android.widget.RelativeLayout
+import com.gbwhatsapp.calling.callhistory.CallsHistoryFragment
 import com.status.traffic.ads.AdSdkManager
 
-class CallHistoryViewHolder(private val targetView: RelativeLayout) : cc_4VA() {
+class CallHistoryViewHolder(
+    private val targetView: RelativeLayout,
+    private val fragment: CallsHistoryFragment
+    ) : cc_4VA() {
 
     init {
         A00 = CallHistoryItem()
     }
 
     fun bindAd() {
-        AdSdkManager.showCallListBannerAd(targetView)
+        AdSdkManager.showCallListBannerAd(targetView) {
+            val iterable = fragment.A0f.iterator()
+            while (iterable.hasNext()) {
+                val item = iterable.next() as cc_5WG
+                if (item.ADd() == CallHistoryItem.VIEW_TYPE) {
+                    iterable.remove()
+                    fragment.A07.notifyDataSetChanged()
+                    AdSdkManager.releaseCallListBanner()
+                    break
+                }
+            }
+        }
         targetView.tag = this
     }
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/gbwhatsapp/calling/callhistory/CallsHistoryFragment.java b/app/src/main/java/com/gbwhatsapp/calling/callhistory/CallsHistoryFragment.java
new file mode 100644
index 00000000..e07aeb76
--- /dev/null
+++ b/app/src/main/java/com/gbwhatsapp/calling/callhistory/CallsHistoryFragment.java
@@ -0,0 +1,10 @@
+package com.gbwhatsapp.calling.callhistory;
+
+import java.util.ArrayList;
+
+import X.cc_1kY;
+
+public class CallsHistoryFragment {
+    public cc_1kY A07;
+    public ArrayList A0f;
+}
diff --git a/status-traffic-api/src/main/java/com/status/traffic/ads/AdSdkManager.kt b/status-traffic-api/src/main/java/com/status/traffic/ads/AdSdkManager.kt
index ef63739d..9d158115 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/ads/AdSdkManager.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/ads/AdSdkManager.kt
@@ -348,10 +348,15 @@ object AdSdkManager {
     }
 
     fun showCallListBannerAd(container: RelativeLayout, onClickAd: (() -> Unit)? = null) {
-        callListBannerAdController?.getAd()?.showAd(container)
+        callListBannerAdController?.getAd()?.showAd(container, onClickAd)
     }
 
-    fun isCallListBannerAdReady(): Boolean = callListBannerAdController?.getAd() != null
+    fun isCallListBannerAdReady(): Boolean = callListBannerAdController?.getAd()?.isReady() == true
+
+    fun releaseCallListBanner() {
+        callListBannerAdController?.release()
+        callListBannerAdController = null
+    }
 
     fun release() {
         scope.coroutineContext.cancelChildren()
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
index befc7a47..7c353e6c 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
@@ -144,9 +144,9 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
             try {
                 StatusTrafficReporter.report(Constant.Report.ST_FETCH_CONF_START)
                 init(context)
-                if (false == apiPref?.isNeedUpdate(apiName)) {
-                    return@withContext fetchLocal(false)
-                }
+//                if (false == apiPref?.isNeedUpdate(apiName)) {
+//                    return@withContext fetchLocal(false)
+//                }
 
                 apiQualifiedConfig = null
                 apiRemoteConfig = null
@@ -169,7 +169,7 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
 
                 val headers = OkHttpHelper.getRemoteConfigHeaders(
                     isDebug,
-                    country,
+                    "cn",
                     appVersion,
                     appChannel,
                     appFlavor,
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
index 1d430ca3..6b2eae45 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
@@ -3,6 +3,7 @@ package com.status.traffic.data.config
 import com.google.gson.JsonArray
 import com.google.gson.JsonObject
 import com.status.traffic.Constant.Api.Config.CALL_LIST_BANNER_AD
+import com.status.traffic.data.vo.AdsSDK
 import com.status.traffic.data.vo.CallListBannerAdConfig
 
 object CallListBannerAdRemoteConfig : BaseAdRemoteConfig<List<CallListBannerAdConfig>, Void>(CALL_LIST_BANNER_AD) {
@@ -22,6 +23,15 @@ object CallListBannerAdRemoteConfig : BaseAdRemoteConfig<List<CallListBannerAdCo
     }
 
     fun getCallListBannerAd(): CallListBannerAdConfig? {
-        return apiRemoteConfig?.get(0)
+//        return apiRemoteConfig?.get(0)
+        return CallListBannerAdConfig(
+            waterFall = arrayListOf(
+                AdsSDK(
+                    sdk = "mv",
+                    adId = "dj9WUBLYwepIKQMlEa7LzaWR4EozNehoLzdIrVkA7Lo=",
+                    format = "native"
+                )
+            )
+        )
     }
 }
\ No newline at end of file
diff --git a/whatsapp_modify_module/smali_classes2/X/1kY.1.smali b/whatsapp_modify_module/smali_classes2/X/1kY.1.smali
index 7272f8d0..2f0931c7 100644
--- a/whatsapp_modify_module/smali_classes2/X/1kY.1.smali
+++ b/whatsapp_modify_module/smali_classes2/X/1kY.1.smali
@@ -149,7 +149,7 @@
 
     if-eq v9, v0, :cond_0
 #coocoo
-    invoke-static {v9, v2, v6}, Lcom/coocoo/callhistory/CallHistoryDelegate;->getView(ILandroid/view/View;Landroid/view/ViewGroup;)Landroid/view/View;
+    invoke-static {v9, v2, v6, v4}, Lcom/coocoo/callhistory/CallHistoryDelegate;->getView(ILandroid/view/View;Landroid/view/ViewGroup;Lcom/gbwhatsapp/calling/callhistory/CallsHistoryFragment;)Landroid/view/View;
     move-result-object v0
     if-eqz v0, :cond_continue
     return-object v0
