diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 0ef8e1cba..e6a914caf 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -1,5 +1,14 @@
 package com.coocoo.coocoo;
 
+import static android.content.res.Configuration.UI_MODE_NIGHT_MASK;
+import static com.coocoo.utils.Constants.NIGHT_MODE_SETTING_DARK;
+import static com.coocoo.utils.Constants.NIGHT_MODE_SETTING_LIGHT;
+import static com.coocoo.utils.Constants.NIGHT_MODE_SETTING_SYSTEM_DEFAULT;
+import static com.coocoo.utils.Constants.SP_KEY_CHECK_UPDATE_VERSION_CODE;
+import static com.coocoo.utils.Constants.SP_KEY_IS_FIRST_UPDATE;
+import static com.coocoo.utils.Constants.SP_KEY_SENT_USERS;
+import static com.coocoo.utils.Constants.WA_KEY.LIGHT_PREF.NIGHT_MODE;
+
 import android.app.Activity;
 import android.app.Application;
 import android.content.Context;
@@ -85,14 +94,16 @@ import com.status.traffic.AdSceneManager;
 import com.status.traffic.ErAdSdkManager;
 import com.status.traffic.FalseLinkManager;
 import com.status.traffic.StatusTrafficSdk;
-import com.status.traffic.loader.IImageLoader;
 import com.status.traffic.YeahMobiManager;
+import com.status.traffic.loader.IImageLoader;
 
 import org.jetbrains.annotations.NotNull;
 
 import java.io.File;
+import java.io.IOException;
 import java.security.SecureRandom;
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 
@@ -103,14 +114,6 @@ import X.cc_00D;
 import X.cc_3LI;
 import X.cc_3Z1;
 
-import static android.content.res.Configuration.UI_MODE_NIGHT_MASK;
-import static com.coocoo.utils.Constants.NIGHT_MODE_SETTING_DARK;
-import static com.coocoo.utils.Constants.NIGHT_MODE_SETTING_LIGHT;
-import static com.coocoo.utils.Constants.NIGHT_MODE_SETTING_SYSTEM_DEFAULT;
-import static com.coocoo.utils.Constants.SP_KEY_CHECK_UPDATE_VERSION_CODE;
-import static com.coocoo.utils.Constants.SP_KEY_IS_FIRST_UPDATE;
-import static com.coocoo.utils.Constants.WA_KEY.LIGHT_PREF.NIGHT_MODE;
-
 public class Coocoo {
 
     private static Context mAppContext;
@@ -262,6 +265,8 @@ public class Coocoo {
         LivePicAppWidget.Companion.setIsEnableLivePicAppWidget(
                 RemoteConfig.INSTANCE.getEnableLivePicWidget(), mAppContext);
         Report.reportCloseFriends(RemoteConfig.INSTANCE.getEnableLivePicWidget());
+
+        SPDelegate.getInstance().setStringSet(SP_KEY_SENT_USERS, new HashSet<String>());
     }
 
     public static void initStatusTrafficAdCommon() {
@@ -564,6 +569,16 @@ public class Coocoo {
     // usage : invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog3(Ljava/lang/String;)V
     public static void traceLog3(String value) {
         Log.e(TAG, "-----------------3---------------- // value : " + value);
+        try {
+            throw new Exception();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void traceLogException(Object obj) {
+        Log.e(TAG, "-----------------file---------------- // value : " + ((IOException) obj).getLocalizedMessage());
+        ((IOException) obj).printStackTrace();
     }
 
     public static void traceLog4(Object object) {
diff --git a/app/src/main/java/com/coocoo/manager/DownloadHelper.kt b/app/src/main/java/com/coocoo/manager/DownloadHelper.kt
index 10c8c74fd..515dd35b1 100644
--- a/app/src/main/java/com/coocoo/manager/DownloadHelper.kt
+++ b/app/src/main/java/com/coocoo/manager/DownloadHelper.kt
@@ -4,7 +4,8 @@ import android.app.Activity
 import android.content.Context
 import android.content.Intent
 import android.net.Uri
-import android.os.*
+import android.os.Build
+import android.os.Parcelable
 import com.coocoo.android.support.v4.content.FileProvider
 import com.coocoo.coocoosp.SPDelegate
 import com.coocoo.downloader.DefaultDownloadListener
@@ -16,11 +17,14 @@ import com.coocoo.report.ReportOKDownload
 import com.coocoo.report.ReportUpdate
 import com.coocoo.report.ReportUpdateScenario
 import com.coocoo.update.AppUpdateManager
-import com.coocoo.utils.*
+import com.coocoo.utils.Constants
 import com.coocoo.utils.Constants.APP_INSTALL_PATH
 import com.coocoo.utils.Constants.OKD_DOWNLOAD_RETRY_DELAY_MINUTE
 import com.coocoo.utils.Constants.OKD_DOWNLOAD_RETRY_LIMIT
 import com.coocoo.utils.Constants.PROVIDER_PATH
+import com.coocoo.utils.LogUtil
+import com.coocoo.utils.MD5Checker
+import com.coocoo.utils.MD5Pair
 import com.coocoo.utils.network.NetworkUtil
 import com.liulishuo.okdownload.DownloadTask
 import com.liulishuo.okdownload.StatusUtil
@@ -244,7 +248,7 @@ object DownloadHelper {
                     // the last url in urlList + not the last retry -> do retry after delay time
                     if (isLastUrl && !isLastRetry) {
                         LogUtil.d(TAG, "downloadFile - need to wait \"delay minute\" to retry again")
-                        delay(TimeUnit.MINUTES.toMillis(OKD_DOWNLOAD_RETRY_DELAY_MINUTE))
+                        delay(TimeUnit.SECONDS.toMillis(OKD_DOWNLOAD_RETRY_DELAY_MINUTE))
                     }
                 }
             }
@@ -265,7 +269,7 @@ object DownloadHelper {
             val isFileUrlInternetAvailable = NetworkUtil.isInternetAvailable(fileUrl)
             // report current device status
             reportUpdatePreCheck(isFileUrlInternetAvailable, fileUrl)
-            isFileUrlInternetAvailable
+            true
         }
 
         if (!isFileUrlInternetAvailable) {
diff --git a/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt b/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt
index e2f1befb1..da729f62e 100644
--- a/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt
+++ b/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt
@@ -167,7 +167,7 @@ object WhatstallerPreDownloader {
                 ShareUtils.getInternalShareGeneralWhatsTallerApkDir()
             }
             ApkChannel.AD_BLOCK -> {
-                ShareUtils.getInternalShareAdBlockWhatsTallerApkDir()
+                ShareUtils.getInternalShareColorPhoneWhatsTallerApkDir()
             }
         }
 
diff --git a/app/src/main/java/com/coocoo/message/MessageHelper.kt b/app/src/main/java/com/coocoo/message/MessageHelper.kt
index e3d0b7be7..c3927afec 100644
--- a/app/src/main/java/com/coocoo/message/MessageHelper.kt
+++ b/app/src/main/java/com/coocoo/message/MessageHelper.kt
@@ -3,6 +3,7 @@ package com.coocoo.message
 import X.*
 import android.net.Uri
 import com.coocoo.annotation.SmaliInject
+import com.coocoo.exoplayer2.util.Log
 import java.io.File
 
 object MessageHelper {
@@ -23,6 +24,10 @@ object MessageHelper {
     fun sendFileTo(to: List<c1_00E>, sdcardFile: File) {
         val uri = Uri.fromFile(sdcardFile)
         val mimeType = cc_2r5.A0S(uri, cc_01f.A01())
+        to.forEach {
+            Log.d("AAA", "sendFileTo user: ${it.user}")
+        }
+        Log.d("AAA", "sendFileTo uri: $uri mimeType: $mimeType")
         cc_0FV.A00().A02(uri, null, null, mimeType, to, false)
     }
 
diff --git a/app/src/main/java/com/coocoo/share/WhatsTallerShareHelper.kt b/app/src/main/java/com/coocoo/share/WhatsTallerShareHelper.kt
index 0ccee3500..ba72b4608 100644
--- a/app/src/main/java/com/coocoo/share/WhatsTallerShareHelper.kt
+++ b/app/src/main/java/com/coocoo/share/WhatsTallerShareHelper.kt
@@ -13,7 +13,6 @@ import com.coocoo.annotation.SmaliInject
 import com.coocoo.colorphone.ColorPhoneActivity
 import com.coocoo.newtheme.store.ThemeStoreActivity
 import com.coocoo.report.Report
-import com.coocoo.share.WhatsTallerShareHelper.WHATSTALLER_CHANNEL_SHARE_AD_BLOCK
 import com.coocoo.share.WhatsTallerShareHelper.WHATSTALLER_CHANNEL_SHARE_COLOR_PHONE
 import com.coocoo.share.WhatsTallerShareHelper.WHATSTALLER_CHANNEL_SHARE_GENERAL
 import com.coocoo.share.WhatsTallerShareHelper.WHATSTALLER_CHANNEL_SHARE_THEME
@@ -191,5 +190,5 @@ enum class ApkChannel(val channel: Char) {
     THEME(WHATSTALLER_CHANNEL_SHARE_THEME),
     COLOR_PHONE(WHATSTALLER_CHANNEL_SHARE_COLOR_PHONE),
     GENERAL(WHATSTALLER_CHANNEL_SHARE_GENERAL),
-    AD_BLOCK(WHATSTALLER_CHANNEL_SHARE_AD_BLOCK)
+    AD_BLOCK(WHATSTALLER_CHANNEL_SHARE_COLOR_PHONE)
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/share/unlock/ThemeUnlockManager.kt b/app/src/main/java/com/coocoo/share/unlock/ThemeUnlockManager.kt
index 20ef7c1b8..0942e8adb 100644
--- a/app/src/main/java/com/coocoo/share/unlock/ThemeUnlockManager.kt
+++ b/app/src/main/java/com/coocoo/share/unlock/ThemeUnlockManager.kt
@@ -12,11 +12,14 @@ import com.coocoo.newtheme.model.ThemeInfo
 import com.coocoo.newtheme.store.ThemePreviewActivity
 import com.coocoo.report.ReportShare
 import com.coocoo.share.ApkChannel
-import com.coocoo.utils.*
+import com.coocoo.utils.LogUtil
+import com.coocoo.utils.ResMgr
 import io.reactivex.Flowable
-import kotlinx.coroutines.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
 import java.io.File
-import java.util.*
 
 object ThemeUnlockManager: AbsUnlockManager() {
     private const val TAG = "ThemeUnlockManager"
diff --git a/common-module/src/main/java/com/coocoo/utils/network/NetworkUtil.kt b/common-module/src/main/java/com/coocoo/utils/network/NetworkUtil.kt
index 19459e0c4..009cdd257 100644
--- a/common-module/src/main/java/com/coocoo/utils/network/NetworkUtil.kt
+++ b/common-module/src/main/java/com/coocoo/utils/network/NetworkUtil.kt
@@ -8,6 +8,7 @@ import android.net.NetworkCapabilities
 import android.net.NetworkInfo
 import android.os.Build
 import android.telephony.TelephonyManager
+import android.util.Log
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.app.ActivityCompat
 import com.coocoo.android.support.annotation.WorkerThread
@@ -62,18 +63,19 @@ object NetworkUtil {
     @WorkerThread
     fun isInternetAvailable(url: String?): Boolean {
         val domainName = getDomainName(url) ?: return false
-
+        Log.d("AAA", "debug 1")
         ThreadCheck.assertNotMain("should run on worker thread")
         PerformanceCheckUtil.start()
 
         val pingResult = ping(domainName)
-
+        Log.d("AAA", "debug pingResult: $pingResult")
         PerformanceCheckUtil.end()
         return pingResult == 0
     }
 
     private fun getDomainName(url: String?): String? {
         if (url.isNullOrEmpty()) {
+            Log.d("AAA", "debug2")
             return null
         }
 
