diff --git a/app/src/main/java/com/coocoo/activity/AdBlockShareActivity.kt b/app/src/main/java/com/coocoo/activity/AdBlockShareActivity.kt
index 3b10377bd..a4db926ff 100644
--- a/app/src/main/java/com/coocoo/activity/AdBlockShareActivity.kt
+++ b/app/src/main/java/com/coocoo/activity/AdBlockShareActivity.kt
@@ -3,13 +3,18 @@ package com.coocoo.activity
 import android.app.Activity
 import android.content.Intent
 import android.os.Bundle
+import android.view.View
 import android.widget.ImageView
+import com.coocoo.exoplayer2.util.Log
 import com.coocoo.utils.ResMgr
+import com.coocoowhatsapp.ContactPicker
 
 class AdBlockShareActivity : Activity() {
 
     companion object {
 
+        private const val REQUEST_CODE = 10
+
         fun startActivity(activity: Activity) {
             val intent = Intent(activity, AdBlockShareActivity::class.java)
             activity.startActivity(intent)
@@ -18,19 +23,39 @@ class AdBlockShareActivity : Activity() {
 
     private lateinit var btnClose: ImageView
 
+    private lateinit var btnShare: View
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(ResMgr.getLayoutId("cc_activity_ad_block_share"))
 
         btnClose = findViewById(ResMgr.getId("cc_close"))
-
+        btnShare = findViewById(ResMgr.getId("cc_btn_share"))
 
         initView()
     }
 
     private fun initView() {
         btnClose.setOnClickListener {
+            finish()
+        }
+
+        btnShare.setOnClickListener {
+            shareMessage()
+        }
+    }
+
+    private fun shareMessage() {
+        val intent = Intent(this, ContactPicker::class.java).apply {
+            putExtra(Intent.EXTRA_TEXT, "Message Test")
+        }
+        startActivityForResult(intent, REQUEST_CODE)
+    }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (resultCode == Activity.RESULT_OK && requestCode == REQUEST_CODE) {
+            Log.d("AAA", "AdBlockShareActivity onActivityResult")
+            finish()
         }
     }
 }
\ No newline at end of file
