diff --git a/app/src/main/java/com/coocoo/backuprestore/FullBackupManager.kt b/app/src/main/java/com/coocoo/backuprestore/FullBackupManager.kt
index 097696082..6680d3778 100644
--- a/app/src/main/java/com/coocoo/backuprestore/FullBackupManager.kt
+++ b/app/src/main/java/com/coocoo/backuprestore/FullBackupManager.kt
@@ -64,7 +64,7 @@ object FullBackupManager {
         ResMgr.getString(Constants.Res.String.APP_NAME), FOLDER_CURRENT_BACKUP)
     private val PATH_MOD_BACKUP_V2_ZIP =
         FileUtil.concatPaths(FOLDER_CURRENT_BACKUP_V2, Constants.FILE_BACKUP_ZIP)
-    private val PATH_MOD_DATA_DATA_BACKUP = FileUtil.concatPaths(FOLDER_CURRENT_BACKUP_V2, DATA_FOLDER_PATTERN)
+    val PATH_MOD_DATA_DATA_BACKUP: String = FileUtil.concatPaths(FOLDER_CURRENT_BACKUP_V2, DATA_FOLDER_PATTERN)
     private val PATH_MOD_DATA_DATA = FileUtil.concatPaths(PATH_SDCARD_DATA, DATA_FOLDER_PATTERN, yo.mpack)
 
     // Theme info
diff --git a/app/src/main/java/com/coocoo/preference/AbsCoocooPrefFragment.kt b/app/src/main/java/com/coocoo/preference/AbsCoocooPrefFragment.kt
index 35696e5ae..d378a5123 100644
--- a/app/src/main/java/com/coocoo/preference/AbsCoocooPrefFragment.kt
+++ b/app/src/main/java/com/coocoo/preference/AbsCoocooPrefFragment.kt
@@ -28,7 +28,7 @@ abstract class AbsCooCooPrefFragment : PreferenceFragment() {
     private var confirmRestartDialog: Dialog? = null
     private var listView: ListView? = null
 
-    private val prefFragmentScope: CoroutineScope
+    protected val prefFragmentScope: CoroutineScope
         get() {
             return CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
         }
diff --git a/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt b/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
index 6ff991b77..6606a7a59 100644
--- a/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
+++ b/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
@@ -8,6 +8,7 @@ import android.text.TextUtils
 import android.view.View
 import android.widget.Toast
 import com.coocoo.backuprestore.*
+import com.coocoo.backuprestore.FullBackupManager.PATH_MOD_DATA_DATA_BACKUP
 import com.coocoo.backuprestore.dialog.ConfirmSignOutDialog
 import com.coocoo.coocoo.Coocoo
 import com.coocoo.coocoosp.SPDelegate
@@ -18,11 +19,13 @@ import com.coocoo.report.ReportConstant
 import com.coocoo.utils.*
 import com.coocoo.utils.ToastUtil.showToast
 import com.coocoo.widget.ConfirmDialog
+import com.coocoo.widget.CyclicProgressDialog
 import com.coocoo.widget.ProgressDialog
 import com.coocoo.widget.SimpleDialog
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import java.lang.ref.WeakReference
 
 class CoocooBackupRestorePrefFragment
@@ -148,7 +151,18 @@ class CoocooBackupRestorePrefFragment
         findPreference(PREF_KEY_CLEAR_BACKUP)?.let { preference ->
             preference.onPreferenceClickListener = this@CoocooBackupRestorePrefFragment
             preference.onPreferenceChangeListener = this@CoocooBackupRestorePrefFragment
-            preference.isEnabled = FullBackupManager.lastLocalBackupTime.isNotEmpty()
+            preference.isEnabled = false
+            val strCalculating = ResMgr.getString("cc_backup_whatsapp_data_clear_backup_calculating")
+            preference.summary = ResMgr.getString("cc_backup_whatsapp_data_clear_backup_content", strCalculating)
+            prefFragmentScope.launch {
+                val size = getFormattedBackUpFileSize()
+                preference.summary = if (size > 0) {
+                    preference.isEnabled = true
+                    ResMgr.getString("cc_backup_whatsapp_data_clear_backup_content", "%.2f MB".format(size))
+                } else {
+                    ResMgr.getString("cc_backup_whatsapp_data_clear_backup_content", "0 MB")
+                }
+            }
         }
 
         setupGoogleDriveTitlePreference()
@@ -180,7 +194,7 @@ class CoocooBackupRestorePrefFragment
                         context = this@CoocooBackupRestorePrefFragment.activity,
                         title = ResMgr.getString("cc_delete_now_title"),
                         message = ResMgr.getString("cc_delete_now_content"),
-                        onConfirmClicked = {},
+                        onConfirmClicked = ::confirmDeleting,
                     ).also {
                         confirmDialog = it
                     }.show()
@@ -407,6 +421,31 @@ class CoocooBackupRestorePrefFragment
         }
     }
 
+    private fun confirmDeleting(view: View) {
+        val dialog = CyclicProgressDialog(
+            context = activity,
+            title = ResMgr.getString("cc_backup_whatsapp_data_clear_backup_dialog_deleting_title"),
+            message = ResMgr.getString("cc_universal_settings_please_wait"),
+        )
+        dialog.show()
+        prefFragmentScope.launch {
+            deleteBackUpFiles()
+            dialog.dismiss()
+            SPDelegate.getInstance().setLongValue(
+                Constants.PREFERENCE_KEY_BACKUP_LOCAL_LAST_TIME,
+                -1
+            )
+        }
+    }
+
+    private suspend fun deleteBackUpFiles() = withContext(Dispatchers.IO) {
+        FileUtil.deleteDir(PATH_MOD_DATA_DATA_BACKUP)
+    }
+
+    private suspend fun getFormattedBackUpFileSize(): Float = withContext(Dispatchers.IO) {
+        FileUtil.getFileSize(PATH_MOD_DATA_DATA_BACKUP) / (1024f * 1024f)
+    }
+
     private fun generateLocalBackupPreferenceSummary(localBackupTime: String): String {
         val never: String = ResMgr.getString(Constants.Res.String.BACKUP_RESTORE_NEVER)
         val time = String.format(
diff --git a/app/src/main/java/com/coocoo/widget/ProgressDialog.kt b/app/src/main/java/com/coocoo/widget/ProgressDialog.kt
index 1a5703a47..693138978 100644
--- a/app/src/main/java/com/coocoo/widget/ProgressDialog.kt
+++ b/app/src/main/java/com/coocoo/widget/ProgressDialog.kt
@@ -2,6 +2,7 @@ package com.coocoo.widget
 
 import android.content.Context
 import android.util.Log
+import android.view.View
 import android.widget.ProgressBar
 import android.widget.TextView
 import com.coocoo.utils.Constants
@@ -13,7 +14,8 @@ import kotlinx.coroutines.launch
 class ProgressDialog(
         context: Context,
         private val title: String?,
-        private val message: String?
+        private val message: String?,
+        private val isIndeterminate: Boolean = false
 ) : CommonDialog(context) {
 
     override fun show() {
@@ -38,6 +40,13 @@ class ProgressDialog(
         progressBar?.run {
             progress = 0
         }
+        
+        val indeterminateProgressBar = findViewById<ProgressBar>(ResMgr.getId("cc_dialog_progress"))
+        if (isIndeterminate) {
+            indeterminateProgressBar.visibility = View.GONE
+        } else {
+            progressBar.visibility = View.GONE
+        }
     }
 
     fun updateProgress(newProgress: Int) {
diff --git a/app/src/main/res/layout/cc_dialog_progress.xml b/app/src/main/res/layout/cc_dialog_progress.xml
index a3863d1fa..67732045c 100644
--- a/app/src/main/res/layout/cc_dialog_progress.xml
+++ b/app/src/main/res/layout/cc_dialog_progress.xml
@@ -30,7 +30,15 @@
         android:progressDrawable="@drawable/cc_update_progress"
         android:layout_marginTop="25dp"
         android:layout_marginStart="35dp"
-        android:layout_marginEnd="35dp"/>
+        android:layout_marginEnd="35dp"
+        tools:visibility="gone"/>
+
+    <ProgressBar
+        android:id="@+id/cc_cyclic_dialog_progress"
+        android:layout_width="24dp"
+        android:layout_height="24dp"
+        android:layout_marginTop="25dp"
+        android:layout_gravity="center_horizontal"/>
 
     <com.coocoo.widget.CoocooTextView
         android:id="@+id/cc_dialog_progress_text"
@@ -43,7 +51,8 @@
         android:layout_marginRight="35dp"
         android:textColor="#06979F"
         android:layout_gravity="center_horizontal"
-        tools:text="40%"/>
+        tools:text="40%"
+        tools:visibility="gone"/>
 
     <View
         android:layout_width="match_parent"
diff --git a/app/src/main/res/values/cc_strings.xml b/app/src/main/res/values/cc_strings.xml
index 1d83a0e90..44a7b5718 100644
--- a/app/src/main/res/values/cc_strings.xml
+++ b/app/src/main/res/values/cc_strings.xml
@@ -856,7 +856,9 @@ Our official Website: http://heymods.com"</string>
     <string name="cc_backup_whatsapp_data_desc_time_cloud">Backup Time: %1$s</string>
     <string name="cc_backup_whatsapp_data_desc_size">Backup Size: %1$s</string>
     <string name="cc_backup_whatsapp_data_clear_backup_title">Clear Backup</string>
-    <string name="cc_backup_whatsapp_data_clear_backup_content">The Clear Backup option deletes all old backup data. More space saving.</string>
+    <string name="cc_backup_whatsapp_data_clear_backup_content">Size: %s\nThe Clear Backup option deletes all old backup data. More space saving.</string>
+    <string name="cc_backup_whatsapp_data_clear_backup_calculating">Calculating&#8320;</string>
+    <string name="cc_backup_whatsapp_data_clear_backup_dialog_deleting_title">Deleting Backup files…</string>
     <string name="cc_backup_whatsapp_data_title_include_media">Include media</string>
     <string name="cc_backup_whatsapp_data_desc_include_media">Enable this option may occupy more storage on your device or cloud storage.</string>
     <string name="cc_restore_whatsApp_data_title">Restore WhatsApp Data</string>
