diff --git a/app/src/main/java/com/coocoo/AppForegroundManager.kt b/app/src/main/java/com/coocoo/AppForegroundManager.kt
index ceeb49517..9263d47c9 100644
--- a/app/src/main/java/com/coocoo/AppForegroundManager.kt
+++ b/app/src/main/java/com/coocoo/AppForegroundManager.kt
@@ -3,7 +3,6 @@ package com.coocoo
 import android.app.Activity
 import android.app.Application
 import android.os.Bundle
-import com.coocoo.activity.ExpiredWarningActivity
 import com.coocoo.report.Report
 import com.coocoo.update.AppUpdateManager
 import com.coocoowhatsapp.HomeActivity
@@ -21,12 +20,6 @@ object AppForegroundManager : Application.ActivityLifecycleCallbacks {
 
     override fun onActivityResumed(activity: Activity?) {
         isAppInForeground = true
-        // not process : ExpiredWarningActivity
-        // process by itself : HomeActivity
-        if (AppUpdateManager.shouldShowWarningActivity() &&
-                !(activity is ExpiredWarningActivity || activity is HomeActivity)) {
-            ExpiredWarningActivity.startActivity();
-        }
 
         if (activity is HomeActivity) {
             isHomeActivityInForeground = true
diff --git a/app/src/main/java/com/coocoo/activity/ExpiredWarningActivity.kt b/app/src/main/java/com/coocoo/activity/ExpiredWarningActivity.kt
deleted file mode 100644
index eaff63143..000000000
--- a/app/src/main/java/com/coocoo/activity/ExpiredWarningActivity.kt
+++ /dev/null
@@ -1,160 +0,0 @@
-package com.coocoo.activity
-
-import android.app.Activity
-import android.content.Intent
-import android.os.Bundle
-import android.view.LayoutInflater
-import android.view.View
-import android.widget.ImageView
-import android.widget.LinearLayout
-import android.widget.TextView
-import com.coocoo.CooCooApp
-import com.coocoo.backuprestore.*
-import com.coocoo.backuprestore.FullBackupManager.startBackup
-import com.coocoo.incrementalupdate.IncrementalUpdateManager
-import com.coocoo.report.ReportConstant
-import com.coocoo.report.ReportUpdate
-import com.coocoo.update.AppUpdateManager
-import com.coocoo.update.UpdateService
-import com.coocoo.update.bean.UpdateData
-import com.coocoo.utils.AppInfoUtil
-import com.coocoo.utils.DialogUtils.showDialogSafely
-import com.coocoo.utils.ResMgr
-import com.coocoo.utils.StringUtil
-import com.coocoo.utils.format
-import com.coocoo.widget.ForceUpdateBackupDialog
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
-import java.lang.ref.WeakReference
-
-class ExpiredWarningActivity : Activity() {
-
-    companion object {
-        private const val TAG = "ExpiredWarningActivity"
-        fun startActivity() {
-            val intent = Intent(CooCooApp.getAppContext(), ExpiredWarningActivity::class.java)
-            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
-            CooCooApp.getAppContext().startActivity(intent)
-        }
-    }
-
-    private var updateData: UpdateData? = null
-    private var updateBtn: TextView? = null
-
-    /**
-     * Backup progress listener.
-     */
-    private val backupListener: BackupRestoreListener by lazy {
-        BackupRestoreListenerHelper.createBackupProgressDialogListener(this)
-    }
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        parseIntent(intent)
-
-        setContentView(ResMgr.getLayoutId("cc_activity_expired_warning"))
-        setupViews()
-
-        reportStrategyInterstitialActivity(ReportConstant.ACTION_SHOW)
-    }
-
-    override fun onResume() {
-        super.onResume()
-        AppUpdateManager.notifyAutoInstallIfNotLatestVersion(this)
-
-
-        GlobalScope.launch(Dispatchers.IO) {
-            withContext(Dispatchers.Main) {
-                updateBtn?.text = ResMgr.getString("cc_update_text")
-
-                val bottomHint = ResMgr.findViewById<TextView>("cc_expired_bottom_hint", this@ExpiredWarningActivity)
-                bottomHint?.run {
-                    visibility = View.GONE
-                    text = ""
-                }
-            }
-        }
-    }
-
-    override fun onPause() {
-        super.onPause()
-        AppUpdateManager.cancelNotifyAutoInstallOrGoToWeb()
-    }
-
-    override fun onBackPressed() {
-        super.onBackPressed()
-        AppUpdateManager.closeWarningActivity()
-        // Send report when user click system back key
-        reportStrategyInterstitialActivity(ReportConstant.ACTION_CLOSE)
-    }
-
-    private fun parseIntent(intent: Intent?) {
-        if (intent == null) {
-            return
-        }
-    }
-
-    private fun setupViews() {
-        val titleContent = StringUtil.format("cc_warning_activity_title", AppInfoUtil.getApplicationName())
-        val title = ResMgr.findViewById<TextView>("cc_expired_title", this)
-        title?.text = titleContent
-
-        val subTitleContent = StringUtil.format("cc_warning_activity_subtitle", AppInfoUtil.getApplicationName())
-        val subTitle = ResMgr.findViewById<TextView>("cc_expired_sub_title", this)
-        subTitle?.text = subTitleContent
-
-        val closeBtn = ResMgr.findViewById<ImageView>("close_self_btn", this)
-        closeBtn?.setOnClickListener {
-            onBackPressed()
-        }
-
-        updateData = AppUpdateManager.getUpdateData()
-        updateBtn = ResMgr.findViewById<TextView>("cc_update_now", this)
-        updateBtn?.run {
-            setOnClickListener {
-                reportStrategyInterstitialActivity(ReportConstant.ACTION_UPDATE)
-                if (updateData?.updateInformation == null) {
-                    AppUpdateManager.gotoOfficialWebsite(this@ExpiredWarningActivity)
-                } else {
-                    AppUpdateManager.startUpdateServiceOrInstallApk(this@ExpiredWarningActivity,
-                            UpdateService.START_UPDATE_SERVICE_FROM_WARNING_ACTIVITY)
-                }
-            }
-        }
-
-        val descContainer = ResMgr.findViewById<LinearLayout>("desc_container", this)
-        val inflater = LayoutInflater.from(this)
-        val warnings = AppUpdateManager.getAllWarnings()
-        for (warning in warnings) {
-            val contentRow = inflater.inflate(ResMgr.getLayoutId("cc_activity_expired_warning_content_row"), null)
-            val tv = ResMgr.findViewById<TextView>("cc_warning_row_tv", contentRow)
-            tv?.text = warning
-            descContainer.addView(contentRow)
-        }
-
-        val backup = ResMgr.findViewById<TextView>("cc_expired_backup", this)
-        backup?.setOnClickListener {
-            val backupDialog = ForceUpdateBackupDialog(this) {
-                // Start backup
-                startBackup(WeakReference(backupListener), ReportConstant.FROM_INTERSTITIAL_ACTIVITY)
-                // Report backup action
-                reportStrategyInterstitialActivity(ReportConstant.ACTION_BACKUP)
-            }
-            showDialogSafely(this, backupDialog)
-        }
-    }
-
-    /**
-     * Report the [action] and [isApkDownloaded] in IO thread as it's a time-consuming
-     * task to check if calculate MD5.
-     */
-    private fun reportStrategyInterstitialActivity(action: String) {
-        GlobalScope.launch(Dispatchers.IO) {
-            val md5Data = AppUpdateManager.getUpdateData()?.updateInformation?.completeUpdatePackageMd5Pair
-            val isApkDownloaded = IncrementalUpdateManager.isApkFullyDownloadedSuspend(md5Data)
-            ReportUpdate.strategyInterstitialActivity(action, isApkDownloaded)
-        }
-    }
-}
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/report/ReportConstant.java b/app/src/main/java/com/coocoo/report/ReportConstant.java
index 34423b902..36134e072 100644
--- a/app/src/main/java/com/coocoo/report/ReportConstant.java
+++ b/app/src/main/java/com/coocoo/report/ReportConstant.java
@@ -282,7 +282,6 @@ public interface ReportConstant {
     String STRATEGY_COMMON = "common";
 
     //Update Interstitial Activity
-    String EVENT_UPDATE_STRATEGY_INTERSTITIAL_ACTIVITY = "update_interstitial_activity";
     String ACTION_UPDATE = "update";
 
     //Message Scheduler
diff --git a/app/src/main/java/com/coocoo/report/ReportCore.java b/app/src/main/java/com/coocoo/report/ReportCore.java
index d1b39ec44..fb312ad8c 100644
--- a/app/src/main/java/com/coocoo/report/ReportCore.java
+++ b/app/src/main/java/com/coocoo/report/ReportCore.java
@@ -525,13 +525,6 @@ class ReportCore implements ReportConstant {
         report(EVENT_UPDATE_STRATEGY_SNACK_BAR, params);
     }
 
-    static void updateStrategyInterstitialActivity(String action, boolean isApkDownloaded) {
-        Map<String, String> params = new HashMap<>();
-        params.put(KEY_ACTION, action);
-        params.put(KEY_DOWNLOAD_SUCCESS, isApkDownloaded ? "1" : "0");
-        report(EVENT_UPDATE_STRATEGY_INTERSTITIAL_ACTIVITY, params);
-    }
-
     static void clickSettingToUpdate(
             String action,
             boolean hasLabel,
diff --git a/app/src/main/java/com/coocoo/report/ReportUpdate.java b/app/src/main/java/com/coocoo/report/ReportUpdate.java
index 89c0d9ebc..afc28f689 100644
--- a/app/src/main/java/com/coocoo/report/ReportUpdate.java
+++ b/app/src/main/java/com/coocoo/report/ReportUpdate.java
@@ -1,5 +1,35 @@
 package com.coocoo.report;
 
+import static com.coocoo.manager.DownloadHelper.DOWNLOAD_MODULE_ORI;
+import static com.coocoo.report.ReportConstant.EVENT_INTERNAL_DOWNLOAD_MODULE;
+import static com.coocoo.report.ReportConstant.EVENT_INTERNAL_UPDATE_FROM;
+import static com.coocoo.report.ReportConstant.EVENT_INTERNAL_UPDATE_NETWORK;
+import static com.coocoo.report.ReportConstant.EVENT_INTERNAL_UPDATE_RESULT;
+import static com.coocoo.report.ReportConstant.EVENT_INTERNAL_UPDATE_SUCCESS;
+import static com.coocoo.report.ReportConstant.EVENT_INTERNAL_UPDATE_THREAD;
+import static com.coocoo.report.ReportConstant.EVENT_INTERNAL_UPDATE_TYPE;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE_APP_CODE;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE_APP_IS_FIRST;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE_APP_NAME;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE_APP_SUCCESS;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE_DIALOG_SHOW;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE_DOWNLOAD_FLOW_ACTION;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE_DOWNLOAD_FLOW_ERROR;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE_DOWNLOAD_FLOW_SHOW;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE_DOWNLOAD_FLOW_SPEED;
+import static com.coocoo.report.ReportConstant.EVENT_UPDATE_DOWNLOAD_FLOW_SPEND;
+import static com.coocoo.report.ReportConstant.KEY_DIFF_TIME;
+import static com.coocoo.report.ReportConstant.KEY_SCENARIO;
+import static com.coocoo.report.ReportConstant.KEY_UPDATE_CLICK;
+import static com.coocoo.report.ReportConstant.KEY_UPDATE_STATUS;
+import static com.coocoo.report.ReportConstant.KEY_UPDATE_STATUS_MESSAGE;
+import static com.coocoo.report.ReportConstant.VALUE_MAKE_INCREMENTAL_UPDATE_APK_BASE_APK_MD5;
+import static com.coocoo.report.ReportConstant.VALUE_MAKE_INCREMENTAL_UPDATE_APK_PATCH_MD5;
+import static com.coocoo.report.ReportConstant.VALUE_STRATEGY_ACTIVE;
+import static com.coocoo.report.ReportConstant.VALUE_STRATEGY_NORMAL;
+import static com.coocoo.report.ReportConstant.VALUE_STRATEGY_RADICAL;
+
 import android.text.TextUtils;
 
 import com.coocoo.CooCooApp;
@@ -9,14 +39,10 @@ import com.coocoo.permission.PermissionManager;
 import com.coocoo.update.StrategyType;
 import com.coocoo.utils.VersionUtil;
 import com.coocoo.utils.network.NetworkUtil;
-import com.flurry.android.FlurryAgent;
 
 import java.util.HashMap;
 import java.util.Map;
 
-import static com.coocoo.manager.DownloadHelper.DOWNLOAD_MODULE_ORI;
-import static com.coocoo.report.ReportConstant.*;
-
 public class ReportUpdate {
 
     public static final String UPDATE_DLG_ACTION_CLOSE = "closeButton";
@@ -201,10 +227,6 @@ public class ReportUpdate {
         ReportCore.updateStrategySnackBar(action, strategy, isApkDownloaded);
     }
 
-    public static void strategyInterstitialActivity(String action, boolean isApkDownloaded) {
-        ReportCore.updateStrategyInterstitialActivity(action, isApkDownloaded);
-    }
-
     public static void clickSettingToUpdate(
         String action,
         boolean hasLabel,
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
index 4649f18ff..625ee7ade 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
@@ -57,7 +57,6 @@ import com.bumptech.glide.request.target.Target;
 import com.coocoo.AppForegroundManager;
 import com.coocoo.CooCooApp;
 import com.coocoo.activity.CoocooSecureQuestionActivity;
-import com.coocoo.activity.ExpiredWarningActivity;
 import com.coocoo.activity.MessageSchedulerActivity;
 import com.coocoo.activity.OpenWhatsAppActivity;
 import com.coocoo.activity.PrivateConversationsActivity;
@@ -1089,9 +1088,7 @@ public class HomeActivityDelegate extends ActivityDelegate<HomeActivityThemeMana
             boolean isBaseApkExpired = AppUpdateManager.INSTANCE.isBaseApkExpired();
             if (isForceUpdate || isBaseApkExpired) {
                 UpdateForceActivity.startForceUpdateActivity(updateData);
-            } else if (AppUpdateManager.INSTANCE.shouldShowWarningActivity()) {
-                ExpiredWarningActivity.Companion.startActivity();
-            } else if (updateData.isNeedUpdate() && isApkFullyDownload(updateData)) {
+            }  else if (updateData.isNeedUpdate() && isApkFullyDownload(updateData)) {
                 showUpdateDialog(updateData, false);
             } else if (SurveyQuestDialog.canShow()) {
                 showSurveyQuestDialog();
diff --git a/app/src/main/res/drawable-xxhdpi/cc_btn_expired_warning_appbar_closed_n.png b/app/src/main/res/drawable-xxhdpi/cc_btn_expired_warning_appbar_closed_n.png
deleted file mode 100644
index bb26e0a89..000000000
Binary files a/app/src/main/res/drawable-xxhdpi/cc_btn_expired_warning_appbar_closed_n.png and /dev/null differ
diff --git a/app/src/main/res/drawable-xxhdpi/cc_img_wa_expired.webp b/app/src/main/res/drawable-xxhdpi/cc_img_wa_expired.webp
deleted file mode 100755
index ffef8c8f4..000000000
Binary files a/app/src/main/res/drawable-xxhdpi/cc_img_wa_expired.webp and /dev/null differ
diff --git a/app/src/main/res/layout/cc_activity_expired_warning.xml b/app/src/main/res/layout/cc_activity_expired_warning.xml
deleted file mode 100644
index 8010cc9e9..000000000
--- a/app/src/main/res/layout/cc_activity_expired_warning.xml
+++ /dev/null
@@ -1,103 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:background="#FFFF4E54"
-    android:orientation="vertical">
-
-    <ImageView
-        android:id="@+id/close_self_btn"
-        android:layout_width="24dp"
-        android:layout_height="24dp"
-        android:layout_marginTop="16dp"
-        android:layout_marginStart="16dp"
-        android:src="@drawable/cc_btn_expired_warning_appbar_closed_n"/>
-
-    <ImageView
-        android:id="@+id/img_wa_expired"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_marginTop="98dp"
-        android:layout_centerHorizontal="true"
-        android:src="@drawable/cc_img_wa_expired"/>
-
-    <com.coocoo.widget.CoocooTextView
-        android:id="@+id/cc_expired_backup"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_marginBottom="48dp"
-        android:layout_marginTop="16dp"
-        android:fontFamily="sans-serif-medium"
-        android:text="@string/cc_update_force_backup"
-        android:textColor="#FFFFFFFF"
-        android:textSize="14sp"
-        android:layout_centerHorizontal="true"
-        android:layout_alignParentBottom="true" />
-
-    <com.coocoo.widget.CoocooTextView
-        android:id="@+id/cc_update_now"
-        android:layout_width="232dp"
-        android:layout_height="36dp"
-        android:layout_marginTop="16dp"
-        android:layout_marginBottom="10dp"
-        android:fontFamily="sans-serif-medium"
-        android:gravity="center"
-        android:text="@string/cc_update_text"
-        android:textColor="#FFFF5043"
-        android:textSize="13sp"
-        android:textAllCaps="true"
-        android:background="@drawable/cc_warning_bar_update_btn_bg"
-        android:layout_centerHorizontal="true"
-        android:layout_above="@id/cc_expired_backup"/>
-
-    <com.coocoo.widget.CoocooTextView
-        android:id="@+id/cc_expired_bottom_hint"
-        android:layout_width="288dp"
-        android:layout_height="wrap_content"
-        android:layout_above="@id/cc_update_now"
-        android:layout_marginTop="16dp"
-        android:layout_centerHorizontal="true"
-        android:text="@string/cc_update_speedy_hint"
-        android:textColor="#FFFFFFFF"
-        android:textSize="13sp"
-        android:gravity="center"
-        android:visibility="gone"
-        tools:visibility="visible"/>
-    
-    <com.coocoo.widget.CoocooTextView
-        android:id="@+id/cc_expired_title"
-        android:layout_width="288dp"
-        android:layout_height="wrap_content"
-        android:layout_below="@id/img_wa_expired"
-        android:layout_marginTop="53dp"
-        android:layout_centerHorizontal="true"
-        android:textColor="#FFFFFFFF"
-        android:textStyle="bold"
-        android:textSize="18sp"/>
-
-    <com.coocoo.widget.CoocooTextView
-        android:id="@+id/cc_expired_sub_title"
-        android:layout_width="288dp"
-        android:layout_height="wrap_content"
-        android:layout_below="@id/cc_expired_title"
-        android:layout_marginTop="24dp"
-        android:layout_centerHorizontal="true"
-        android:textColor="#FFFFFFFF"
-        android:textSize="13sp"/>
-
-    <ScrollView
-        android:layout_width="288dp"
-        android:layout_height="match_parent"
-        android:layout_below="@id/cc_expired_sub_title"
-        android:layout_above="@id/cc_expired_bottom_hint"
-        android:layout_centerHorizontal="true">
-        <LinearLayout
-            android:id="@+id/desc_container"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:orientation="vertical"/>
-    </ScrollView>
-
-</RelativeLayout>
diff --git a/app/src/main/res/layout/cc_activity_expired_warning_content_row.xml b/app/src/main/res/layout/cc_activity_expired_warning_content_row.xml
deleted file mode 100644
index be3c90e84..000000000
--- a/app/src/main/res/layout/cc_activity_expired_warning_content_row.xml
+++ /dev/null
@@ -1,25 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="wrap_content"
-    android:orientation="horizontal">
-
-    <com.coocoo.widget.CoocooTextView
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_gravity="center_horizontal"
-        android:text="- "
-        android:textColor="#FFFFFFFF"
-        android:textSize="13sp"
-        />
-    
-    <com.coocoo.widget.CoocooTextView
-        android:id="@+id/cc_warning_row_tv"
-        android:layout_width="288dp"
-        android:layout_height="wrap_content"
-        android:textColor="#FFFFFFFF"
-        android:textSize="13sp"/>
-
-</LinearLayout>
diff --git a/script/remove_ad/AndroidManifest.xml b/script/remove_ad/AndroidManifest.xml
index 4147fe220..cf090a891 100644
--- a/script/remove_ad/AndroidManifest.xml
+++ b/script/remove_ad/AndroidManifest.xml
@@ -2601,14 +2601,6 @@
             </intent-filter>
         </activity>
 
-        <!-- #coocoo -->
-       <activity
-           android:name="com.coocoo.activity.ExpiredWarningActivity"
-           android:screenOrientation="portrait"
-           android:launchMode="singleTop"
-           android:theme="@style/Theme.App.NoActionBar" />
-        <!-- #coocoo -->
-
         <activity-alias
             android:name="com.coocoo.activity.Alias01"
             android:enabled="false"
diff --git a/script/remove_ad/AndroidManifest.xml.bk b/script/remove_ad/AndroidManifest.xml.bk
index ca401b827..0d3dc67d3 100644
--- a/script/remove_ad/AndroidManifest.xml.bk
+++ b/script/remove_ad/AndroidManifest.xml.bk
@@ -2629,14 +2629,6 @@
             </intent-filter>
         </activity>
 
-        <!-- #coocoo -->
-       <activity
-           android:name="com.coocoo.activity.ExpiredWarningActivity"
-           android:screenOrientation="portrait"
-           android:launchMode="singleTop"
-           android:theme="@style/Theme.App.NoActionBar" />
-        <!-- #coocoo -->
-
         <activity-alias
             android:name="com.coocoo.activity.Alias01"
             android:enabled="false"
diff --git a/whatsapp_modify_module/AndroidManifest.xml b/whatsapp_modify_module/AndroidManifest.xml
index 0a04c3fd7..4eb33edbb 100644
--- a/whatsapp_modify_module/AndroidManifest.xml
+++ b/whatsapp_modify_module/AndroidManifest.xml
@@ -1081,10 +1081,6 @@
             </intent-filter>
         </activity>
 
-        <!-- #coocoo -->
-        <activity android:name="com.coocoo.activity.ExpiredWarningActivity" android:launchMode="singleTop" android:screenOrientation="portrait" android:theme="@style/Theme.App.NoActionBar" />
-        <!-- #coocoo -->
-
         <activity-alias android:name="com.coocoo.activity.Alias01" android:enabled="false" android:icon="@drawable/cc_icon_01" android:label="@string/cc_app_name" android:targetActivity="com.coocoowhatsapp.Main">
             <intent-filter android:label="@string/launcher_app_name">
                 <action android:name="android.intent.action.MAIN" />
