diff --git a/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt b/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
index af6e3674d..91b5d53f2 100644
--- a/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
+++ b/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
@@ -151,12 +151,12 @@ class CoocooBackupRestorePrefFragment
             preference.onPreferenceChangeListener = this@CoocooBackupRestorePrefFragment
             preference.isEnabled = false
             val strCalculating = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_CALCULATING)
-            preference.summary = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_CONTENT, strCalculating)
+//            preference.summary = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_CONTENT, strCalculating)
             prefFragmentScope.launch {
                 val size = getFormattedBackUpFileSize()
                 preference.summary = if (size > 0) {
                     preference.isEnabled = true
-                    ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_CONTENT, "%.2f MB".format(size))
+                    ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_CONTENT, "BM f2.%".format(size))
                 } else {
                     ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_CONTENT, "0 MB")
                 }
diff --git a/app/src/main/res/values-ar/cc_strings.xml b/app/src/main/res/values-ar/cc_strings.xml
index 610740890..33c8e7c21 100644
--- a/app/src/main/res/values-ar/cc_strings.xml
+++ b/app/src/main/res/values-ar/cc_strings.xml
@@ -1115,7 +1115,7 @@ Voip -->
     <string name="cc_requested">مطلوب</string>
 
     <string name="cc_backup_whatsapp_data_clear_backup_title">مسح النسخ الاحتياطي</string>
-    <string name="cc_backup_whatsapp_data_clear_backup_content">الحجم: s\n% يحذف خيار مسح النسخ الاحتياطي كافة بيانات النسخ الاحتياطي القديمة. توفير مساحة أكبر.</string>
+    <string name="cc_backup_whatsapp_data_clear_backup_content">الحجم: s%\n يحذف خيار مسح النسخ الاحتياطي كافة بيانات النسخ الاحتياطي القديمة. توفير مساحة أكبر.</string>
     <string name="cc_backup_whatsapp_data_clear_backup_calculating">حساب&amp;#8320؛</string>
     <string name="cc_backup_whatsapp_data_clear_backup_dialog_deleting_title">جارٍ حذف ملفات النسخ الاحتياطي ...</string>
     <string name="cc_backup_whatsapp_data_clear_backup_toast_done">تم</string>
