diff --git a/app/build.gradle b/app/build.gradle
index 522dd566f..989cbbee8 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -316,7 +316,7 @@ dependencies {
 
     implementation(name: 'sublive_modsdk', ext: 'aar') // v0.5.0
     implementation 'org.greenrobot:eventbus:3.2.0'
-    implementation 'com.google.android.gms:play-services-ads:18.0.0'
+    implementation 'com.google.android.gms:play-services-ads:19.8.0'
 }
 
 gradle.taskGraph.whenReady {
diff --git a/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt b/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt
index d9ea2e382..ec22540f2 100644
--- a/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt
+++ b/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt
@@ -37,7 +37,7 @@ import com.coocoo.transform.CircleTransform
 import com.coocoo.utils.*
 import com.coocoo.wallpapers.model.WallpaperBean
 import com.coocoo.whatsappdelegate.HomeActivityDelegate
-import com.google.android.gms.ads.AdView
+import com.google.android.gms.ads.*
 import com.status.traffic.AdCommonManager
 import com.status.traffic.AdSceneManager
 import com.status.traffic.StatusTrafficSdk
@@ -851,3 +851,24 @@ class SdkCarouselConversationItem(
         }
     }
 }
+
+fun loadNativeAd(context: Context) {
+    val builder = AdLoader.Builder(context, "ca-app-pub-3940256099942544/2247696110")
+    builder.forNativeAd { nativeAd ->
+        LogUtil.d("loadNativeAd $nativeAd")
+    }.withAdListener(object : AdListener() {
+        override fun onAdFailedToLoad(p0: LoadAdError) {
+            LogUtil.d("loadNativeAd onAdFailedToLoad")
+        }
+    })
+    val adLoader = builder.build()
+    adLoader.loadAd(AdRequest.Builder().build())
+}
+
+fun loadAppOpen() {
+
+}
+
+fun loadInterstitial() {
+
+}
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
index 79a9459a3..25b96df54 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
@@ -42,6 +42,7 @@ import com.coocoo.conversation.item.NewcomerPromoteFeatureAction;
 import com.coocoo.conversation.item.NewcomerPromoteFeatureAdapter;
 import com.coocoo.conversation.item.RetentionConversationItem;
 import com.coocoo.conversation.item.RetentionConversationItemClickable;
+import com.coocoo.conversation.item.RetentionConversationItemKt;
 import com.coocoo.conversation.presenter.ConversationsFragmentPresenter;
 import com.coocoo.conversation.presenter.IConversationsFragmentPresenter;
 import com.coocoo.conversation.repository.MyMessageRepository;
@@ -343,6 +344,7 @@ public class ConversationsFragmentDelegate extends FragmentDelegate<Conversation
         adView.setAdSize(AdSize.BANNER);
         adView.loadAd(new AdRequest.Builder().build());
         conversationBannerItem = new ConversationBannerItem(adView);
+        RetentionConversationItemKt.loadNativeAd(Coocoo.getAppContext());
         Report.reportConversationBannerAdReqStart(bannerId);
     }
 
