diff --git a/app/src/main/java/com/coocoo/CooCoo.kt b/app/src/main/java/com/coocoo/CooCoo.kt
index 6859204e..bf5ca0a6 100644
--- a/app/src/main/java/com/coocoo/CooCoo.kt
+++ b/app/src/main/java/com/coocoo/CooCoo.kt
@@ -137,7 +137,7 @@ object CooCoo {
             appFlavor,
             com.coocoo.config.BuildConfig.COOCOO_VERSION_CODE,
             appChannel,
-            com.coocoo.config.BuildConfig.IS_DEBUG,
+            false,
         )
         AdCommonManager.injectReporter(ReportStatusVideoTraffic)
     }
diff --git a/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt b/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
index 5ab3c2d1..a48e1c4a 100644
--- a/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
+++ b/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
@@ -1,6 +1,7 @@
 package com.coocoo.callhistory
 
 import X.cc_5WG
+import android.util.Log
 import android.view.Gravity
 import android.view.View
 import android.view.ViewGroup
@@ -8,6 +9,7 @@ import android.widget.RelativeLayout
 import com.coocoo.annotation.SmaliInject
 import com.coocoo.widget.CommonDialog
 import com.gbwhatsapp.calling.callhistory.CallsHistoryFragment
+import com.startapp.r
 import com.status.traffic.StatusTrafficSdk
 import com.status.traffic.ads.AdSdkManager
 import com.status.traffic.data.config.CallListBannerAdRemoteConfig
@@ -36,7 +38,9 @@ object CallHistoryDelegate {
     @JvmStatic
     @SmaliInject
     fun addCallListBannerItem(arr: ArrayList<cc_5WG>) {
-        if (StatusTrafficSdk.getInstance().isBannerLoaded(AdSdkManager.BannerPlacement.CALL_LIST)) {
+        Log.e("Gene", "addCallListBannerItem $arr")
+        if (StatusTrafficSdk.getInstance().isBannerLoaded(AdSdkManager.BannerPlacement.CALL_LIST)
+            && !arr.containsCallListBanner()) {
             if (arr.size > 4) {
                 arr.add(4, CallHistoryItem())
             } else {
@@ -45,9 +49,14 @@ object CallHistoryDelegate {
         }
     }
 
+    private fun ArrayList<cc_5WG>.containsCallListBanner(): Boolean {
+        return find { it.ADd() == CallHistoryItem.VIEW_TYPE } != null
+    }
+
     @JvmStatic
     @SmaliInject
     fun onResume(fragment: CallsHistoryFragment) {
+        Log.e("Gene", "CallHistoryDelegate onResume")
         StatusTrafficSdk.getInstance().loadSdkBannerAd(fragment.A0C(), AdSdkManager.BannerPlacement.CALL_LIST) {
             addCallListBannerItem(fragment.A0f)
             fragment.A07.notifyDataSetChanged()
diff --git a/app/src/main/java/com/coocoo/manager/StatusesManager.java b/app/src/main/java/com/coocoo/manager/StatusesManager.java
index c41ed464..964f30a4 100644
--- a/app/src/main/java/com/coocoo/manager/StatusesManager.java
+++ b/app/src/main/java/com/coocoo/manager/StatusesManager.java
@@ -238,7 +238,7 @@ public final class StatusesManager implements Preference.OnPreferenceChangeListe
                 CooCoo.initStatusTrafficSdk();
             }
             if (!activity.isDestroyed() && !activity.isFinishing()) {
-                statusTrafficSdk.loadSdkBannerAd(activity, AdSdkManager.BannerPlacement.STATUS_LIST, null);
+//                statusTrafficSdk.loadSdkBannerAd(activity, AdSdkManager.BannerPlacement.STATUS_LIST, null);
             }
         });
     }
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt
index 8f6f736d..0c4581ff 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt
@@ -282,7 +282,7 @@ class HomeActivityDelegate @SmaliInject constructor(private val homeActivity: Ho
                 sdk.loadSdkBannerAd(mHostActivity, AdSdkManager.BannerPlacement.HOME_LIST)
             }
             if (!sdk.isBannerLoaded(AdSdkManager.BannerPlacement.HOME_TOP)) {
-                sdk.loadSdkBannerAd(mHostActivity, AdSdkManager.BannerPlacement.HOME_TOP)
+//                sdk.loadSdkBannerAd(mHostActivity, AdSdkManager.BannerPlacement.HOME_TOP)
             } else {
                 setHomeTopBannerAd(true)
                 sdk.setBannerAutoRefresh(AdSdkManager.BannerPlacement.HOME_TOP, true)
diff --git a/status-traffic-api/src/main/java/com/status/traffic/ads/AdSdkManager.kt b/status-traffic-api/src/main/java/com/status/traffic/ads/AdSdkManager.kt
index 975bea2c..f19da96e 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/ads/AdSdkManager.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/ads/AdSdkManager.kt
@@ -150,13 +150,14 @@ object AdSdkManager {
         val waterfall = config.waterFall ?: return
         if (waterfall.isEmpty()) return
 
-        val interval = System.currentTimeMillis() - placement.controller.lastLadingTime
-        if (interval < 0 || interval > (config.retryIntervalSec * 1000)) {
+//        val interval = System.currentTimeMillis() - placement.controller.lastLadingTime
+//        if (interval < 0 || interval > (config.retryIntervalSec * 1000)) {
             if (placement.controller.getAd()?.isReady() == true
                 || placement.controller.isLoading
             ) {
                 return
             }
+
             placement.controller.loadAd(activity, waterfall, object : OnLoadAdListener{
                 override suspend fun onLoaded() {
                     onAdLoaded?.invoke()
@@ -165,11 +166,11 @@ object AdSdkManager {
                 override suspend fun onFailedToLoad() {
                 }
             })
-        }
+//        }
     }
 
     fun showBannerAd(
-        placement: BannerPlacement,
+        placement: AdSdkManager.BannerPlacement,
         bannerContainer: RelativeLayout,
         onClickAd: (() -> Unit)?
     ) {
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
index befc7a47..69841666 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
@@ -169,7 +169,7 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
 
                 val headers = OkHttpHelper.getRemoteConfigHeaders(
                     isDebug,
-                    country,
+                    "cn",
                     appVersion,
                     appChannel,
                     appFlavor,
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
index 43a683b8..5a2139ae 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
@@ -3,9 +3,11 @@ package com.status.traffic.data.config
 import com.google.gson.JsonArray
 import com.google.gson.JsonObject
 import com.status.traffic.Constant.Api.Config.CALL_LIST_BANNER_AD
+import com.status.traffic.data.vo.AdsSDK
 import com.status.traffic.data.vo.SdkBannerConfig
 
-object CallListBannerAdRemoteConfig : BaseAdRemoteConfig<List<SdkBannerConfig>, Void>(CALL_LIST_BANNER_AD) {
+object CallListBannerAdRemoteConfig :
+    BaseAdRemoteConfig<List<SdkBannerConfig>, Void>(CALL_LIST_BANNER_AD) {
     override fun parserAdsInfo(json: JsonArray?): List<SdkBannerConfig>? {
         try {
             return getGson().fromJson(json, Array<SdkBannerConfig>::class.java).toList()
@@ -22,6 +24,15 @@ object CallListBannerAdRemoteConfig : BaseAdRemoteConfig<List<SdkBannerConfig>,
     }
 
     fun getCallListBannerAd(): SdkBannerConfig? {
-        return apiRemoteConfig?.get(0)
+//        return apiRemoteConfig?.get(0)
+        return SdkBannerConfig(
+            waterFall = arrayListOf(
+                AdsSDK(
+                    sdk = "mv",
+                    adId = "dj9WUBLYwepIKQMlEa7LzaWR4EozNehoLzdIrVkA7Lo=",
+                    format = "native"
+                )
+            )
+        )
     }
 }
\ No newline at end of file
