diff --git a/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt b/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt
index 6bba4fd89..e2f1befb1 100644
--- a/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt
+++ b/app/src/main/java/com/coocoo/manager/WhatstallerPreDownloader.kt
@@ -44,6 +44,7 @@ object WhatstallerPreDownloader {
         downloadTasks.add(DownloadTask(ApkChannel.THEME))
         downloadTasks.add(DownloadTask(ApkChannel.COLOR_PHONE))
         downloadTasks.add(DownloadTask(ApkChannel.GENERAL))
+        downloadTasks.add(DownloadTask(ApkChannel.AD_BLOCK))
     }
 
     fun handleDownload() {
@@ -165,6 +166,9 @@ object WhatstallerPreDownloader {
             ApkChannel.GENERAL -> {
                 ShareUtils.getInternalShareGeneralWhatsTallerApkDir()
             }
+            ApkChannel.AD_BLOCK -> {
+                ShareUtils.getInternalShareAdBlockWhatsTallerApkDir()
+            }
         }
 
         return@withContext apkDict?.run {
diff --git a/app/src/main/java/com/coocoo/share/WhatsTallerShareHelper.kt b/app/src/main/java/com/coocoo/share/WhatsTallerShareHelper.kt
index 79d78c3f3..cfcacc9d9 100644
--- a/app/src/main/java/com/coocoo/share/WhatsTallerShareHelper.kt
+++ b/app/src/main/java/com/coocoo/share/WhatsTallerShareHelper.kt
@@ -13,11 +13,12 @@ import com.coocoo.annotation.SmaliInject
 import com.coocoo.colorphone.ColorPhoneActivity
 import com.coocoo.newtheme.store.ThemeStoreActivity
 import com.coocoo.report.Report
-import com.coocoo.share.unlock.ColorPhoneUnlockManager
-import com.coocoo.share.unlock.ThemeUnlockManager
+import com.coocoo.share.WhatsTallerShareHelper.WHATSTALLER_CHANNEL_SHARE_AD_BLOCK
 import com.coocoo.share.WhatsTallerShareHelper.WHATSTALLER_CHANNEL_SHARE_COLOR_PHONE
 import com.coocoo.share.WhatsTallerShareHelper.WHATSTALLER_CHANNEL_SHARE_GENERAL
 import com.coocoo.share.WhatsTallerShareHelper.WHATSTALLER_CHANNEL_SHARE_THEME
+import com.coocoo.share.unlock.ColorPhoneUnlockManager
+import com.coocoo.share.unlock.ThemeUnlockManager
 import com.coocoo.utils.Constants
 import com.coocoo.utils.DialogUtils.showDialogSafely
 import com.coocoo.utils.LogUtil
@@ -36,6 +37,7 @@ object WhatsTallerShareHelper {
     internal const val WHATSTALLER_CHANNEL_SHARE_THEME = 's'
     internal const val WHATSTALLER_CHANNEL_SHARE_COLOR_PHONE = 'c'
     internal const val WHATSTALLER_CHANNEL_SHARE_GENERAL = 'n'
+    internal const val WHATSTALLER_CHANNEL_SHARE_AD_BLOCK = 'a' //TODO adblock: lack of channel name
 
     private val coocooModsPackageNameList = listOf(
             "com.whatsapp",
@@ -182,5 +184,6 @@ object WhatsTallerShareHelper {
 enum class ApkChannel(val channel: Char) {
     THEME(WHATSTALLER_CHANNEL_SHARE_THEME),
     COLOR_PHONE(WHATSTALLER_CHANNEL_SHARE_COLOR_PHONE),
-    GENERAL(WHATSTALLER_CHANNEL_SHARE_GENERAL)
+    GENERAL(WHATSTALLER_CHANNEL_SHARE_GENERAL),
+    AD_BLOCK(WHATSTALLER_CHANNEL_SHARE_AD_BLOCK)
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/share/unlock/AdBlockShareUnlockManager.kt b/app/src/main/java/com/coocoo/share/unlock/AdBlockShareUnlockManager.kt
index e11271741..29be6d1d5 100644
--- a/app/src/main/java/com/coocoo/share/unlock/AdBlockShareUnlockManager.kt
+++ b/app/src/main/java/com/coocoo/share/unlock/AdBlockShareUnlockManager.kt
@@ -1,20 +1,42 @@
 package com.coocoo.share.unlock
 
 import X.c1_00E
+import com.coocoo.manager.WhatstallerPreDownloader
+import com.coocoo.share.ApkChannel
 import com.coocoo.utils.ResMgr
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import java.io.File
 
 object AdBlockShareUnlockManager : AbsUnlockManager() {
 
     const val KEY: String = "AdBlockShareUnlockManager"
 
+    private val managerScope: CoroutineScope by lazy { CoroutineScope(SupervisorJob() + Dispatchers.Main) }
+
     override fun handleSendMessage(shareData: Any?, jids: ArrayList<String>, sourceText: String?) {
         val targetUser = jids.map { jid ->
             c1_00E.A01(jid)
         }
 
-        handleSendTextMessage(targetUser, ResMgr.getString("cc_ad_block_share_message"))
-
-        showSharedSuccessfulToast()
+        var message = ResMgr.getString("cc_ad_block_share_message")
+        var apk: File? = null
+        managerScope.launch {
+            val hasDownload = WhatstallerPreDownloader.hasDownloaded(ApkChannel.AD_BLOCK)
+            if (hasDownload) {
+                apk = WhatstallerPreDownloader.getFile(ApkChannel.AD_BLOCK)
+            } else {
+                message += "\nTap to install GBWhatsApp\uD83D\uDC47\nhttps://go.gbwhatsapp.blog/GBWhatsApp_2022a.apk"
+            }
+
+            handleSendTextMessage(targetUser, message)
+            apk?.let {
+                handleSendApkFileMessage(targetUser, it)
+            }
+            showSharedSuccessfulToast()
+        }
     }
 
     override fun getUnlockTag(): String = KEY
diff --git a/app/src/main/java/com/coocoo/utils/ShareUtils.kt b/app/src/main/java/com/coocoo/utils/ShareUtils.kt
index f1bc7e181..258cd0ca5 100644
--- a/app/src/main/java/com/coocoo/utils/ShareUtils.kt
+++ b/app/src/main/java/com/coocoo/utils/ShareUtils.kt
@@ -92,6 +92,13 @@ object ShareUtils {
         }
     }
 
+    // ad block (related to AdBlockShare)
+    fun getInternalShareAdBlockWhatsTallerApkDir(): File? {
+        return Coocoo.getAppContext()?.run {
+            File(cacheDir, "$MOD${File.separator}ad_block_whatstaller_apk")
+        }
+    }
+
     @JvmStatic
     fun getInternalShareImageDir(): File? {
         return Coocoo.getAppContext()?.run {
