diff --git a/app/src/main/java/com/coocoo/backuprestore/FullBackupManager.kt b/app/src/main/java/com/coocoo/backuprestore/FullBackupManager.kt
index 097696082..6680d3778 100644
--- a/app/src/main/java/com/coocoo/backuprestore/FullBackupManager.kt
+++ b/app/src/main/java/com/coocoo/backuprestore/FullBackupManager.kt
@@ -64,7 +64,7 @@ object FullBackupManager {
         ResMgr.getString(Constants.Res.String.APP_NAME), FOLDER_CURRENT_BACKUP)
     private val PATH_MOD_BACKUP_V2_ZIP =
         FileUtil.concatPaths(FOLDER_CURRENT_BACKUP_V2, Constants.FILE_BACKUP_ZIP)
-    private val PATH_MOD_DATA_DATA_BACKUP = FileUtil.concatPaths(FOLDER_CURRENT_BACKUP_V2, DATA_FOLDER_PATTERN)
+    val PATH_MOD_DATA_DATA_BACKUP: String = FileUtil.concatPaths(FOLDER_CURRENT_BACKUP_V2, DATA_FOLDER_PATTERN)
     private val PATH_MOD_DATA_DATA = FileUtil.concatPaths(PATH_SDCARD_DATA, DATA_FOLDER_PATTERN, yo.mpack)
 
     // Theme info
diff --git a/app/src/main/java/com/coocoo/preference/AbsCoocooPrefFragment.kt b/app/src/main/java/com/coocoo/preference/AbsCoocooPrefFragment.kt
index 35696e5ae..d378a5123 100644
--- a/app/src/main/java/com/coocoo/preference/AbsCoocooPrefFragment.kt
+++ b/app/src/main/java/com/coocoo/preference/AbsCoocooPrefFragment.kt
@@ -28,7 +28,7 @@ abstract class AbsCooCooPrefFragment : PreferenceFragment() {
     private var confirmRestartDialog: Dialog? = null
     private var listView: ListView? = null
 
-    private val prefFragmentScope: CoroutineScope
+    protected val prefFragmentScope: CoroutineScope
         get() {
             return CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
         }
diff --git a/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt b/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
index 6ff991b77..36bdbe272 100644
--- a/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
+++ b/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
@@ -8,6 +8,7 @@ import android.text.TextUtils
 import android.view.View
 import android.widget.Toast
 import com.coocoo.backuprestore.*
+import com.coocoo.backuprestore.FullBackupManager.PATH_MOD_DATA_DATA_BACKUP
 import com.coocoo.backuprestore.dialog.ConfirmSignOutDialog
 import com.coocoo.coocoo.Coocoo
 import com.coocoo.coocoosp.SPDelegate
@@ -23,6 +24,7 @@ import com.coocoo.widget.SimpleDialog
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import java.lang.ref.WeakReference
 
 class CoocooBackupRestorePrefFragment
@@ -148,7 +150,15 @@ class CoocooBackupRestorePrefFragment
         findPreference(PREF_KEY_CLEAR_BACKUP)?.let { preference ->
             preference.onPreferenceClickListener = this@CoocooBackupRestorePrefFragment
             preference.onPreferenceChangeListener = this@CoocooBackupRestorePrefFragment
-            preference.isEnabled = FullBackupManager.lastLocalBackupTime.isNotEmpty()
+            preference.isEnabled = false
+            if (FullBackupManager.lastLocalBackupTime.isNotEmpty()) {
+                preference.summary = ResMgr.getString("cc_backup_whatsapp_data_clear_backup_calculating")
+                prefFragmentScope.launch {
+                    preference.isEnabled = true
+                    val size = getBackUpFileSize()
+                    preference.summary = ResMgr.getString("cc_backup_whatsapp_data_clear_backup_content", size)
+                }
+            }
         }
 
         setupGoogleDriveTitlePreference()
@@ -407,6 +417,13 @@ class CoocooBackupRestorePrefFragment
         }
     }
 
+    private fun deleteBackUpFiles(view: View) {
+    }
+
+    private suspend fun getBackUpFileSize(): Float = withContext(Dispatchers.IO) {
+        FileUtil.getFileSize(PATH_MOD_DATA_DATA_BACKUP) / (1024f * 1024f)
+    }
+
     private fun generateLocalBackupPreferenceSummary(localBackupTime: String): String {
         val never: String = ResMgr.getString(Constants.Res.String.BACKUP_RESTORE_NEVER)
         val time = String.format(
diff --git a/app/src/main/res/values/cc_strings.xml b/app/src/main/res/values/cc_strings.xml
index 1d83a0e90..7839d17ee 100644
--- a/app/src/main/res/values/cc_strings.xml
+++ b/app/src/main/res/values/cc_strings.xml
@@ -856,7 +856,8 @@ Our official Website: http://heymods.com"</string>
     <string name="cc_backup_whatsapp_data_desc_time_cloud">Backup Time: %1$s</string>
     <string name="cc_backup_whatsapp_data_desc_size">Backup Size: %1$s</string>
     <string name="cc_backup_whatsapp_data_clear_backup_title">Clear Backup</string>
-    <string name="cc_backup_whatsapp_data_clear_backup_content">The Clear Backup option deletes all old backup data. More space saving.</string>
+    <string name="cc_backup_whatsapp_data_clear_backup_content">Size: %.2f MB\nThe Clear Backup option deletes all old backup data. More space saving.</string>
+    <string name="cc_backup_whatsapp_data_clear_backup_calculating">Calculating&#8239;</string>
     <string name="cc_backup_whatsapp_data_title_include_media">Include media</string>
     <string name="cc_backup_whatsapp_data_desc_include_media">Enable this option may occupy more storage on your device or cloud storage.</string>
     <string name="cc_restore_whatsApp_data_title">Restore WhatsApp Data</string>
