diff --git a/common-module/src/main/java/com/coocoo/utils/DialogUtils.kt b/common-module/src/main/java/com/coocoo/utils/DialogUtils.kt
index ca0c1a4d1..bc0e73e87 100644
--- a/common-module/src/main/java/com/coocoo/utils/DialogUtils.kt
+++ b/common-module/src/main/java/com/coocoo/utils/DialogUtils.kt
@@ -43,5 +43,7 @@ object DialogUtils {
 
 
 fun Dialog.safelyShow() {
-    showDialogSafely(context as Activity, this)
+    ownerActivity?.let {
+        showDialogSafely(it, this)
+    }
 }
