diff --git a/app/src/main/java/com/coocoo/CooCooAppShell.kt b/app/src/main/java/com/coocoo/CooCooAppShell.kt
index ed84640..13138e2 100644
--- a/app/src/main/java/com/coocoo/CooCooAppShell.kt
+++ b/app/src/main/java/com/coocoo/CooCooAppShell.kt
@@ -5,11 +5,14 @@ import android.os.Bundle
 import com.coocoo.coocoosp.SPDelegate
 import com.coocoo.manager.PerformanceAnalyticsManager
 import com.coocoo.report.Report
+import com.coocoo.utils.LogUtil
 import com.coocoo.utils.ResMgr
 import com.gbwhatsapp.AppShell
-import com.gbwhatsapp.yo.yo
 
 class CooCooAppShell : AppShell() {
+    companion object {
+        private const val FLURRY_APP_KEY = "4G4SG6ZXQ6PJ5HK9PNFG"
+    }
 
     override fun onCreate() {
         CooCoo.context = this
@@ -53,5 +56,6 @@ class CooCooAppShell : AppShell() {
         Report.init(this)
         SPDelegate.getInstance().init(this)
         ResMgr.init(this)
+        LogUtil.d(FLURRY_APP_KEY)
     }
 }
\ No newline at end of file
