diff --git a/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt b/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt
index 24fc257db..4f1b8349d 100644
--- a/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt
+++ b/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt
@@ -114,6 +114,7 @@ class StatusHeaderViewHolder(
             A0R = arrayListOf(1)
             A00()
         }
+        intent.putExtra(IS_FROM_STATUS_FEED, true)
         fragment.startActivityForResult(intent, REQUEST_CODE)
     }
 
@@ -131,5 +132,6 @@ class StatusHeaderViewHolder(
 
     companion object {
         private const val REQUEST_CODE = 0x2
+        const val IS_FROM_STATUS_FEED = "is_from_status_feed"
     }
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
new file mode 100644
index 000000000..0dcf692db
--- /dev/null
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
@@ -0,0 +1,13 @@
+package com.coocoo.whatsappdelegate
+
+import com.coocoo.annotation.SmaliInject
+import com.coocoo.utils.LogUtil
+import com.coocoowhatsapp.contact.picker.ContactPickerFragment
+
+object ContactPickerSendBtnDelegate {
+    @SmaliInject
+    @JvmStatic
+    fun onSendBtnClick(fragment: ContactPickerFragment) {
+        LogUtil.d("onSendBtnClick")
+    }
+}
\ No newline at end of file
diff --git a/whatsapp_modify_module/smali/X/1Fe.1.smali b/whatsapp_modify_module/smali/X/1Fe.1.smali
index eea1a1953..37372812c 100644
--- a/whatsapp_modify_module/smali/X/1Fe.1.smali
+++ b/whatsapp_modify_module/smali/X/1Fe.1.smali
@@ -26,6 +26,9 @@
 
     .line 220959
     iget-object v4, p0, LX/1Fe;->A00:Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;
+#coocoo
+    invoke-static {v4}, Lcom/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate;->onSendBtnClick(Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;)V
+#coocoo
 
 #coocoo
     invoke-virtual {v4}, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->A0A()LX/01K;
