diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt
index c7b33d828..51701144b 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt
@@ -10,6 +10,7 @@ import android.hardware.Camera
 import android.net.ConnectivityManager
 import android.opengl.GLSurfaceView
 import android.opengl.Matrix
+import android.os.Build
 import android.os.Handler
 import android.view.MotionEvent
 import android.view.View
@@ -49,13 +50,7 @@ import com.coocoo.newtheme.store.model.DownloadUnInitialised
 import com.coocoo.newtheme.store.model.Downloading
 import com.coocoo.report.Report
 import com.coocoo.report.ReportConstant
-import com.coocoo.utils.Constants
-import com.coocoo.utils.DialogUtils
-import com.coocoo.utils.LogUtil
-import com.coocoo.utils.ResMgr
-import com.coocoo.utils.ToastUtil
-import com.coocoo.utils.UnzipError
-import com.coocoo.utils.UnzipResult
+import com.coocoo.utils.*
 import com.coocoo.utils.network.NetworkUtil
 import com.coocoo.widget.CameraUiFilterControlLayout
 import com.coocoo.widget.CameraUiFilterSwitchLayout
@@ -700,9 +695,15 @@ class CameraUiDelegate(val activity: cc_0Ay, val uiHelperProvider: cc_0MI) {
         if (show) {
             uiHelperProvider.AAN().A05.visibility = View.VISIBLE
         }
+        val parent = (uiHelperProvider.AAN().A0C.parent as View)
         uiHelperProvider.AAN().A0C.visibility = if (show) View.VISIBLE else View.GONE
         uiHelperProvider.AAN().A0C.alpha = if (show) 1.0F else 0.0F
         uiHelperProvider.AAN().A06.visibility = if (show) View.INVISIBLE else View.VISIBLE  // Hide the camera_control view at showing the gallery view.
+        uiHelperProvider.AAN().A05.postDelayed({
+            LogUtil.d("1 parent.visibility: ${parent.visibility}, alpha: ${parent.alpha}, show: ${show}")
+            uiHelperProvider.AAN().A05.visibility = View.VISIBLE
+            LogUtil.d("2 parent.visibility: ${parent.visibility}, alpha: ${parent.alpha}")
+        },1000)
     }
 
     private fun switchFilter() {
diff --git a/whatsapp_modify_module/smali/X/0GV.smali b/whatsapp_modify_module/smali/X/0GV.smali
index ce59cd6c8..ab53aa4fa 100644
--- a/whatsapp_modify_module/smali/X/0GV.smali
+++ b/whatsapp_modify_module/smali/X/0GV.smali
@@ -1793,7 +1793,7 @@
 
     const-wide/16 v0, 0x96
 
-    const/16 v3, 0x8
+    const/16 v3, 0x0
 
     if-nez v5, :cond_3
 
@@ -6028,8 +6028,9 @@
 
     .line 64544
     :cond_1
-    const/16 v1, 0x8
-
+#coocoo
+    const/16 v1, 0x0
+#coocoo
     if-eq v0, v1, :cond_2
 
     .line 64545
