diff --git a/app/src/main/java/com/coocoo/openchat/activity/OpenChatWebActivity.kt b/app/src/main/java/com/coocoo/openchat/activity/OpenChatWebActivity.kt
index 32599765c..95b9c07db 100644
--- a/app/src/main/java/com/coocoo/openchat/activity/OpenChatWebActivity.kt
+++ b/app/src/main/java/com/coocoo/openchat/activity/OpenChatWebActivity.kt
@@ -36,6 +36,7 @@ import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.MainScope
 import kotlinx.coroutines.launch
 import java.lang.ref.WeakReference
+import java.net.URL
 
 class OpenChatWebActivity : cc_01X() {
 
@@ -287,7 +288,8 @@ class OpenChatWebActivity : cc_01X() {
 
     private fun handleExternalUrl(url: String?): Boolean {
         url?.let {
-            if (!it.startsWith(WhisperOpenChatViewModel.OPENCHAT_HOME_URL) &&
+            LogUtil.d("$it ${WhisperOpenChatViewModel.OPENCHAT_HOME_URL}")
+            if (!it.contains(URL(WhisperOpenChatViewModel.OPENCHAT_HOME_URL).host) &&
                 !viewModel.isMatchHandleUrl(it)
             ) {
                 return try {
