diff --git a/app/build.gradle b/app/build.gradle
index 522dd566f..64d651d1f 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -316,7 +316,7 @@ dependencies {
 
     implementation(name: 'sublive_modsdk', ext: 'aar') // v0.5.0
     implementation 'org.greenrobot:eventbus:3.2.0'
-    implementation 'com.google.android.gms:play-services-ads:18.0.0'
+    implementation 'com.google.android.gms:play-services-ads:20.3.0'
 }
 
 gradle.taskGraph.whenReady {
diff --git a/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt b/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt
index d9ea2e382..b968cb237 100644
--- a/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt
+++ b/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt
@@ -38,6 +38,7 @@ import com.coocoo.utils.*
 import com.coocoo.wallpapers.model.WallpaperBean
 import com.coocoo.whatsappdelegate.HomeActivityDelegate
 import com.google.android.gms.ads.AdView
+import com.google.android.gms.ads.nativead.NativeAdView
 import com.status.traffic.AdCommonManager
 import com.status.traffic.AdSceneManager
 import com.status.traffic.StatusTrafficSdk
@@ -769,45 +770,22 @@ class AppletsConversationItem(val title : String, val des: String, val pushMessa
     }
 }
 
-class ConversationBannerItem(
-    private val adView: AdView?) : RetentionConversationItem() {
+class ConversationBannerItem(private val adView: NativeAdView) : RetentionConversationItem() {
 
     var isAdLoadSuccess = false
 
     fun resume() {
-        adView?.resume()
     }
 
     fun pause() {
-        adView?.pause()
     }
 
     fun destroy() {
-        adView?.destroy()
+//        adView.destroy()
     }
 
     override fun getView(parent: ViewGroup): View {
-        if (adView?.parent != null) {
-            return adView.parent as View
-        }
-
-        val frameLayout = FrameLayout(CooCooApp.getAppContext())
-        val params: ViewGroup.LayoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
-        frameLayout.layoutParams = params
-        if (!isAdLoadSuccess) {
-            return frameLayout
-        }
-
-        if (adView != null) {
-            val childParams: FrameLayout.LayoutParams = FrameLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT)
-            childParams.width = adView.adSize.getWidthInPixels(CooCooApp.getAppContext())
-            childParams.height = adView.adSize.getHeightInPixels(CooCooApp.getAppContext())
-            childParams.gravity = Gravity.CENTER
-            frameLayout.addView(adView, childParams)
-        }
-
-
-        return frameLayout
+        return adView
     }
 }
 
diff --git a/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfig.kt b/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfig.kt
index edad124a8..ecf01838b 100644
--- a/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfig.kt
+++ b/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfig.kt
@@ -991,6 +991,7 @@ object RemoteConfig {
     val conversationBannerAd: String
         @JvmStatic
         get() {
-            return getString(RemoteConfigKeys.CONVERSATION_BANNER)
+//            return getString(RemoteConfigKeys.CONVERSATION_BANNER)
+            return "ca-app-pub-3940256099942544/2247696110"
         }
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfigKeys.kt b/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfigKeys.kt
index affcc597e..d80114505 100644
--- a/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfigKeys.kt
+++ b/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfigKeys.kt
@@ -124,7 +124,7 @@ object RemoteConfigKeys {
     const val THEME_STORE_BANNER = "modad_theme_store_banner"
 
     // Conversation banner ad
-    const val CONVERSATION_BANNER = "modad_homelist_banner"
+    const val CONVERSATION_BANNER = "modad_homelist_banner_native"
     const val APPLETS_PUSH_LAST_CODE = "applets_push_switch_last_code"
     const val APPLETS_BLOCK_PUSH_MESSAGE = "applets_block_push_message"
     const val APPLETS_PUSH_MESSAGE_VERSION = "applets_push_message_version"
diff --git a/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java b/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
index 1e1a9d2b8..d78b64c61 100644
--- a/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
+++ b/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
@@ -11,6 +11,8 @@ import android.widget.Button;
 import android.widget.FrameLayout;
 import android.widget.ImageView;
 
+import androidx.annotation.NonNull;
+
 import com.bumptech.glide.Glide;
 import com.coocoo.android.support.v4.app.Fragment;
 import com.coocoo.android.support.v4.app.FragmentManager;
@@ -32,6 +34,7 @@ import com.google.android.gms.ads.AdListener;
 import com.google.android.gms.ads.AdRequest;
 import com.google.android.gms.ads.AdSize;
 import com.google.android.gms.ads.AdView;
+import com.google.android.gms.ads.LoadAdError;
 import com.status.traffic.ThemeFabAdManager;
 
 import java.io.File;
@@ -124,9 +127,14 @@ public class ThemeStoreActivity extends CCBaseActivity implements View.OnClickLi
                 Report.reportThemeStoreBannerAdShow(BANNER_PLACEMENT, bannerId);
             }
 
+//            @Override
+//            public void onAdFailedToLoad(int code) {
+//                Report.reportThemeStoreBannerAdReqFail(BANNER_PLACEMENT, bannerId, code);
+//            }
+
             @Override
-            public void onAdFailedToLoad(int code) {
-                Report.reportThemeStoreBannerAdReqFail(BANNER_PLACEMENT, bannerId, code);
+            public void onAdFailedToLoad(@NonNull LoadAdError loadAdError) {
+                Report.reportThemeStoreBannerAdReqFail(BANNER_PLACEMENT, bannerId, loadAdError.getCode());
             }
         });
         adView.setAdUnitId(bannerId);
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
index 79a9459a3..95c398379 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
@@ -12,10 +12,10 @@ import android.graphics.Point;
 import android.net.Uri;
 import android.os.Bundle;
 import android.os.Handler;
-import android.os.Looper;
 import android.text.TextUtils;
 import android.view.LayoutInflater;
 import android.view.View;
+import android.view.ViewGroup;
 import android.widget.BaseAdapter;
 import android.widget.FrameLayout;
 import android.widget.ListView;
@@ -79,9 +79,11 @@ import com.coocoo.widget.indicator.ViewPagerIndicator;
 import com.coocoowhatsapp.conversationslist.ConversationsFragment;
 import com.coocoowhatsapp.conversationslist.ViewHolder;
 import com.google.android.gms.ads.AdListener;
+import com.google.android.gms.ads.AdLoader;
 import com.google.android.gms.ads.AdRequest;
-import com.google.android.gms.ads.AdSize;
-import com.google.android.gms.ads.AdView;
+import com.google.android.gms.ads.LoadAdError;
+import com.google.android.gms.ads.nativead.NativeAd;
+import com.google.android.gms.ads.nativead.NativeAdView;
 import com.status.traffic.StatusTrafficSdk;
 import com.status.traffic.data.enums.DownloadApkDialogType;
 import com.status.traffic.data.enums.InstallTag;
@@ -89,7 +91,6 @@ import com.status.traffic.data.vo.ApkData;
 import com.status.traffic.presenter.ILaunchInstallActivityPresenter;
 import com.status.traffic.presenter.LaunchInstallActivityPresenter;
 import com.status.traffic.report.AdReporter;
-import com.status.traffic.report.AdsFillReporter;
 import com.sublive.mod.LarkSdk;
 import com.sublive.mod.applets.Applets;
 import com.sublive.mod.applets.bean.AppletsPushMessage;
@@ -97,7 +98,6 @@ import com.whatsapp.jid.UserJid;
 
 import org.jetbrains.annotations.NotNull;
 
-import java.io.File;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
@@ -246,7 +246,7 @@ public class ConversationsFragmentDelegate extends FragmentDelegate<Conversation
         LogUtil.debug("ConversationsFragmentDelegate.onCreateView");
         super.onCreateView(view, savedInstanceState);
         getHomeActivity();
-        initBannerAd();
+        initBannerAd(view);
     }
 
     @SmaliInject
@@ -303,49 +303,52 @@ public class ConversationsFragmentDelegate extends FragmentDelegate<Conversation
         listView.addHeaderView(newcomerPromoteFeatureContainer);
     }
 
-    private void initBannerAd() {
+    private void initBannerAd(View view) {
 
         final String bannerId = RemoteConfig.getConversationBannerAd();
+        LogUtil.d("bannerId" + bannerId);
         if (TextUtils.isEmpty(bannerId)) {
             return;
         }
-
-        AdView adView = new AdView(Coocoo.getAppContext());
-        adView.setAdListener(new AdListener() {
-            @Override
-            public void onAdLoaded() {
-                Report.reportConversationBannerAdReqOk(bannerId);
-                new Handler(Looper.getMainLooper()).post(new Runnable() {
-                    @Override
-                    public void run() {
-                        if (conversationBannerItem != null) {
-                            conversationBannerItem.setAdLoadSuccess(true);
-                        }
-                        BaseAdapter adapter = ((ConversationsFragment) mHostFragment).A0r;
-                        if (adapter != null) {
-                            adapter.notifyDataSetChanged();
-                        }
-                    }
-                });
-            }
-
+        AdLoader.Builder builder = new AdLoader.Builder(Coocoo.getAppContext(), bannerId);
+        NativeAdView nativeAdView = (NativeAdView) LayoutInflater.from(view.getContext()).inflate(ResMgr.getLayoutId("cc_ad_banner"), (ViewGroup) view, false);
+        builder.forNativeAd((nativeAd) -> {
+            LogUtil.d("onAdLoaded" + nativeAd.getHeadline());
+            populateNativeAdView(nativeAd, nativeAdView);
+            new Handler().post(() -> {
+                if (conversationBannerItem != null) {
+                    conversationBannerItem.setAdLoadSuccess(true);
+                }
+                BaseAdapter adapter = ((ConversationsFragment) mHostFragment).A0r;
+                if (adapter != null) {
+                    adapter.notifyDataSetChanged();
+                }
+            });
+        });
+        AdLoader adLoader = builder.withAdListener(new AdListener() {
             @Override
             public void onAdImpression() {
                 Report.reportConversationBannerAdShow(bannerId);
             }
 
             @Override
-            public void onAdFailedToLoad(int code) {
-                Report.reportConversationBannerAdReqFail(bannerId, code);
+            public void onAdFailedToLoad(@NonNull LoadAdError loadAdError) {
+                Report.reportConversationBannerAdReqFail(bannerId, loadAdError.getCode());
             }
-        });
-        adView.setAdUnitId(bannerId);
-        adView.setAdSize(AdSize.BANNER);
-        adView.loadAd(new AdRequest.Builder().build());
-        conversationBannerItem = new ConversationBannerItem(adView);
+        }).build();
+        LogUtil.d("loadAd");
+        adLoader.loadAd(new AdRequest.Builder().build());
+        conversationBannerItem = new ConversationBannerItem(nativeAdView);
         Report.reportConversationBannerAdReqStart(bannerId);
     }
 
+    private void populateNativeAdView(NativeAd nativeAd, NativeAdView nativeAdView) {
+        nativeAdView.setHeadlineView(nativeAdView.findViewById(ResMgr.getId("cc_tv_title")));
+        nativeAdView.setBodyView(nativeAdView.findViewById(ResMgr.getId("cc_tv_subtitle")));
+        nativeAdView.setIconView(nativeAdView.findViewById(ResMgr.getId("cc_iv_ad")));
+        nativeAdView.setNativeAd(nativeAd);
+    }
+
     public void create(Bundle savedInstanceState) {
 
     }
diff --git a/app/src/main/res/layout/cc_ad_banner.xml b/app/src/main/res/layout/cc_ad_banner.xml
new file mode 100644
index 000000000..e6f7faa6b
--- /dev/null
+++ b/app/src/main/res/layout/cc_ad_banner.xml
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.google.android.gms.ads.nativead.NativeAdView xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="76dp"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <com.coocoowhatsapp.WaImageView
+            android:id="@+id/cc_iv_ad"
+            android:layout_width="48dp"
+            android:layout_height="48dp"
+            android:layout_marginStart="12dp"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            tools:src="@android:drawable/stat_sys_warning"/>
+
+        <com.coocoowhatsapp.WaTextView
+            android:id="@+id/cc_tv_title"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="12dp"
+            style="@style/ListItemTitle"
+            app:layout_constraintTop_toTopOf="@+id/cc_iv_ad"
+            app:layout_constraintStart_toEndOf="@+id/cc_iv_ad"
+            tools:text="Adventure Duck"/>
+
+        <com.coocoowhatsapp.WaImageView
+            android:id="@+id/cc_tv_label"
+            android:layout_height="wrap_content"
+            android:layout_width="wrap_content"
+            android:src="@drawable/cc_ic_ad_tag_chatlist"
+            android:layout_marginStart="6dp"
+            app:layout_constraintTop_toTopOf="@+id/cc_tv_title"
+            app:layout_constraintBottom_toBottomOf="@+id/cc_tv_title"
+            app:layout_constraintStart_toEndOf="@+id/cc_tv_title"/>
+
+        <com.coocoowhatsapp.WaTextView
+            android:id="@+id/cc_tv_subtitle"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            style="?listItemSubtitle"
+            app:layout_constraintStart_toStartOf="@+id/cc_tv_title"
+            app:layout_constraintBottom_toBottomOf="@+id/cc_iv_ad"
+            tools:text="Slash &amp; swim your way down…"/>
+
+        <com.coocoowhatsapp.WaTextView
+            android:id="@+id/btn_download"
+            android:layout_width="85dp"
+            android:layout_height="36dp"
+            android:textSize="13sp"
+            android:textColor="#ffffff"
+            android:layout_marginEnd="4dp"
+            android:background="@drawable/cc_btn_green_gradient"
+            android:gravity="center"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            android:text="@string/cc_carousel_conversation_download"/>
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+</com.google.android.gms.ads.nativead.NativeAdView>
\ No newline at end of file
