diff --git a/app/build.gradle b/app/build.gradle
index 50e40530d..cbf792786 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -42,8 +42,8 @@ android {
 //                || cur_flavor == "YoUnclone"
 //                || cur_flavor == "GbUnclone"
 //                || cur_flavor == "WhatsappPlus") {
-//            rootProject.ext.version_name = '"11.60.0"' // HeyMods
-//            rootProject.ext.version_code = "1160000" // HeyMods
+            rootProject.ext.version_name = '"9.0.0"' // HeyMods
+            rootProject.ext.version_code = "9000000" // HeyMods
 //        } else if (cur_flavor == "FM") {
 //            rootProject.ext.version_name = '"10.1.0"'
 //            rootProject.ext.version_code = "1001000"
@@ -84,8 +84,8 @@ android {
 //            rootProject.ext.version_name = '"0.20.197"'
 //            rootProject.ext.version_code = "20197"
 //        }
-//            rootProject.ext.version_name = '"11.60.0"' // HeyMods
-//            rootProject.ext.version_code = "1160000" // HeyMods
+            rootProject.ext.version_name = '"9.0.0"' // HeyMods
+            rootProject.ext.version_code = "9000000" // HeyMods
     }
 
     flavorDimensions "version"
diff --git a/app/src/main/java/com/coocoo/statuses/HistoryStatusActivity.kt b/app/src/main/java/com/coocoo/statuses/HistoryStatusActivity.kt
index 44bd507b1..e179ec0cd 100644
--- a/app/src/main/java/com/coocoo/statuses/HistoryStatusActivity.kt
+++ b/app/src/main/java/com/coocoo/statuses/HistoryStatusActivity.kt
@@ -15,6 +15,7 @@ import android.widget.TextView
 import androidx.lifecycle.CCViewModelProvider
 import com.bumptech.glide.Glide
 import com.coocoo.base.CCBaseActivity
+import com.coocoo.utils.LogUtil
 import com.coocoo.utils.ResMgr
 import java.text.DateFormat
 import java.text.SimpleDateFormat
@@ -33,6 +34,7 @@ class HistoryStatusActivity : CCBaseActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
         super.onCreate(savedInstanceState, persistentState)
+        LogUtil.d("onCreate")
         setContentView(ResMgr.getLayoutId("cc_activity_history_status"))
 
         findViewById<GridView>(ResMgr.getId("cc_grid_view")).adapter = statusGridAdapter
@@ -42,6 +44,7 @@ class HistoryStatusActivity : CCBaseActivity() {
 
     private fun bindView() {
         viewModel.statusList.observe(lifecycleOwner) {
+            LogUtil.d("bindView")
             statusGridAdapter.addAll(it)
         }
     }
@@ -62,7 +65,7 @@ class HistoryStatusActivity : CCBaseActivity() {
             }
 
             onBindViewHolder(position, vh)
-
+            LogUtil.d("getView")
             return vh.itemView
         }
 
diff --git a/app/src/main/java/com/coocoo/statuses/StatusListRepository.kt b/app/src/main/java/com/coocoo/statuses/StatusListRepository.kt
index 6b5dad405..a9e55c323 100644
--- a/app/src/main/java/com/coocoo/statuses/StatusListRepository.kt
+++ b/app/src/main/java/com/coocoo/statuses/StatusListRepository.kt
@@ -1,9 +1,11 @@
 package com.coocoo.statuses
 
+import com.coocoo.utils.LogUtil
 import java.util.*
 
 class StatusListRepository {
     fun getStatusList(): List<HistoryStatusActivity.StatusInfo> {
+        LogUtil.d("getStatusList")
         return List(20) { position ->
             HistoryStatusActivity.StatusInfo(
                 url = "https://picsum.photos/300/400",
diff --git a/app/src/main/res/layout/cc_activity_history_status.xml b/app/src/main/res/layout/cc_activity_history_status.xml
index c93016a88..83b15c8ed 100644
--- a/app/src/main/res/layout/cc_activity_history_status.xml
+++ b/app/src/main/res/layout/cc_activity_history_status.xml
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <LinearLayout
     xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical"
     android:layout_width="match_parent"
     android:layout_height="match_parent">
 
