diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
index dac76e7..f26590d 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
@@ -20,6 +20,8 @@ public class HomeActivityDelegate extends ActivityDelegate {
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        String str = null;
+        str.length();
         Log.d(TAG, "onCreate");
     }
 
