diff --git a/app/build.gradle b/app/build.gradle
index ac7c94191..74ca898b6 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -45,7 +45,7 @@ android {
 //                || cur_flavor == "GbUnclone"
 //                || cur_flavor == "WhatsappPlus") {
             rootProject.ext.version_name = '"9.0.0"' // HeyMods
-            rootProject.ext.version_code = "900000" // HeyMods
+            rootProject.ext.version_code = "9000000" // HeyMods
 //        } else if (cur_flavor == "FM") {
 //            rootProject.ext.version_name = '"10.1.0"'
 //            rootProject.ext.version_code = "1001000"
@@ -87,7 +87,7 @@ android {
 //            rootProject.ext.version_code = "20197"
 //        }
             rootProject.ext.version_name = '"9.0.0"' // HeyMods
-            rootProject.ext.version_code = "900000" // HeyMods
+            rootProject.ext.version_code = "9000000" // HeyMods
     }
 
     flavorDimensions "version"
diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 3b51bd9c5..6142e8d80 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -116,6 +116,7 @@ import javax.crypto.SecretKey;
 
 import X.c1_01f;
 import X.c1_2Oc;
+import X.c1_2Ot;
 import X.c1_3Ul;
 import X.c1_2P6;
 import X.c1_2ua;
@@ -640,7 +641,11 @@ public class Coocoo {
         Log.e(TAG, "-----------------3---------------- // value : " + value);
     }
 
+    // invoke-static {vx}, Lcom/coocoo/coocoo/Coocoo;->traceLog4(Ljava/lang/Object;)V
     public static void traceLog4(Object object) {
+        if (object instanceof c1_2Ot) {
+            Log.e(TAG, "-----------------4---------------- // object : " + object);
+        }
         Log.e(TAG, "-----------------4---------------- // object : " + object);
     }
 
@@ -648,6 +653,31 @@ public class Coocoo {
         Log.e(TAG, "-----------------5---------------- // value : " + value);
     }
 
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog6(Z)V
+    public static void traceLog6(boolean value) {
+        Log.e(TAG, "-----------------6---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog7(Ljava/lang/CharSequence;)V
+    public static void traceLog7(CharSequence value) {
+        Log.e(TAG, "-----------------7---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog8(Ljava/util/ArrayList;)V
+    public static void traceLog8(ArrayList value) {
+        Log.e(TAG, "-----------------8---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog9(Landroid/view/View;)V
+    public static void traceLog9(View value) {
+        Log.e(TAG, "-----------------9---------------- // value : " + value.getId());
+    }
+
+    // invoke-static {}, Lcom/coocoo/coocoo/Coocoo;->stackTrace()V
+    public static void stackTrace() throws Exception {
+        throw new Exception("Debug");
+    }
+
     public static void traceLogCodeStack() {
         try {
             throw new IllegalArgumentException("traceLogCodeStack");
diff --git a/app/src/main/java/com/coocoo/openchat/activity/viewmodel/WhisperOpenChatViewModel.kt b/app/src/main/java/com/coocoo/openchat/activity/viewmodel/WhisperOpenChatViewModel.kt
index 663d6a067..d53ec983f 100644
--- a/app/src/main/java/com/coocoo/openchat/activity/viewmodel/WhisperOpenChatViewModel.kt
+++ b/app/src/main/java/com/coocoo/openchat/activity/viewmodel/WhisperOpenChatViewModel.kt
@@ -14,6 +14,7 @@ import com.coocoo.openchat.activity.OpenChatWebActivity.Companion.KEY_TYPE_LIVEH
 import com.coocoo.openchat.activity.OpenChatWebActivity.Companion.KEY_TYPE_OPENCHAT
 import com.coocoo.profile.ProfileManager
 import com.coocoo.report.ReportOpenChat
+import com.coocoo.utils.LogUtil
 import com.coocoo.utils.PrivacyUtils
 import com.coocoo.utils.ResMgr
 import com.coocoo.utils.SystemUtil
@@ -93,6 +94,9 @@ class WhisperOpenChatViewModel(private val whisper: Whisper) {
                 "${uri.toString()}&telephone=$phoneNumber"
             }
             else -> OPENCHAT_HOME_URL
+        }.also {
+            LogUtil.d("type: $type")
+            LogUtil.d(it)
         }
     }
 
