diff --git a/app/src/main/java/com/coocoo/CooCoo.kt b/app/src/main/java/com/coocoo/CooCoo.kt
index 6859204e..bf5ca0a6 100644
--- a/app/src/main/java/com/coocoo/CooCoo.kt
+++ b/app/src/main/java/com/coocoo/CooCoo.kt
@@ -137,7 +137,7 @@ object CooCoo {
             appFlavor,
             com.coocoo.config.BuildConfig.COOCOO_VERSION_CODE,
             appChannel,
-            com.coocoo.config.BuildConfig.IS_DEBUG,
+            false,
         )
         AdCommonManager.injectReporter(ReportStatusVideoTraffic)
     }
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
index befc7a47..7c353e6c 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
@@ -144,9 +144,9 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
             try {
                 StatusTrafficReporter.report(Constant.Report.ST_FETCH_CONF_START)
                 init(context)
-                if (false == apiPref?.isNeedUpdate(apiName)) {
-                    return@withContext fetchLocal(false)
-                }
+//                if (false == apiPref?.isNeedUpdate(apiName)) {
+//                    return@withContext fetchLocal(false)
+//                }
 
                 apiQualifiedConfig = null
                 apiRemoteConfig = null
@@ -169,7 +169,7 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
 
                 val headers = OkHttpHelper.getRemoteConfigHeaders(
                     isDebug,
-                    country,
+                    "cn",
                     appVersion,
                     appChannel,
                     appFlavor,
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
index 1d430ca3..5361c3ec 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
@@ -3,9 +3,11 @@ package com.status.traffic.data.config
 import com.google.gson.JsonArray
 import com.google.gson.JsonObject
 import com.status.traffic.Constant.Api.Config.CALL_LIST_BANNER_AD
+import com.status.traffic.data.vo.AdsSDK
 import com.status.traffic.data.vo.CallListBannerAdConfig
 
-object CallListBannerAdRemoteConfig : BaseAdRemoteConfig<List<CallListBannerAdConfig>, Void>(CALL_LIST_BANNER_AD) {
+object CallListBannerAdRemoteConfig :
+    BaseAdRemoteConfig<List<CallListBannerAdConfig>, Void>(CALL_LIST_BANNER_AD) {
     override fun parserAdsInfo(json: JsonArray?): List<CallListBannerAdConfig>? {
         try {
             return getGson().fromJson(json, Array<CallListBannerAdConfig>::class.java).toList()
@@ -22,6 +24,15 @@ object CallListBannerAdRemoteConfig : BaseAdRemoteConfig<List<CallListBannerAdCo
     }
 
     fun getCallListBannerAd(): CallListBannerAdConfig? {
-        return apiRemoteConfig?.get(0)
+//        return apiRemoteConfig?.get(0)
+        return CallListBannerAdConfig(
+            waterFall = arrayListOf(
+                AdsSDK(
+                    sdk = "mv",
+                    adId = "dj9WUBLYwepIKQMlEa7LzaWR4EozNehoLzdIrVkA7Lo=",
+                    format = "native"
+                )
+            )
+        )
     }
 }
\ No newline at end of file
diff --git a/whatsapp_modify_module/smali_classes2/X/1kx.2.smali b/whatsapp_modify_module/smali_classes2/X/1kx.2.smali
index e0b3708a..7fdbff5a 100644
--- a/whatsapp_modify_module/smali_classes2/X/1kx.2.smali
+++ b/whatsapp_modify_module/smali_classes2/X/1kx.2.smali
@@ -611,7 +611,7 @@
 
     :goto_0
 #coocoo
-    invoke-static {v0}, Lcom/coocoo/callhistory/CallHistoryDelegate;->addCallListBannerItem(Ljava/util/ArrayList;)V
+    #invoke-static {v0}, Lcom/coocoo/callhistory/CallHistoryDelegate;->addCallListBannerItem(Ljava/util/ArrayList;)V
 #coocoo
     iput-object v0, v2, Lcom/gbwhatsapp/calling/callhistory/CallsHistoryFragment;->A0f:Ljava/util/ArrayList;
 
