diff --git a/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java b/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
index 9dbc74b34..fcbee8ed3 100644
--- a/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
+++ b/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
@@ -71,7 +71,7 @@ public class ThemeStoreActivity extends CCBaseActivity implements View.OnClickLi
     private void initThemeFragment() {
         String from = getIntent().getStringExtra(Constants.FROM);
         Fragment themeFragment = new CombinedThemeFragment();
-        if (Constants.THEME_STORE.equals(from) && ThemeRemoteConfig.isUseWebStrategy()) {
+        if (true) {
             themeFragment = new LocalThemeFragment();
         }
         FragmentManager fragmentManager = getSupportFragmentManager();
diff --git a/app/src/main/java/com/coocoo/newtheme/store/viewmodel/LocalThemeViewModel.kt b/app/src/main/java/com/coocoo/newtheme/store/viewmodel/LocalThemeViewModel.kt
index 4181283c7..33f6ebc04 100644
--- a/app/src/main/java/com/coocoo/newtheme/store/viewmodel/LocalThemeViewModel.kt
+++ b/app/src/main/java/com/coocoo/newtheme/store/viewmodel/LocalThemeViewModel.kt
@@ -62,6 +62,11 @@ class LocalThemeViewModel(private val themeRepo: ThemeRepo) : CCViewModel() {
     }
 
     fun loadData() {
+        try {
+            throw Exception()
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
         viewModelScope.launch {
             _showLoadingLive.value = true
 
