diff --git a/app/src/main/java/com/coocoo/CooCoo.kt b/app/src/main/java/com/coocoo/CooCoo.kt
index 6859204e..bf5ca0a6 100644
--- a/app/src/main/java/com/coocoo/CooCoo.kt
+++ b/app/src/main/java/com/coocoo/CooCoo.kt
@@ -137,7 +137,7 @@ object CooCoo {
             appFlavor,
             com.coocoo.config.BuildConfig.COOCOO_VERSION_CODE,
             appChannel,
-            com.coocoo.config.BuildConfig.IS_DEBUG,
+            false,
         )
         AdCommonManager.injectReporter(ReportStatusVideoTraffic)
     }
diff --git a/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt b/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
index 6eecf980..f8564ba4 100644
--- a/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
+++ b/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
@@ -1,11 +1,13 @@
 package com.coocoo.callhistory
 
 import X.cc_5WG
+import android.util.Log
 import android.view.Gravity
 import android.view.View
 import android.view.ViewGroup
 import android.widget.RelativeLayout
 import com.coocoo.annotation.SmaliInject
+import com.coocoo.callhistory.CallHistoryDelegate.addCallListBannerItem
 import com.coocoo.widget.CommonDialog
 import com.gbwhatsapp.calling.callhistory.CallsHistoryFragment
 import com.status.traffic.ads.AdSdkManager
@@ -35,6 +37,7 @@ object CallHistoryDelegate {
     @JvmStatic
     @SmaliInject
     fun addCallListBannerItem(arr: ArrayList<cc_5WG>) {
+        Log.e("Gene", "addCallListBannerItem: ${AdSdkManager.isCallListBannerAdReady()}")
         if (AdSdkManager.isCallListBannerAdReady()) {
             if (arr.size > 4) {
                 arr.add(4, CallHistoryItem())
@@ -44,6 +47,10 @@ object CallHistoryDelegate {
         }
     }
 
+    private fun ArrayList<cc_5WG>.containsCallListBanner(): Boolean {
+        return find { it.ADd() == CallHistoryItem.VIEW_TYPE } != null
+    }
+
     @JvmStatic
     @SmaliInject
     fun onResume(fragment: CallsHistoryFragment) {
@@ -51,10 +58,17 @@ object CallHistoryDelegate {
             val adsSdk = config.waterFall
             if (adsSdk != null) {
                 AdSdkManager.loadCallListBannerAd(fragment.A0C(), adsSdk) {
+                    Log.e("Gene", "onLoadAdListener: ${fragment.A0f}")
                     addCallListBannerItem(fragment.A0f)
                     fragment.A07.notifyDataSetChanged()
                 }
             }
         }
     }
+
+    @JvmStatic
+    @SmaliInject
+    fun onPause(fragment: CallsHistoryFragment) {
+
+    }
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt b/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt
index 1b85ca04..16699e4f 100644
--- a/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt
+++ b/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt
@@ -2,9 +2,11 @@ package com.coocoo.callhistory
 
 import X.cc_4VA
 import X.cc_5WG
+import android.util.Log
 import android.widget.RelativeLayout
 import com.gbwhatsapp.calling.callhistory.CallsHistoryFragment
 import com.status.traffic.ads.AdSdkManager
+import com.status.traffic.data.config.CallListBannerAdRemoteConfig
 
 class CallHistoryViewHolder(
     private val targetView: RelativeLayout,
@@ -16,6 +18,7 @@ class CallHistoryViewHolder(
     }
 
     fun bindAd() {
+        Log.e("Gene", "targetView.childCount: ${targetView.childCount}")
         AdSdkManager.showCallListBannerAd(targetView) {
             val iterable = fragment.A0f.iterator()
             while (iterable.hasNext()) {
diff --git a/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseAdController.kt b/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseAdController.kt
index e067332e..759f03bd 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseAdController.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseAdController.kt
@@ -31,7 +31,7 @@ abstract class BaseAdController <T: BaseAdController.BaseAd?> {
             return
         }
 
-        ad = null
+//        ad = null
         adConfig = null
         loadAdJob =  scope.launch(Dispatchers.IO) {
             if (timeout == null) {
diff --git a/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseBannerAd.kt b/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseBannerAd.kt
index 053c2982..aa31eeb1 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseBannerAd.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseBannerAd.kt
@@ -1,5 +1,6 @@
 package com.status.traffic.ads.ad
 
+import android.util.Log
 import android.view.View
 import android.view.ViewGroup
 import android.widget.RelativeLayout
@@ -53,6 +54,7 @@ open class BaseBannerAd (
                         setUpView(bannerContainer, nativeHandler.adViewGroup)?.run {
                             bannerContainer.addView(this)
                         }
+                        Log.e("Report", "nativeHandler.onResume() parent:${bannerContainer} ad:${nativeHandler.adViewGroup}")
                         nativeHandler.onResume()
                     }
                 }
diff --git a/status-traffic-api/src/main/java/com/status/traffic/ads/ad/CallListBannerAdController.kt b/status-traffic-api/src/main/java/com/status/traffic/ads/ad/CallListBannerAdController.kt
index 75289422..1054608e 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/ads/ad/CallListBannerAdController.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/ads/ad/CallListBannerAdController.kt
@@ -1,3 +1,11 @@
 package com.status.traffic.ads.ad
 
-class CallListBannerAdController: BaseBannerAdController()
\ No newline at end of file
+import android.util.Log
+
+class CallListBannerAdController: BaseBannerAdController() {
+    override fun release() {
+        super.release()
+        Log.e("Gene", "release")
+        java.lang.Exception().printStackTrace()
+    }
+}
\ No newline at end of file
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
index befc7a47..7c353e6c 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
@@ -144,9 +144,9 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
             try {
                 StatusTrafficReporter.report(Constant.Report.ST_FETCH_CONF_START)
                 init(context)
-                if (false == apiPref?.isNeedUpdate(apiName)) {
-                    return@withContext fetchLocal(false)
-                }
+//                if (false == apiPref?.isNeedUpdate(apiName)) {
+//                    return@withContext fetchLocal(false)
+//                }
 
                 apiQualifiedConfig = null
                 apiRemoteConfig = null
@@ -169,7 +169,7 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
 
                 val headers = OkHttpHelper.getRemoteConfigHeaders(
                     isDebug,
-                    country,
+                    "cn",
                     appVersion,
                     appChannel,
                     appFlavor,
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
index 1d430ca3..5361c3ec 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
@@ -3,9 +3,11 @@ package com.status.traffic.data.config
 import com.google.gson.JsonArray
 import com.google.gson.JsonObject
 import com.status.traffic.Constant.Api.Config.CALL_LIST_BANNER_AD
+import com.status.traffic.data.vo.AdsSDK
 import com.status.traffic.data.vo.CallListBannerAdConfig
 
-object CallListBannerAdRemoteConfig : BaseAdRemoteConfig<List<CallListBannerAdConfig>, Void>(CALL_LIST_BANNER_AD) {
+object CallListBannerAdRemoteConfig :
+    BaseAdRemoteConfig<List<CallListBannerAdConfig>, Void>(CALL_LIST_BANNER_AD) {
     override fun parserAdsInfo(json: JsonArray?): List<CallListBannerAdConfig>? {
         try {
             return getGson().fromJson(json, Array<CallListBannerAdConfig>::class.java).toList()
@@ -22,6 +24,15 @@ object CallListBannerAdRemoteConfig : BaseAdRemoteConfig<List<CallListBannerAdCo
     }
 
     fun getCallListBannerAd(): CallListBannerAdConfig? {
-        return apiRemoteConfig?.get(0)
+//        return apiRemoteConfig?.get(0)
+        return CallListBannerAdConfig(
+            waterFall = arrayListOf(
+                AdsSDK(
+                    sdk = "mv",
+                    adId = "dj9WUBLYwepIKQMlEa7LzaWR4EozNehoLzdIrVkA7Lo=",
+                    format = "native"
+                )
+            )
+        )
     }
 }
\ No newline at end of file
diff --git a/whatsapp_modify_module/smali_classes2/com/gbwhatsapp/calling/callhistory/CallsHistoryFragment.smali b/whatsapp_modify_module/smali_classes2/com/gbwhatsapp/calling/callhistory/CallsHistoryFragment.smali
index b2f2de20..833dcf2f 100644
--- a/whatsapp_modify_module/smali_classes2/com/gbwhatsapp/calling/callhistory/CallsHistoryFragment.smali
+++ b/whatsapp_modify_module/smali_classes2/com/gbwhatsapp/calling/callhistory/CallsHistoryFragment.smali
@@ -487,7 +487,9 @@
 
 .method public A0r()V
     .locals 1
-
+#coocoo
+    invoke-static {p0}, Lcom/coocoo/callhistory/CallHistoryDelegate;->onPause(Lcom/gbwhatsapp/calling/callhistory/CallsHistoryFragment;)V
+#coocoo
     const-string/jumbo v0, "voip/CallsFragment/onPause"
 
     invoke-static {v0}, Lcom/whatsapp/util/Log;->i(Ljava/lang/String;)V
