diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 77bd78c7b..bf93eb98e 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -589,7 +589,9 @@ public class Coocoo {
 
     @SmaliInject
     public static boolean isHideReceipt(c1_2P6 param) {
-        return PrivacyPreferenceManager.Companion.getInstance().isHideReceipt(param);
+        boolean rtn = PrivacyPreferenceManager.Companion.getInstance().isHideReceipt(param);
+        LogUtil.d("Coocoo", rtn);
+        return rtn;
     }
 
     @SmaliInject
