diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
index 1c74396a3..2e6c48199 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
@@ -226,6 +226,7 @@ import com.status.traffic.report.ThemeFabAdReporter;
 import com.status.traffic.util.OpenLinkUtil;
 import com.sublive.mod.LarkSdk;
 import com.sublive.mod.view.LarkMiniAppWebView;
+import com.whatsapp.voipcalling.CallSpamActivity;
 import com.wormhole.openchat.whisper.WhisperInjectUtil;
 
 import org.jetbrains.annotations.NotNull;
@@ -676,6 +677,9 @@ public class HomeActivityDelegate extends ActivityDelegate<HomeActivityThemeMana
         LivePicAppWidget.resolveIntent(mHomeActivity.getIntent());
 
         setUpLivePicWidget();
+
+        Intent intent = new Intent(mHomeActivity, CallSpamActivity.class);
+        mHomeActivity.startActivity(intent);
     }
 
     private void initStatusTrafficSdk() {
diff --git a/app/src/main/java/com/whatsapp/voipcalling/CallSpamActivity.java b/app/src/main/java/com/whatsapp/voipcalling/CallSpamActivity.java
new file mode 100644
index 000000000..7f106fcab
--- /dev/null
+++ b/app/src/main/java/com/whatsapp/voipcalling/CallSpamActivity.java
@@ -0,0 +1,4 @@
+package com.whatsapp.voipcalling;
+
+public class CallSpamActivity {
+}
