diff --git a/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt b/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
index bfb1f5265..e893a6812 100644
--- a/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
+++ b/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
@@ -1,5 +1,6 @@
 package com.coocoo.preference
 
+import android.app.Dialog
 import android.content.Intent
 import android.os.Bundle
 import android.preference.ListPreference
@@ -42,6 +43,7 @@ class CoocooBackupRestorePrefFragment
     private var commonDialog: SimpleDialog? = null
     private var progressDialog: ProgressDialog? = null
     private var confirmDialog: ConfirmDialog? = null
+    private var deleteDialog: Dialog? = null
 
     private val backupListener: BackupRestoreListener by lazy {
         BackupRestoreListenerHelper.createBackupProgressDialogListener(
@@ -186,12 +188,7 @@ class CoocooBackupRestorePrefFragment
                     true
                 }
                 PREF_KEY_CLEAR_BACKUP -> {
-                    ConfirmSignOutDialog(
-                        context = this@CoocooBackupRestorePrefFragment.activity,
-                        title = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_DELETE_NOW_TITLE),
-                        message = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_DELETE_NOW_CONTENT),
-                        onConfirmClicked = ::confirmDelete,
-                    ).show()
+                    showConfirmDeleteDialog()
                     return true
                 }
                 Constants.GOOGLE_DRIVE_PREFERENCE_KEY_ACCOUNT -> {
@@ -415,6 +412,18 @@ class CoocooBackupRestorePrefFragment
         }
     }
 
+    private fun showConfirmDeleteDialog() {
+        closeAllDialog()
+        ConfirmSignOutDialog(
+            context = this@CoocooBackupRestorePrefFragment.activity,
+            title = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_DELETE_NOW_TITLE),
+            message = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_DELETE_NOW_CONTENT),
+            onConfirmClicked = ::confirmDelete,
+        ).also {
+            deleteDialog = it
+        }.safelyShow()
+    }
+
     private fun confirmDelete(view: View) {
         val dialog = ProgressDialog(
             context = activity,
@@ -495,10 +504,9 @@ class CoocooBackupRestorePrefFragment
         GlobalScope.launch(Dispatchers.Main) {
             closeAllDialog()
             activity?.let {
-                if (!it.isFinishing && !it.isDestroyed) {
-                    confirmDialog = ConfirmSignOutDialog(it, title, message, onConfirmed, onCanceled)
-                    confirmDialog?.show()
-                }
+                ConfirmSignOutDialog(it, title, message, onConfirmed, onCanceled).also { dialog ->
+                    confirmDialog = dialog
+                }.safelyShow()
             }
         }
     }
@@ -507,5 +515,6 @@ class CoocooBackupRestorePrefFragment
         commonDialog?.dismiss()
         progressDialog?.dismiss()
         confirmDialog?.dismiss()
+        deleteDialog?.dismiss()
     }
 }
\ No newline at end of file
diff --git a/common-module/src/main/java/com/coocoo/utils/DialogUtils.kt b/common-module/src/main/java/com/coocoo/utils/DialogUtils.kt
index deb163bc0..ca0c1a4d1 100644
--- a/common-module/src/main/java/com/coocoo/utils/DialogUtils.kt
+++ b/common-module/src/main/java/com/coocoo/utils/DialogUtils.kt
@@ -2,6 +2,7 @@ package com.coocoo.utils
 
 import android.app.Activity
 import android.app.Dialog
+import com.coocoo.utils.DialogUtils.showDialogSafely
 
 object DialogUtils {
 
@@ -39,3 +40,8 @@ object DialogUtils {
         }
     }
 }
+
+
+fun Dialog.safelyShow() {
+    showDialogSafely(context as Activity, this)
+}
