diff --git a/app/src/main/java/com/coocoo/newtheme/store/LocalThemeFragment.kt b/app/src/main/java/com/coocoo/newtheme/store/LocalThemeFragment.kt
index 740e1847b..f50148cb6 100644
--- a/app/src/main/java/com/coocoo/newtheme/store/LocalThemeFragment.kt
+++ b/app/src/main/java/com/coocoo/newtheme/store/LocalThemeFragment.kt
@@ -39,8 +39,7 @@ class LocalThemeFragment : CCBaseFragment() {
 
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
-        setUpViewModel()
-        viewModel.loadData()
+
     }
 
     override fun createView(
@@ -57,7 +56,7 @@ class LocalThemeFragment : CCBaseFragment() {
 
     override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
+        setUpViewModel()
         view ?: return
 
         gridView = view.findViewById<GridView>(ResMgr.getId(Constants.Res.Id.THEME_STORE_GRID_VIEW))
@@ -112,6 +111,7 @@ class LocalThemeFragment : CCBaseFragment() {
         }
         gridView?.adapter = themeListAdapter
         gridView?.setRecyclerListener { v: View? -> themeListAdapter.recycleView(v) }
+        viewModel.loadData()
     }
 
     private fun setUpViewModel() {
diff --git a/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java b/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
index 9dbc74b34..8de8c711a 100644
--- a/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
+++ b/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
@@ -23,6 +23,7 @@ import com.coocoo.newtheme.notification.ThemeNotificationManager;
 import com.coocoo.report.Report;
 import com.coocoo.utils.Constants;
 import com.coocoo.utils.FileUtil;
+import com.coocoo.utils.LogUtil;
 import com.coocoo.utils.ResMgr;
 import com.coocoo.widget.LoadingDialog;
 import com.status.traffic.ThemeFabAdManager;
@@ -34,6 +35,11 @@ import java.util.concurrent.atomic.AtomicReference;
 public class ThemeStoreActivity extends CCBaseActivity implements View.OnClickListener {
 
     public static Intent newIntent(Context context, String from) {
+        try {
+            throw new Exception();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
         Intent intent = new Intent(context, ThemeStoreActivity.class);
         intent.putExtra(Constants.FROM, from);
         return intent;
@@ -69,9 +75,10 @@ public class ThemeStoreActivity extends CCBaseActivity implements View.OnClickLi
     }
 
     private void initThemeFragment() {
+        LogUtil.d("initThemeFragment");
         String from = getIntent().getStringExtra(Constants.FROM);
         Fragment themeFragment = new CombinedThemeFragment();
-        if (Constants.THEME_STORE.equals(from) && ThemeRemoteConfig.isUseWebStrategy()) {
+        if (true) {
             themeFragment = new LocalThemeFragment();
         }
         FragmentManager fragmentManager = getSupportFragmentManager();
diff --git a/app/src/main/java/com/coocoo/newtheme/store/viewmodel/LocalThemeViewModel.kt b/app/src/main/java/com/coocoo/newtheme/store/viewmodel/LocalThemeViewModel.kt
index 4181283c7..33f6ebc04 100644
--- a/app/src/main/java/com/coocoo/newtheme/store/viewmodel/LocalThemeViewModel.kt
+++ b/app/src/main/java/com/coocoo/newtheme/store/viewmodel/LocalThemeViewModel.kt
@@ -62,6 +62,11 @@ class LocalThemeViewModel(private val themeRepo: ThemeRepo) : CCViewModel() {
     }
 
     fun loadData() {
+        try {
+            throw Exception()
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
         viewModelScope.launch {
             _showLoadingLive.value = true
 
