diff --git a/app/src/main/java/com/coocoo/downloader/Downloader.java b/app/src/main/java/com/coocoo/downloader/Downloader.java
index 1280545b0..d73f109d7 100644
--- a/app/src/main/java/com/coocoo/downloader/Downloader.java
+++ b/app/src/main/java/com/coocoo/downloader/Downloader.java
@@ -1,5 +1,6 @@
 package com.coocoo.downloader;
 
+import android.os.Environment;
 import android.os.Handler;
 import android.os.Looper;
 import android.os.Message;
@@ -8,6 +9,7 @@ import android.text.TextUtils;
 import com.coocoo.CooCooApp;
 import com.coocoo.downloader.model.DownloadData;
 import com.coocoo.downloader.net.NetworkManager;
+import com.coocoo.utils.AppUtil;
 import com.coocoo.utils.FileUtil;
 
 import java.io.File;
@@ -29,6 +31,11 @@ public class Downloader implements IDownloader {
         return InstanceHolder.INSTANCE;
     }
 
+    public static void deleteExternalDownloadDir() {
+        String mDownloadRootPath = new File(Environment.getExternalStorageDirectory().getAbsolutePath(), AppUtil.getAppName() + "/download/").getAbsolutePath();
+        FileUtil.deleteDir(mDownloadRootPath);
+    }
+
     public synchronized boolean init() {
         if (mInit) {
             return true;
diff --git a/app/src/main/java/com/coocoo/newtheme/ExternalThemeDeleter.kt b/app/src/main/java/com/coocoo/newtheme/ExternalThemeDeleter.kt
new file mode 100644
index 000000000..29329e41f
--- /dev/null
+++ b/app/src/main/java/com/coocoo/newtheme/ExternalThemeDeleter.kt
@@ -0,0 +1,22 @@
+package com.coocoo.newtheme
+
+import com.coocoo.downloader.Downloader
+import com.coocoo.newtheme.store.ThemeRepo
+import com.coocoo.permission.PermissionManager
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+object ExternalThemeDeleter {
+
+    @JvmStatic
+    fun delete() {
+        if (PermissionManager.getInstance().isStorageGranted) {
+            GlobalScope.launch(Dispatchers.IO) {
+                ThemeManager.deleteExternalThemeDir()
+                ThemeRepo.deleteExternalColorPhoneDir()
+                Downloader.deleteExternalDownloadDir()
+            }
+        }
+    }
+}
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/newtheme/ThemeManager.java b/app/src/main/java/com/coocoo/newtheme/ThemeManager.java
index 70044fd51..9b38a5d50 100644
--- a/app/src/main/java/com/coocoo/newtheme/ThemeManager.java
+++ b/app/src/main/java/com/coocoo/newtheme/ThemeManager.java
@@ -5,7 +5,6 @@ import android.text.TextUtils;
 import com.coocoo.CooCooApp;
 import com.coocoo.android.support.annotation.NonNull;
 import com.coocoo.android.support.annotation.Nullable;
-import com.coocoo.coocoo.Coocoo;
 import com.coocoo.coocoosp.SPDelegate;
 import com.coocoo.db.ThemeMessageManager;
 import com.coocoo.db.conversation.ThemeMessage;
@@ -26,7 +25,6 @@ import com.coocoo.utils.SystemUtil;
 import org.json.JSONArray;
 import org.json.JSONObject;
 
-import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicBoolean;
@@ -35,15 +33,15 @@ import static com.coocoo.newtheme.model.ThemeConstants.ID_DEFAULT;
 import static com.coocoo.newtheme.model.ThemeConstants.ID_MIN_BUILDIN;
 import static com.coocoo.newtheme.model.ThemeConstants.ID_MIN_CUSTOM;
 import static com.coocoo.newtheme.model.ThemeConstants.ID_MIN_IMPORT;
+import static com.coocoo.utils.Constant.PATH_SDCARD_THEME;
 import static com.coocoo.utils.Constants.FILE_EXTENSION_MP4;
 import static com.coocoo.utils.Constants.OFFICIAL_THEME_HOME_VIDEO_URL;
 
 public class ThemeManager {
     private static final String TAG = "ThemeManager";
     public static int VERSION = 2;
-    public static String THEMES_PATH = Coocoo.getAppContext().getFilesDir() + File.separator + "Themes";
-    public static String EXPORT_DIR_PATH = THEMES_PATH + File.separator +  "export";
-    public static String IMPORT_DIR_PATH = THEMES_PATH + File.separator + "import";
+    public static String EXPORT_DIR_PATH = FileUtil.concatPaths(PATH_SDCARD_THEME, "export");
+    public static String IMPORT_DIR_PATH = FileUtil.concatPaths(PATH_SDCARD_THEME, "import");
     public static int IMPORT_COUNT_LIMIT = 50;
     public static int IMPORT_COUNT_SHOW = 100;
     public static int CREATE_THEME_COUNT_LIMIT = 50;
@@ -92,6 +90,10 @@ public class ThemeManager {
         return themeId;
     }
 
+    public static void deleteExternalThemeDir() {
+        FileUtil.deleteDir(PATH_SDCARD_THEME);
+    }
+
     /**
      * 按id获取一个主题
      * @param id
@@ -506,7 +508,7 @@ public class ThemeManager {
     }
 
     private synchronized ThemeInfo migrateDiy() {
-        String dirPath = FileUtil.concatPaths(THEMES_PATH, "diy");
+        String dirPath = FileUtil.concatPaths(PATH_SDCARD_THEME, "diy");
         if (!FileUtil.isDirExists(dirPath, false)) {
             return null;
         }
@@ -521,7 +523,7 @@ public class ThemeManager {
     private synchronized ThemeInfo migrateOffical(String themeName) {
         String url = "https://static.cdn.wamodshost.com/coocoo/config/images/wallpaper/" + themeName + ".zip";
         String dirName = MD5Util.getMD5(url);
-        String dirPath = FileUtil.concatPaths(THEMES_PATH, dirName);
+        String dirPath = FileUtil.concatPaths(PATH_SDCARD_THEME, dirName);
         if (!FileUtil.isDirExists(dirPath, false)) {
             return null;
         }
diff --git a/app/src/main/java/com/coocoo/newtheme/store/ThemeRepo.java b/app/src/main/java/com/coocoo/newtheme/store/ThemeRepo.java
index 8b0ea1e37..a7b543e41 100644
--- a/app/src/main/java/com/coocoo/newtheme/store/ThemeRepo.java
+++ b/app/src/main/java/com/coocoo/newtheme/store/ThemeRepo.java
@@ -1,6 +1,7 @@
 package com.coocoo.newtheme.store;
 
 import android.content.Context;
+import android.os.Environment;
 import android.text.TextUtils;
 
 import androidx.lifecycle.CCLiveData;
@@ -8,6 +9,7 @@ import androidx.lifecycle.CCMutableLiveData;
 
 import com.coocoo.android.support.annotation.Nullable;
 import com.coocoo.colorphone.ColorPhoneThemeData;
+import com.coocoo.colorphone.Constant;
 import com.coocoo.config.BuildConfig;
 import com.coocoo.firebase.remoteConfig.RemoteConfig;
 import com.coocoo.manager.DownloadHelper;
@@ -44,6 +46,7 @@ import java.util.Map;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
+import static com.coocoo.colorphone.Constant.WORK_DIR;
 import static com.coocoo.utils.Constants.ERROR_HOT_THEME_UNSUPPORTED_MSG;
 import static com.coocoo.utils.Constants.FILE_EXTENSION_MP4;
 
@@ -73,6 +76,11 @@ public class ThemeRepo {
         void onFailure(Throwable throwable);
     }
 
+    public static void deleteExternalColorPhoneDir() {
+        String dirPath = Environment.getExternalStorageDirectory().getAbsolutePath() + "/" + WORK_DIR;
+        FileUtil.deleteDir(dirPath);
+    }
+
     public ThemeRepo(Context applicationContext) {
         this.appContext = applicationContext;
         executorService = Executors.newFixedThreadPool(5);
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
index bfcb9f0cb..ec3644976 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.java
@@ -128,6 +128,7 @@ import com.coocoo.manager.RoutingManager;
 import com.coocoo.manager.StatusesItemTypeManager;
 import com.coocoo.manager.WhatstallerPreDownloader;
 import com.coocoo.masssender.MassMessageSenderListActivity;
+import com.coocoo.newtheme.ExternalThemeDeleter;
 import com.coocoo.newtheme.TabConstant;
 import com.coocoo.newtheme.ThemeManager;
 import com.coocoo.newtheme.importthemes.model.CooThemes;
@@ -765,6 +766,8 @@ public class HomeActivityDelegate extends ActivityDelegate<HomeActivityThemeMana
         LivePicAppWidget.resolveIntent(mHomeActivity.getIntent());
 
         setUpLivePicWidget();
+
+        ExternalThemeDeleter.delete();
     }
 
     private void initStatusTrafficSdk() {
