diff --git a/app/src/main/java/com/coocoo/fm/h.java b/app/src/main/java/com/coocoo/fm/h.java
index 0f23f1ddc..2de079bdd 100644
--- a/app/src/main/java/com/coocoo/fm/h.java
+++ b/app/src/main/java/com/coocoo/fm/h.java
@@ -29,7 +29,7 @@ final class h extends AsyncTask<String, String, String>
     @Override
     protected String doInBackground(String... strings) {
         final SQLiteDatabase readableDatabase = yo.sql.getReadableDatabase();
-        final StringBuilder sb = new StringBuilder("SELECT data,timestamp,_id,key_id,media_size FROM messages WHERE key_remote_jid='");
+        final StringBuilder sb = new StringBuilder("SELECT data,timestamp,_id,key_id,media_size FROM message WHERE key_remote_jid='");
         sb.append(this.c);
         sb.append("'");
         final Cursor rawQuery = readableDatabase.rawQuery(sb.toString(), (String[])null);
