diff --git a/app/src/main/java/X/cc_3Ao.java b/app/src/main/java/X/cc_3Ao.java
new file mode 100644
index 000000000..1b5f0acdf
--- /dev/null
+++ b/app/src/main/java/X/cc_3Ao.java
@@ -0,0 +1,8 @@
+package X;
+
+public class cc_3Ao {
+
+    public long A01(int r6) {
+        return 0L;
+    }
+}
diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 7ff796fbf..97cbf03f5 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -117,6 +117,7 @@ import X.c1_0gC;
 import X.c1_02d;
 import X.c1_2Ql;
 import X.c1_2pa;
+import X.cc_3Ao;
 import X.cc_3LJ;
 import X.cc_3Xe;
 
@@ -641,6 +642,8 @@ public class Coocoo {
     }
 
     public static void traceLog4(Object object) {
+        cc_3Ao cc3Ao = (cc_3Ao) object;
+        Log.d("Gene", "0x0:" + cc3Ao.A01(0x0) + " 0x8:" + cc3Ao.A01(0x8) + " 0xd:" + cc3Ao.A01(0xdd));
         Log.e(TAG, "-----------------4---------------- // object : " + object);
     }
 
@@ -648,6 +651,10 @@ public class Coocoo {
         Log.e(TAG, "-----------------5---------------- // value : " + value);
     }
 
+    public static void traceLog6(long value) {
+        Log.e(TAG, "-----------------5---------------- // value : " + value);
+    }
+
     public static void traceLogCodeStack() {
         try {
             throw new IllegalArgumentException("traceLogCodeStack");
diff --git a/whatsapp_modify_module/smali_classes2/X/2i2.smali b/whatsapp_modify_module/smali_classes2/X/2i2.smali
index b3a0af478..6f4afa769 100644
--- a/whatsapp_modify_module/smali_classes2/X/2i2.smali
+++ b/whatsapp_modify_module/smali_classes2/X/2i2.smali
@@ -14979,6 +14979,10 @@
     :cond_130
     check-cast v0, LX/3zo;
 
+    const-string v1, "Gene Debug 1"
+
+    invoke-static {v1}, Lcom/coocoo/coocoo/Coocoo;->traceLog3(Ljava/lang/String;)V
+
     .line 436103
     iget-object v0, v0, LX/3zo;->A00:Ljava/lang/ref/WeakReference;
 
@@ -14990,12 +14994,21 @@
 
     if-nez v0, :cond_131
 
+    const-string v1, "Gene Debug 2"
+
+    invoke-static {v1}, Lcom/coocoo/coocoo/Coocoo;->traceLog3(Ljava/lang/String;)V
+
     const/4 v6, 0x0
 
     return-object v6
 
     .line 436104
     :cond_131
+
+    const-string v1, "Gene Debug 3"
+
+    invoke-static {v1}, Lcom/coocoo/coocoo/Coocoo;->traceLog3(Ljava/lang/String;)V
+
     iget-object v1, v0, LX/2xw;->A03:LX/2Ql;
 
     .line 436105
@@ -15044,6 +15057,8 @@
 
     check-cast v1, LX/3Ao;
 
+    invoke-static {v1}, Lcom/coocoo/coocoo/Coocoo;->traceLog4(Ljava/lang/Object;)V
+
     const/16 v0, 0xd
 
     .line 436111
