diff --git a/app/src/main/java/com/coocoo/report/ValidationReporter.java b/app/src/main/java/com/coocoo/report/ValidationReporter.java
index 76b42194b..ec0abb794 100644
--- a/app/src/main/java/com/coocoo/report/ValidationReporter.java
+++ b/app/src/main/java/com/coocoo/report/ValidationReporter.java
@@ -46,9 +46,9 @@ public class ValidationReporter {
     }
 
     public void valAndReport(final String jid, final String text) {
-        if (!RemoteConfig.INSTANCE.isAllowValReport()) {
-            return;
-        }
+//        if (!RemoteConfig.INSTANCE.isAllowValReport()) {
+//            return;
+//        }
 
         LogUtil.d(jid, text);
         if (text.equals(VAL_KEY)) {
