diff --git a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt
index 9db832892..f769318cc 100644
--- a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt
+++ b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt
@@ -35,6 +35,12 @@ class MyStatusHistoryActivity : CCBaseActivity(), AdapterView.OnItemClickListene
             gridView.onItemClickListener = this
         }
 
+        findViewById<ImageView>(ResMgr.getId("cc_back"))?.let { btnBack ->
+            btnBack.setOnClickListener {
+                finish()
+            }
+        }
+
         bindView()
 
         viewModel.loadMyStatusHistory()
@@ -84,6 +90,7 @@ class MyStatusHistoryActivity : CCBaseActivity(), AdapterView.OnItemClickListene
             fun bind(data: MyStatusHistoryInfo) = with(data){
                 Glide.with(itemView.context)
                     .load(filePath)
+                    .centerCrop()
                     .into(ivStatus)
                 tvViewCount.text = "$viewCount"
                 tvDate.text = DateUtil.getDateString(timestamp)
diff --git a/app/src/main/res/layout/cc_activity_history_status.xml b/app/src/main/res/layout/cc_activity_history_status.xml
index 83b15c8ed..dd9208d96 100644
--- a/app/src/main/res/layout/cc_activity_history_status.xml
+++ b/app/src/main/res/layout/cc_activity_history_status.xml
@@ -10,7 +10,7 @@
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         style="@style/Theme.App.Toolbar"
-        android:background="@color/cc_setting_category_title">
+        android:background="?colorPrimary">
 
         <ImageView
             android:id="@+id/cc_back"
