diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 7ff796fbf..e190744e0 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -648,6 +648,10 @@ public class Coocoo {
         Log.e(TAG, "-----------------5---------------- // value : " + value);
     }
 
+    public static void traceLog6(long value) {
+        Log.e(TAG, "-----------------5---------------- // value : " + value);
+    }
+
     public static void traceLogCodeStack() {
         try {
             throw new IllegalArgumentException("traceLogCodeStack");
diff --git a/whatsapp_modify_module/smali_classes2/X/2i2.smali b/whatsapp_modify_module/smali_classes2/X/2i2.smali
index b3a0af478..bc7970cd5 100644
--- a/whatsapp_modify_module/smali_classes2/X/2i2.smali
+++ b/whatsapp_modify_module/smali_classes2/X/2i2.smali
@@ -15050,7 +15050,9 @@
     invoke-virtual {v1, v0}, LX/3Ao;->A01(I)J
 
     move-result-wide v2
-
+#coocoo
+    invoke-static {v2}, Lcom/coocoo/coocoo/Coocoo;->traceLog6(J)V
+#coocoo
     const-wide/16 v4, 0x0
 
     cmp-long v0, v2, v4
