diff --git a/app/src/main/java/com/coocoo/openchat/activity/OpenChatWebActivity.kt b/app/src/main/java/com/coocoo/openchat/activity/OpenChatWebActivity.kt
index c33f49c8e..6c5460999 100644
--- a/app/src/main/java/com/coocoo/openchat/activity/OpenChatWebActivity.kt
+++ b/app/src/main/java/com/coocoo/openchat/activity/OpenChatWebActivity.kt
@@ -330,6 +330,11 @@ class OpenChatWebActivity : cc_01X() {
     }
 
     private fun showNoNetworkView(noNetwork: Boolean) {
+        try {
+            throw Exception("showNoNetworkView")
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
         notNetworkView?.visibility = View.VISIBLE
         if (noNetwork) {
             reloadButton?.visibility = View.VISIBLE
