diff --git a/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt b/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
index 06c9db456..e3b74a732 100644
--- a/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
+++ b/app/src/main/java/com/coocoo/preference/CoocooBackupRestorePrefFragment.kt
@@ -148,15 +148,15 @@ class CoocooBackupRestorePrefFragment
             preference.onPreferenceClickListener = this@CoocooBackupRestorePrefFragment
             preference.onPreferenceChangeListener = this@CoocooBackupRestorePrefFragment
             preference.isEnabled = false
-            val strCalculating = ResMgr.getString("cc_backup_whatsapp_data_clear_backup_calculating")
-            preference.summary = ResMgr.getString("cc_backup_whatsapp_data_clear_backup_content", strCalculating)
+            val strCalculating = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_CALCULATING)
+            preference.summary = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_CONTENT, strCalculating)
             prefFragmentScope.launch {
                 val size = getFormattedBackUpFileSize()
                 preference.summary = if (size > 0) {
                     preference.isEnabled = true
-                    ResMgr.getString("cc_backup_whatsapp_data_clear_backup_content", "%.2f MB".format(size))
+                    ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_CONTENT, "%.2f MB".format(size))
                 } else {
-                    ResMgr.getString("cc_backup_whatsapp_data_clear_backup_content", "0 MB")
+                    ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_CONTENT, "0 MB")
                 }
             }
         }
@@ -188,8 +188,8 @@ class CoocooBackupRestorePrefFragment
                 PREF_KEY_CLEAR_BACKUP -> {
                     ConfirmSignOutDialog(
                         context = this@CoocooBackupRestorePrefFragment.activity,
-                        title = ResMgr.getString("cc_delete_now_title"),
-                        message = ResMgr.getString("cc_delete_now_content"),
+                        title = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_DELETE_NOW_TITLE),
+                        message = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_DELETE_NOW_CONTENT),
                         onConfirmClicked = ::confirmDelete,
                     ).also {
                         confirmDialog = it
@@ -420,7 +420,7 @@ class CoocooBackupRestorePrefFragment
     private fun confirmDelete(view: View) {
         val dialog = ProgressDialog(
             context = activity,
-            title = ResMgr.getString("cc_backup_whatsapp_data_clear_backup_dialog_deleting_title"),
+            title = ResMgr.getString(Constants.Res.String.CLEAR_BACKUP_DATA_DELETING),
             message = ResMgr.getString("cc_universal_settings_please_wait"),
             isIndeterminate = true
         )
diff --git a/app/src/main/java/com/coocoo/utils/Constants.kt b/app/src/main/java/com/coocoo/utils/Constants.kt
index 8ebe05488..22ad7c755 100644
--- a/app/src/main/java/com/coocoo/utils/Constants.kt
+++ b/app/src/main/java/com/coocoo/utils/Constants.kt
@@ -517,8 +517,6 @@ object Constants {
             const val BACKUP_PROGRESS_MESSAGE = "cc_dialog_backup_progress_message"
             const val BACKUP_COMPLETE_MESSAGE = "cc_dialog_backup_complete_message"
             const val BACKUP_FAILED_MESSAGE = "cc_dialog_backup_failed_title"
-            const val BACKUP_RESTORE_CLEAR_TOAST_DONE = "cc_backup_whatsapp_data_clear_backup_toast_done"
-            const val BACKUP_RESTORE_CLEAR_TOAST_FAIL = "cc_backup_whatsapp_data_clear_backup_toast_fail"
 
             // Backup
             const val BACKUP_STATUS_MESSAGE_PREPARE = "cc_backup_status_message_prepare"
@@ -556,6 +554,15 @@ object Constants {
             const val RESTORE_SUCCESS_RESTART_DIALOG_TITLE = "cc_dialog_restore_restart_title"
             const val RESTORE_SUCCESS_RESTART_DIALOG_MESSAGE = "cc_dialog_restore_restart_message"
 
+            // clear BackUp Data
+            const val CLEAR_BACKUP_DATA_CALCULATING = "cc_backup_whatsapp_data_clear_backup_calculating"
+            const val CLEAR_BACKUP_DATA_CONTENT = "cc_backup_whatsapp_data_clear_backup_content"
+            const val CLEAR_BACKUP_DATA_DELETE_NOW_TITLE = "cc_delete_now_title"
+            const val CLEAR_BACKUP_DATA_DELETE_NOW_CONTENT = "cc_delete_now_content"
+            const val CLEAR_BACKUP_DATA_DELETING = "cc_backup_whatsapp_data_clear_backup_dialog_deleting_title"
+            const val BACKUP_RESTORE_CLEAR_TOAST_DONE = "cc_backup_whatsapp_data_clear_backup_toast_done"
+            const val BACKUP_RESTORE_CLEAR_TOAST_FAIL = "cc_backup_whatsapp_data_clear_backup_toast_fail"
+
             // Google drive backup
             const val SETTINGS_GDRIVE_GOTO_BACKUP_AND_RESTORE = "cc_settings_gdrive_goto_backup_and_restore"
 
