diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt
index c7b33d828..aa22ade7f 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt
@@ -10,6 +10,7 @@ import android.hardware.Camera
 import android.net.ConnectivityManager
 import android.opengl.GLSurfaceView
 import android.opengl.Matrix
+import android.os.Build
 import android.os.Handler
 import android.view.MotionEvent
 import android.view.View
@@ -703,6 +704,18 @@ class CameraUiDelegate(val activity: cc_0Ay, val uiHelperProvider: cc_0MI) {
         uiHelperProvider.AAN().A0C.visibility = if (show) View.VISIBLE else View.GONE
         uiHelperProvider.AAN().A0C.alpha = if (show) 1.0F else 0.0F
         uiHelperProvider.AAN().A06.visibility = if (show) View.INVISIBLE else View.VISIBLE  // Hide the camera_control view at showing the gallery view.
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            Handler().postDelayed({
+                LogUtil.d("=============================" +
+                        "Is A0C attached:${uiHelperProvider.AAN().A0C.isAttachedToWindow}\n, " +
+                        "id: ${uiHelperProvider.AAN().A0C.id}\n" +
+                        "visibility: ${uiHelperProvider.AAN().A0C.visibility}\n" +
+                        "alpha: ${uiHelperProvider.AAN().A0C.alpha}\n" +
+                        "width: ${uiHelperProvider.AAN().A0C.width} height: ${uiHelperProvider.AAN().A0C.height}\n" +
+                        "position: (${uiHelperProvider.AAN().A0C.x}, ${uiHelperProvider.AAN().A0C.y})\n"
+                )
+            }, 1000)
+        }
     }
 
     private fun switchFilter() {
