diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 5622d7dcf..3f9bd0e78 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -648,6 +648,10 @@ public class Coocoo {
         Log.e(TAG, "-----------------5---------------- // value : " + value);
     }
 
+    public static void traceLog6(int value1, int value) {
+        Log.e(TAG, "-----------------5---------------- // version old : " + value1 + " version new: " + value);
+    }
+
     public static void traceMediaComposerNewIntent(cc_3LJ value) {
         String valueMsg = new StringBuilder()
                 .append("max_items: " + value.A00)
diff --git a/app/src/main/java/com/coocoo/fm/h.java b/app/src/main/java/com/coocoo/fm/h.java
index 0f23f1ddc..73c74f66f 100644
--- a/app/src/main/java/com/coocoo/fm/h.java
+++ b/app/src/main/java/com/coocoo/fm/h.java
@@ -29,7 +29,7 @@ final class h extends AsyncTask<String, String, String>
     @Override
     protected String doInBackground(String... strings) {
         final SQLiteDatabase readableDatabase = yo.sql.getReadableDatabase();
-        final StringBuilder sb = new StringBuilder("SELECT data,timestamp,_id,key_id,media_size FROM messages WHERE key_remote_jid='");
+        final StringBuilder sb = new StringBuilder("SELECT _id FROM chat_view WHERE raw_string_jid='");
         sb.append(this.c);
         sb.append("'");
         final Cursor rawQuery = readableDatabase.rawQuery(sb.toString(), (String[])null);
@@ -41,9 +41,7 @@ final class h extends AsyncTask<String, String, String>
                 }
             } while (rawQuery.moveToNext());
             rawQuery.close();
-            final StringBuilder sb2 = new StringBuilder("SELECT key_id FROM messages WHERE key_remote_jid='");
-            sb2.append(this.c);
-            sb2.append("' AND _id=");
+            final StringBuilder sb2 = new StringBuilder("SELECT key_id FROM message WHERE chat_row_id=");
             sb2.append(this.b);
             final Cursor rawQuery2 = readableDatabase.rawQuery(sb2.toString(), (String[])null);
             rawQuery2.moveToFirst();
diff --git a/whatsapp_modify_module/smali_classes2/X/2jf.smali b/whatsapp_modify_module/smali_classes2/X/2jf.smali
index c0c6cb232..2faf8aec9 100644
--- a/whatsapp_modify_module/smali_classes2/X/2jf.smali
+++ b/whatsapp_modify_module/smali_classes2/X/2jf.smali
@@ -116,9 +116,10 @@
 
 .method public constructor <init>(Landroid/content/Context;LX/02d;LX/2SZ;LX/2sw;LX/2Vr;LX/2SS;LX/2R0;Ljava/io/File;)V
     .locals 3
-
-    const/4 v2, 0x1
-
+#coocoo
+    #const/4 v2, 0x1
+    const/4 v2, 0x3
+#coocoo
     const-string v0, "msgstore.db"
 
     const/4 v1, 0x0
@@ -12017,6 +12018,9 @@
 
     .line 450754
     invoke-static {v1, v0, p2, p3}, LX/0FM;->A00(Ljava/lang/String;Ljava/lang/String;II)V
+#coocoo
+    invoke-static {p2, p3}, Lcom/coocoo/coocoo/Coocoo;->traceLog6(II)V
+#coocoo
 
     .line 450755
     invoke-virtual {p0, p1}, Landroid/database/sqlite/SQLiteOpenHelper;->onCreate(Landroid/database/sqlite/SQLiteDatabase;)V
