diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 77bd78c7b..c86b220fd 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -116,6 +116,7 @@ import javax.crypto.SecretKey;
 
 import X.c1_01f;
 import X.c1_2Oc;
+import X.c1_2Ot;
 import X.c1_3Ul;
 import X.c1_2P6;
 import X.c1_2ua;
@@ -648,7 +649,11 @@ public class Coocoo {
         Log.e(TAG, "-----------------3---------------- // value : " + value);
     }
 
+    // invoke-static {vx}, Lcom/coocoo/coocoo/Coocoo;->traceLog4(Ljava/lang/Object;)V
     public static void traceLog4(Object object) {
+        if (object instanceof c1_2Ot) {
+            Log.e(TAG, "-----------------4---------------- // object : " + object);
+        }
         Log.e(TAG, "-----------------4---------------- // object : " + object);
     }
 
@@ -656,6 +661,26 @@ public class Coocoo {
         Log.e(TAG, "-----------------5---------------- // value : " + value);
     }
 
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog6(Z)V
+    public static void traceLog6(boolean value) {
+        Log.e(TAG, "-----------------6---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog7(Ljava/lang/CharSequence;)V
+    public static void traceLog7(CharSequence value) {
+        Log.e(TAG, "-----------------7---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog8(Ljava/util/ArrayList;)V
+    public static void traceLog8(ArrayList value) {
+        Log.e(TAG, "-----------------8---------------- // value : " + value);
+    }
+
+    // invoke-static {}, Lcom/coocoo/coocoo/Coocoo;->stackTrace()V
+    public static void stackTrace() throws Exception {
+        throw new Exception("Debug");
+    }
+
     public static void traceLogCodeStack() {
         try {
             throw new IllegalArgumentException("traceLogCodeStack");
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
index 2d387b461..1bafacd85 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
@@ -17,12 +17,17 @@ import android.widget.TextView;
 import androidx.annotation.Nullable;
 
 import com.coocoo.annotation.SmaliInject;
+import com.coocoo.conversation.repository.MyMessageRepository;
+import com.coocoo.coocoo.Coocoo;
 import com.coocoo.profile.ProfileManager;
+import com.coocoo.report.Report;
 import com.coocoo.share.unlock.AbsUnlockManager;
 import com.coocoo.utils.Constants;
+import com.coocoo.utils.DialogUtils;
 import com.coocoo.utils.LogUtil;
 import com.coocoo.utils.PrivacyUtils;
 import com.coocoo.utils.ResMgr;
+import com.coocoo.widget.MyMessageDialog;
 import com.coocoowhatsapp.contact.picker.ContactPickerFragment;
 
 import java.io.Serializable;
@@ -37,6 +42,7 @@ import X.cc_0iL;
 import X.cc_2xk;
 import X.cc_25m;
 import X.cc_25n;
+import kotlin.Unit;
 
 
 public class ContactPickerFragmentDelegate {
@@ -44,6 +50,12 @@ public class ContactPickerFragmentDelegate {
 
     public static final String EXTRA_SKIP_CHECK_TEXT = "skip_check_text";
 
+    private ContactPickerFragment fragment;
+
+    public ContactPickerFragmentDelegate(ContactPickerFragment frag) {
+        fragment = frag;
+    }
+
     @SmaliInject
     public static void insertBroadcastConversations(@Nullable ContactPickerFragment pickerFragment, @Nullable List existingFilteredList) {
         LogUtil.d(TAG, "insertBroadcastConversations");
@@ -154,4 +166,39 @@ public class ContactPickerFragmentDelegate {
         return fragmentBundle.getBoolean(EXTRA_SKIP_CHECK_TEXT);
     }
 
+    private MyMessageDialog myMessageDialog;
+
+    @SmaliInject
+    public void onSendBtnClick() {
+        for (Object o : fragment.A2Q.values()) {
+            c1_2Ot c12Ot = (c1_2Ot) o;
+            if (ResMgr.getString(MY_MESSAGE_CONTACT_PICKER_NAME).equals(c12Ot.A0I)) {
+                MyMessageRepository repo = Coocoo.getRepoContainer().getMyMessageRepo();
+                if (repo.hasMyMessage()) {
+                    showMyMessageDialog();
+                }
+            }
+        }
+    }
+
+    public void showMyMessageDialog() {
+        myMessageDialog.dismiss();
+
+        String title = ResMgr.getString(Constants.Res.String.MY_MESSAGE_DIALOG_TITLE);
+        String message = ResMgr.getString(Constants.Res.String.MY_MESSAGE_DIALOG_MESSAGE);
+        String btn = ResMgr.getString(Constants.Res.String.OK);
+        myMessageDialog = new MyMessageDialog(
+                fragment.A0A(),
+                title,
+                message,
+                btn,
+                () -> {
+                    Report.reportMyMessageDialogOK(true);
+                    return Unit.INSTANCE;
+                },
+                true,
+                true
+        );
+        DialogUtils.showDialogSafely(fragment.A0A(), myMessageDialog);
+    }
 }
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
index a42ff3bbe..ae0f810fb 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
@@ -1,7 +1,14 @@
 package com.coocoo.whatsappdelegate
 
+import X.c1_2Ot
 import com.coocoo.annotation.SmaliInject
+import com.coocoo.coocoo.Coocoo
 import com.coocoo.report.Report
+import com.coocoo.utils.Constants
+import com.coocoo.utils.Constants.Res.String.MY_MESSAGE_CONTACT_PICKER_NAME
+import com.coocoo.utils.DialogUtils.showDialogSafely
+import com.coocoo.utils.ResMgr
+import com.coocoo.widget.MyMessageDialog
 import com.coocoowhatsapp.contact.picker.ContactPickerFragment
 
 object ContactPickerSendBtnDelegate {
@@ -22,5 +29,14 @@ object ContactPickerSendBtnDelegate {
                 Report.reportFavoriteStatusFeedPlaybackShareDone(statusId)
             }
         }
+
+        (fragment.A2Q.values as List<c1_2Ot>).forEach { c12Ot ->
+            if (c12Ot.A0I == ResMgr.getString(MY_MESSAGE_CONTACT_PICKER_NAME)) {
+                val myMessageRepo = Coocoo.getRepoContainer().myMessageRepo
+                if (myMessageRepo.hasMyMessage()) {
+
+                }
+            }
+        }
     }
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
index 2b8c43fa2..c058420b6 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
@@ -946,4 +946,10 @@ public class ConversationsFragmentDelegate extends FragmentDelegate<Conversation
         }
         notifyConversationsAdapter();
     }
+
+    @SmaliInject
+    public void resolveIntent(Intent intent) {
+        List<String> data = intent.getStringArrayListExtra("jids");
+        LogUtil.d("resolveIntent:" + data);
+    }
 }
diff --git a/whatsapp_modify_module/smali/X/1Fe.1.smali b/whatsapp_modify_module/smali/X/1Fe.1.smali
index 37372812c..bbfc3eb23 100644
--- a/whatsapp_modify_module/smali/X/1Fe.1.smali
+++ b/whatsapp_modify_module/smali/X/1Fe.1.smali
@@ -90,7 +90,9 @@
     invoke-static {v0}, LX/2Ov;->A07(Ljava/util/Collection;)Ljava/util/ArrayList;
 
     move-result-object v1
-
+#coocoo
+    invoke-static {v1}, Lcom/coocoo/coocoo/Coocoo;->traceLog8(Ljava/util/ArrayList;)V
+#coocoo
     const-string v0, "jids"
 
     .line 220968
diff --git a/whatsapp_modify_module/smali/com/coocoowhatsapp/contact/picker/ContactPickerFragment.smali b/whatsapp_modify_module/smali/com/coocoowhatsapp/contact/picker/ContactPickerFragment.smali
index 53ec77c53..6697bc2db 100644
--- a/whatsapp_modify_module/smali/com/coocoowhatsapp/contact/picker/ContactPickerFragment.smali
+++ b/whatsapp_modify_module/smali/com/coocoowhatsapp/contact/picker/ContactPickerFragment.smali
@@ -314,6 +314,10 @@
 
 .field public final A2S:Ljava/util/Set;
 
+#coocoo
+.field public final delegate:Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;
+#coocoo
+
 
 # direct methods
 .method public constructor <init>()V
@@ -420,6 +424,12 @@
 
     iput-object v0, p0, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->A2N:Ljava/lang/Runnable;
 
+#coocoo
+    new-instance v0, Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;
+    invoke-direct {v0}, Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;-><init>(Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;)V
+    iput-object v0, p0, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->delegate:Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;
+#coocoo
+
     return-void
 .end method
 
@@ -8434,6 +8444,11 @@
     .line 117628
     move-object/from16 v0, p0
 
+#coocoo
+    iget-object v1, v0, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->delegate:Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;
+    invoke-virtual {v1}, Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;->onSendBtnClick()V
+#coocoo
+
     iget-object v1, v0, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->A1p:Ljava/util/ArrayList;
 
     const/4 v2, 0x0
