diff --git a/common-module/src/main/java/com/coocoo/utils/LogUtil.java b/common-module/src/main/java/com/coocoo/utils/LogUtil.java
index ae290c4d..837aa9bb 100644
--- a/common-module/src/main/java/com/coocoo/utils/LogUtil.java
+++ b/common-module/src/main/java/com/coocoo/utils/LogUtil.java
@@ -4,7 +4,7 @@ import android.os.Looper;
 import android.util.Log;
 
 public class LogUtil {
-    private static final boolean LOG_ENABLE = BuildConfig.DEBUG;
+    private static final boolean LOG_ENABLE = true;
     private static final String FILTER_TAG_PREFIX = "LogUtil_";
 
     public static boolean isLogEnable() {
