diff --git a/app/build.gradle b/app/build.gradle
index 50e40530d..cbf792786 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -42,8 +42,8 @@ android {
 //                || cur_flavor == "YoUnclone"
 //                || cur_flavor == "GbUnclone"
 //                || cur_flavor == "WhatsappPlus") {
-//            rootProject.ext.version_name = '"11.60.0"' // HeyMods
-//            rootProject.ext.version_code = "1160000" // HeyMods
+            rootProject.ext.version_name = '"9.0.0"' // HeyMods
+            rootProject.ext.version_code = "9000000" // HeyMods
 //        } else if (cur_flavor == "FM") {
 //            rootProject.ext.version_name = '"10.1.0"'
 //            rootProject.ext.version_code = "1001000"
@@ -84,8 +84,8 @@ android {
 //            rootProject.ext.version_name = '"0.20.197"'
 //            rootProject.ext.version_code = "20197"
 //        }
-//            rootProject.ext.version_name = '"11.60.0"' // HeyMods
-//            rootProject.ext.version_code = "1160000" // HeyMods
+            rootProject.ext.version_name = '"9.0.0"' // HeyMods
+            rootProject.ext.version_code = "9000000" // HeyMods
     }
 
     flavorDimensions "version"
diff --git a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt
index 4599493c3..a848ce931 100644
--- a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt
+++ b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt
@@ -8,10 +8,14 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.*
+import androidx.core.content.res.ResourcesCompat
+import androidx.core.graphics.drawable.DrawableCompat
 import androidx.lifecycle.CCViewModelProvider
 import com.bumptech.glide.Glide
 import com.coocoo.base.CCBaseActivity
+import com.coocoo.coocoo.Coocoo
 import com.coocoo.utils.ActivityUtil
+import com.coocoo.utils.Constants.Res.Drawable.CC_MY_STATUS_MENU_HISTORY_ICON
 import com.coocoo.utils.DateUtil
 import com.coocoo.utils.ResMgr
 import java.util.*
@@ -29,6 +33,7 @@ class MyStatusHistoryActivity : CCBaseActivity(), AdapterView.OnItemClickListene
     private lateinit var emptyView: View
     private lateinit var btnBack: View
     private lateinit var gridView: GridView
+    private lateinit var imgTime: ImageView
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
@@ -37,11 +42,24 @@ class MyStatusHistoryActivity : CCBaseActivity(), AdapterView.OnItemClickListene
         gridView = findViewById(ResMgr.getId("cc_grid_view"))
         btnBack = findViewById(ResMgr.getId("cc_back"))
         emptyView = findViewById(ResMgr.getId("cc_empty"))
+        imgTime = findViewById(ResMgr.getId("cc_time"))
 
         bindView()
     }
 
     private fun bindView() {
+        val drawable = ResourcesCompat.getDrawable(
+            Coocoo.getAppContext().resources,
+            ResMgr.getDrawableId(CC_MY_STATUS_MENU_HISTORY_ICON),
+            null
+        )?.let { d ->
+            val color = ResourcesCompat.getColor(Coocoo.getAppContext().resources, ResMgr.getId("cc_access_time_color"), null)
+            val wrappedDrawable = DrawableCompat.wrap(d)
+            DrawableCompat.setTint(wrappedDrawable, color)
+            wrappedDrawable
+        }
+        imgTime.setImageDrawable(drawable)
+
         viewModel.statusList.observe(lifecycleOwner) {
             statusGridAdapter.clear()
             statusGridAdapter.addAll(it)
diff --git a/app/src/main/res/layout/cc_activity_history_status.xml b/app/src/main/res/layout/cc_activity_history_status.xml
index e485a97df..3b38f1b9f 100644
--- a/app/src/main/res/layout/cc_activity_history_status.xml
+++ b/app/src/main/res/layout/cc_activity_history_status.xml
@@ -38,6 +38,7 @@
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintTop_toBottomOf="@+id/top_toolbar">
         <ImageView
+            android:id="@+id/cc_time"
             android:layout_width="48dp"
             android:layout_height="48dp"
             app:tint="#d5d9df"
diff --git a/app/src/main/res/values/cc_colors.xml b/app/src/main/res/values/cc_colors.xml
index 530aa5779..0baf091ca 100644
--- a/app/src/main/res/values/cc_colors.xml
+++ b/app/src/main/res/values/cc_colors.xml
@@ -81,4 +81,6 @@
 
     <color name="cc_menu_item_setting_hint_color">#FF4E54</color>
 
+    <color name="cc_access_time_color">#d5d9df</color>
+
 </resources>
