diff --git a/app/build.gradle b/app/build.gradle
index 50e40530d..cbf792786 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -42,8 +42,8 @@ android {
 //                || cur_flavor == "YoUnclone"
 //                || cur_flavor == "GbUnclone"
 //                || cur_flavor == "WhatsappPlus") {
-//            rootProject.ext.version_name = '"11.60.0"' // HeyMods
-//            rootProject.ext.version_code = "1160000" // HeyMods
+            rootProject.ext.version_name = '"9.0.0"' // HeyMods
+            rootProject.ext.version_code = "9000000" // HeyMods
 //        } else if (cur_flavor == "FM") {
 //            rootProject.ext.version_name = '"10.1.0"'
 //            rootProject.ext.version_code = "1001000"
@@ -84,8 +84,8 @@ android {
 //            rootProject.ext.version_name = '"0.20.197"'
 //            rootProject.ext.version_code = "20197"
 //        }
-//            rootProject.ext.version_name = '"11.60.0"' // HeyMods
-//            rootProject.ext.version_code = "1160000" // HeyMods
+            rootProject.ext.version_name = '"9.0.0"' // HeyMods
+            rootProject.ext.version_code = "9000000" // HeyMods
     }
 
     flavorDimensions "version"
diff --git a/app/src/main/java/com/coocoo/db/room/mystatus/dao/MyStatusDao.kt b/app/src/main/java/com/coocoo/db/room/mystatus/dao/MyStatusDao.kt
index 694dfb2f1..6ab0c8329 100644
--- a/app/src/main/java/com/coocoo/db/room/mystatus/dao/MyStatusDao.kt
+++ b/app/src/main/java/com/coocoo/db/room/mystatus/dao/MyStatusDao.kt
@@ -10,8 +10,8 @@ interface MyStatusDao {
     fun insertMyStatusHistory(vararg myStatusHistoryEntity: MyStatusHistoryEntity)
 
     @Query("SELECT * FROM my_status_history")
-    fun getMyStatusHistoryEntities(): List<MyStatusHistoryEntity>
+    suspend fun getMyStatusHistoryEntities(): List<MyStatusHistoryEntity>
 
     @Query("DELETE FROM my_status_history WHERE message_row_id = :messageRowId")
-    fun deleteMyStatusHistory(messageRowId: Long)
+    suspend fun deleteMyStatusHistory(messageRowId: Long)
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt
index fa29b64e9..3b6a0bf1c 100644
--- a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt
+++ b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryActivity.kt
@@ -11,6 +11,7 @@ import android.widget.*
 import androidx.lifecycle.CCViewModelProvider
 import com.bumptech.glide.Glide
 import com.coocoo.base.CCBaseActivity
+import com.coocoo.exoplayer2.util.Log
 import com.coocoo.utils.ActivityUtil
 import com.coocoo.utils.DateUtil
 import com.coocoo.utils.ResMgr
@@ -42,12 +43,11 @@ class MyStatusHistoryActivity : CCBaseActivity(), AdapterView.OnItemClickListene
         }
 
         bindView()
-
-        viewModel.loadMyStatusHistory()
     }
 
     private fun bindView() {
         viewModel.statusList.observe(lifecycleOwner) {
+            statusGridAdapter.clear()
             statusGridAdapter.addAll(it)
         }
     }
diff --git a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryListRepository.kt b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryListRepository.kt
index 30329698c..a5811e697 100644
--- a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryListRepository.kt
+++ b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryListRepository.kt
@@ -2,16 +2,21 @@ package com.coocoo.statuses.myhistory
 
 import X.c1_2Ql
 import X.cc_2Qk
-import android.database.Cursor
 import android.os.Environment
 import android.util.Log
 import androidx.annotation.WorkerThread
+import androidx.lifecycle.CCLiveData
+import androidx.lifecycle.CCMutableLiveData
+import androidx.lifecycle.CCTransformations
 import com.coocoo.annotation.SmaliInject
 import com.coocoo.db.room.mystatus.MyStatusDatabase
 import com.coocoo.db.room.mystatus.entity.MyStatusHistoryEntity
 import com.coocoo.fm.yo
 import com.coocoo.utils.Constant
 import com.coocoo.utils.FileUtil
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
 import java.io.File
 
 class MyStatusHistoryListRepository(private val database: MyStatusDatabase) {
@@ -29,21 +34,35 @@ class MyStatusHistoryListRepository(private val database: MyStatusDatabase) {
         const val MEDIA_TYPE_PHOTO = 1
     }
 
-    fun getMyStatusHistoryInfoList(): List<MyStatusHistoryInfo> {
-        val myStatusHistoryEntities = database.getMyStatusDao().getMyStatusHistoryEntities()
+    private val _statusList: CCMutableLiveData<List<MyStatusHistoryEntity>> by lazy {
+        CCMutableLiveData<List<MyStatusHistoryEntity>>().also {
+            GlobalScope.launch(Dispatchers.Main) {
+                updateMyStatusHistoryInfoList(it)
+            }
+        }
+    }
 
-        val statusList = mutableListOf<MyStatusHistoryInfo>()
-        myStatusHistoryEntities.forEach { myStatusHistoryEntity ->
-            statusList.add(
+    fun getMyStatusHistoryInfoList(): CCLiveData<List<MyStatusHistoryInfo>> {
+        return CCTransformations.map(_statusList) { list ->
+            list.map { myStatusHistoryEntity ->
                 MyStatusHistoryInfo(
                     myStatusHistoryEntity.messageRowId,
                     myStatusHistoryEntity.filePath,
                     myStatusHistoryEntity.timestamp,
                     myStatusHistoryEntity.viewCount,
-                    myStatusHistoryEntity.messageType))
+                    myStatusHistoryEntity.messageType)
+            }
         }
+    }
+
+    private suspend fun updateMyStatusHistoryInfoList(liveData: CCMutableLiveData<List<MyStatusHistoryEntity>>) {
+        liveData.value = database.getMyStatusDao().getMyStatusHistoryEntities()
+    }
 
-        return statusList
+    suspend fun deleteStatus(messageRowId: Long) {
+        val dao = database.getMyStatusDao()
+        dao.deleteMyStatusHistory(messageRowId)
+        updateMyStatusHistoryInfoList(_statusList)
     }
 
     @SmaliInject
diff --git a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryPlaybackActivity.kt b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryPlaybackActivity.kt
index 2ad159790..494a590ee 100644
--- a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryPlaybackActivity.kt
+++ b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryPlaybackActivity.kt
@@ -1,5 +1,7 @@
 package com.coocoo.statuses.myhistory
 
+import X.c1_2Ql
+import androidx.lifecycle.CCViewModelProvider
 import com.coocoowhatsapp.status.playback.StatusPlaybackActivity
 import com.coocoo.statuses.myhistory.MyStatusHistoryInfo
 import com.coocoo.statuses.myhistory.MyStatusHistoryActivity
@@ -10,17 +12,22 @@ import java.util.ArrayList
 
 class MyStatusHistoryPlaybackActivity : StatusPlaybackActivity() {
 
+    val viewModel: MyStatusHistoryPlaybackViewModel = MyStatusHistoryPlaybackViewModel()
+
     val myStatusHistoryInfoList: ArrayList<MyStatusHistoryInfo>
         get() = intent.getParcelableArrayListExtra(MyStatusHistoryActivity.KEY_MY_STATUS_HISTORY_INFO_LIST)
 
     val statusStartPosition: Int
         get() = intent.getIntExtra(MyStatusHistoryActivity.KEY_MY_STATUS_START_POSITION, 0)
 
-    fun showConfirmDeleteDialog() {
+    fun showConfirmDeleteDialog(c12Ql: c1_2Ql) {
         CoocooAlertDialog(this,
             title = ResMgr.getString("cc_my_status_delete_now"),
             message = ResMgr.getString("cc_my_status_delete_now_message"),
-            onConfirmClicked = {},
+            onConfirmClicked = {
+                viewModel.deleteStatus(c12Ql)
+                finish()
+            },
             onCancelClicked = {}
         ).safelyShow()
     }
diff --git a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryPlaybackViewModel.kt b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryPlaybackViewModel.kt
new file mode 100644
index 000000000..a566a2ac1
--- /dev/null
+++ b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryPlaybackViewModel.kt
@@ -0,0 +1,18 @@
+package com.coocoo.statuses.myhistory
+
+import X.c1_2Ql
+import com.coocoo.coocoo.Coocoo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+class MyStatusHistoryPlaybackViewModel(
+    private val repo: MyStatusHistoryListRepository = Coocoo.getRepoContainer().myStatusHistoryListRepo
+) {
+
+    fun deleteStatus(c12Ql: c1_2Ql) {
+        GlobalScope.launch(Dispatchers.Main) {
+            repo.deleteStatus(c12Ql.A0y)
+        }
+    }
+}
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryViewModel.kt b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryViewModel.kt
index 25cc5a6be..d6c51afe9 100644
--- a/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryViewModel.kt
+++ b/app/src/main/java/com/coocoo/statuses/myhistory/MyStatusHistoryViewModel.kt
@@ -2,14 +2,21 @@ package com.coocoo.statuses.myhistory
 
 import androidx.lifecycle.*
 import com.coocoo.coocoo.Coocoo
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
 
-class MyStatusHistoryViewModel(private val repo: MyStatusHistoryListRepository) : CCViewModel() {
+class MyStatusHistoryViewModel(val repo: MyStatusHistoryListRepository) : CCViewModel() {
 
-    private val _statusListLive: CCMutableLiveData<List<MyStatusHistoryInfo>> = CCMutableLiveData<List<MyStatusHistoryInfo>>()
-    val statusList: CCLiveData<List<MyStatusHistoryInfo>> get() = _statusListLive
+    private val _statusList: CCMediatorLiveData<List<MyStatusHistoryInfo>> = CCMediatorLiveData()
+    val statusList: CCLiveData<List<MyStatusHistoryInfo>> get() = _statusList
+
+    init {
+        _statusList.addSource(repo.getMyStatusHistoryInfoList()) {
+            _statusList.value = it
+        }
+    }
+
+    fun getCurrentMyStatusHistoryInfoList(): List<MyStatusHistoryInfo> {
+        return _statusList.value ?: emptyList()
+    }
 
     @Suppress("UNCHECKED_CAST")
     class Factory : CCViewModelProvider.Factory {
@@ -18,16 +25,4 @@ class MyStatusHistoryViewModel(private val repo: MyStatusHistoryListRepository)
             return MyStatusHistoryViewModel(container.myStatusHistoryListRepo) as T
         }
     }
-
-    fun loadMyStatusHistory() {
-        viewModelScope.launch {
-            _statusListLive.value = withContext(Dispatchers.IO) {
-                repo.getMyStatusHistoryInfoList()
-            }
-        }
-    }
-
-    fun getCurrentMyStatusHistoryInfoList(): List<MyStatusHistoryInfo> {
-        return _statusListLive.value ?: emptyList()
-    }
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/StatusPlaybackContactFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/StatusPlaybackContactFragmentDelegate.java
index 042c5bde2..bd0ad0e7b 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/StatusPlaybackContactFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/StatusPlaybackContactFragmentDelegate.java
@@ -302,14 +302,14 @@ public class StatusPlaybackContactFragmentDelegate implements ILaunchInstallActi
     }
 
     @SmaliInject
-    public void onPageViewCreated(View view) {
+    public void onPageViewCreated(View view, c1_2Ql c12Ql) {
         Activity activity = mPlaybackFragment.A09();
         Log.d("StatusPlaybackContactFragmentDelegate", "activity" + activity);
         if (activity instanceof MyStatusHistoryPlaybackActivity) {
             ImageView btnDelete = view.findViewById(ResMgr.getId("delete"));
             if (btnDelete != null) {
                 btnDelete.setOnClickListener((v) -> {
-                    ((MyStatusHistoryPlaybackActivity) activity).showConfirmDeleteDialog();
+                    ((MyStatusHistoryPlaybackActivity) activity).showConfirmDeleteDialog(c12Ql);
                 });
             }
         }
diff --git a/whatsapp_modify_module/smali/com/coocoowhatsapp/status/playback/fragment/StatusPlaybackContactFragment.smali b/whatsapp_modify_module/smali/com/coocoowhatsapp/status/playback/fragment/StatusPlaybackContactFragment.smali
index d9133da71..527fa1a62 100644
--- a/whatsapp_modify_module/smali/com/coocoowhatsapp/status/playback/fragment/StatusPlaybackContactFragment.smali
+++ b/whatsapp_modify_module/smali/com/coocoowhatsapp/status/playback/fragment/StatusPlaybackContactFragment.smali
@@ -1258,9 +1258,11 @@
     .line 444385
     invoke-virtual {v2, v0}, LX/2xx;->A0K(Landroid/view/View;)V
 #coocoo
+    move-object/from16 v14, p1
+
     iget-object v15, v6, Lcom/coocoowhatsapp/status/playback/fragment/StatusPlaybackContactFragment;->statusPlaybackContactFragmentDelegate:Lcom/coocoo/whatsappdelegate/StatusPlaybackContactFragmentDelegate;
 
-    invoke-virtual {v15, v0}, Lcom/coocoo/whatsappdelegate/StatusPlaybackContactFragmentDelegate;->onPageViewCreated(Landroid/view/View;)V
+    invoke-virtual {v15, v0, v14}, Lcom/coocoo/whatsappdelegate/StatusPlaybackContactFragmentDelegate;->onPageViewCreated(Landroid/view/View;LX/2Ql;)V
 #coocoo
 
     .line 444386
