diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt
index c7b33d828..dedc38f1b 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/CameraUiDelegate.kt
@@ -697,6 +697,12 @@ class CameraUiDelegate(val activity: cc_0Ay, val uiHelperProvider: cc_0MI) {
     }
 
     private fun showGallery(show: Boolean) {
+        LogUtil.d("showGallery")
+        try {
+            throw Exception()
+        } catch (e: Exception) {
+            e.printStackTrace()
+        }
         if (show) {
             uiHelperProvider.AAN().A05.visibility = View.VISIBLE
         }
