diff --git a/app/build.gradle b/app/build.gradle
index 589ab98c..3c300024 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -57,8 +57,8 @@ android {
 //                || cur_flavor == "YoUnclone"
 //                || cur_flavor == "GbUnclone"
 //                || cur_flavor == "WhatsappPlus") {
-            rootProject.ext.version_name = '"19.41.4"' // HeyMods
-            rootProject.ext.version_code = "1941" // HeyMods
+            rootProject.ext.version_name = '"19.41.5"' // HeyMods
+            rootProject.ext.version_code = "11941509" // HeyMods
 //        } else if (cur_flavor == "FM") {
 //            rootProject.ext.version_name = '"10.1.0"'
 //            rootProject.ext.version_code = "1001000"
@@ -99,8 +99,8 @@ android {
 //            rootProject.ext.version_name = '"0.20.197"'
 //            rootProject.ext.version_code = "20197"
 //        }
-            rootProject.ext.version_name = '"19.41.4"' // HeyMods
-            rootProject.ext.version_code = "1941" // HeyMods
+            rootProject.ext.version_name = '"19.41.5"' // HeyMods
+            rootProject.ext.version_code = "11941509" // HeyMods
     }
 }
 
diff --git a/app/build.gradle.bak b/app/build.gradle.bak
index 58a79b39..589ab98c 100644
--- a/app/build.gradle.bak
+++ b/app/build.gradle.bak
@@ -144,4 +144,5 @@ dependencies {
     implementation 'androidx.browser:browser:1.3.0'
     // get gaid for max
     implementation 'com.google.android.gms:play-services-ads-identifier:17.0.0'
+    compileOnly 'com.airbnb.android:lottie:5.2.0'
 }
\ No newline at end of file
diff --git a/whatsapp_modify_module/AndroidManifest.xml b/whatsapp_modify_module/AndroidManifest.xml
index 42a4cf9b..19f49df7 100644
--- a/whatsapp_modify_module/AndroidManifest.xml
+++ b/whatsapp_modify_module/AndroidManifest.xml
@@ -1,5 +1,5 @@
 <?xml version='1.0' encoding='UTF-8'?>
-<manifest xmlns:android="http://schemas.android.com/apk/res/android" xmlns:tools="http://schemas.android.com/tools" android:compileSdkVersion="23" android:compileSdkVersionCodename="6.0-2438415" package="com.gbwhatsapp" platformBuildVersionCode="31" platformBuildVersionName="12" android:versionCode="1941" android:versionName="19.41.4">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" xmlns:tools="http://schemas.android.com/tools" android:compileSdkVersion="23" android:compileSdkVersionCodename="6.0-2438415" package="com.gbwhatsapp" platformBuildVersionCode="31" platformBuildVersionName="12" android:versionCode="11941509" android:versionName="19.41.5">
     <uses-feature android:name="android.hardware.bluetooth" android:required="false"/>
     <uses-feature android:name="android.hardware.location" android:required="false"/>
     <uses-feature android:name="android.hardware.location.network" android:required="false"/>
