diff --git a/app/src/main/java/com/coocoo/CooCoo.kt b/app/src/main/java/com/coocoo/CooCoo.kt
index c4cf8130..24fae3a1 100644
--- a/app/src/main/java/com/coocoo/CooCoo.kt
+++ b/app/src/main/java/com/coocoo/CooCoo.kt
@@ -143,7 +143,7 @@ object CooCoo {
             appFlavor,
             com.coocoo.config.BuildConfig.COOCOO_VERSION_CODE,
             appChannel,
-            com.coocoo.config.BuildConfig.IS_DEBUG,
+            false,
         )
         AdCommonManager.injectReporter(ReportStatusVideoTraffic)
     }
diff --git a/app/src/main/java/com/coocoo/conversation/ConversationRowTextDelegate.kt b/app/src/main/java/com/coocoo/conversation/ConversationRowTextDelegate.kt
index 4283c9d8..9283aa28 100644
--- a/app/src/main/java/com/coocoo/conversation/ConversationRowTextDelegate.kt
+++ b/app/src/main/java/com/coocoo/conversation/ConversationRowTextDelegate.kt
@@ -7,33 +7,63 @@ import android.net.Uri
 import com.coocoo.CooCoo
 import com.coocoo.annotation.SmaliInject
 import com.coocoo.utils.SystemUtil
-import com.coocoo.widget.InstallMaxBrowserDialog
+import com.status.traffic.data.DownloadManager
+import com.status.traffic.data.config.BrowserAdRemoteConfig
+import com.status.traffic.data.enums.InstallTag
+import com.status.traffic.data.vo.BrowserAdConfig
+import com.status.traffic.install.InstallActivity
+import com.status.traffic.ui.InstallMaxBrowserDialog
 
 object ConversationRowTextDelegate {
 
-    private const val MAX_PACKAGE_NAME = "com.kiwibrowser.browser"
+    private val config: BrowserAdConfig? = BrowserAdRemoteConfig.getRemoteConfig()?.get(0)
 
-    private val isMaxBrowserInstalled: Boolean = SystemUtil.checkAppInstalled(MAX_PACKAGE_NAME)
+    private val isBrowserInstalled: Boolean get() = SystemUtil.checkAppInstalled(config?.packageName)
 
     @JvmStatic
     @SmaliInject
     fun onHyperLinkClick(cc12P: cc_12P, context: Context, uri: Uri) {
         if (uri.scheme == "http" || uri.scheme == "https") {
-            if (isMaxBrowserInstalled) {
-                val intent = context.packageManager.getLaunchIntentForPackage(MAX_PACKAGE_NAME) ?: return
+            if (isBrowserInstalled) {
+                val intent = context.packageManager.getLaunchIntentForPackage(config?.packageName.orEmpty()) ?: return
                 intent.addCategory(Intent.CATEGORY_LAUNCHER)
                 intent.data = uri
                 context.startActivity(intent)
             } else {
-                showDownloadBrowserDialog(context)
+                showDownloadBrowserDialog(context, cc12P, uri)
             }
         } else {
             cc12P.Ab8(context, uri)
         }
     }
 
-    private fun showDownloadBrowserDialog(context: Context) {
-        InstallMaxBrowserDialog(context).show()
+    private fun showDownloadBrowserDialog(context: Context, cc12P: cc_12P, uri: Uri) {
+        InstallMaxBrowserDialog(context).apply {
+            content = config?.content.orEmpty()
+            setOnDismissListener {
+                cc12P.Ab8(context, uri)
+            }
+            onInstallClick = {
+                config?.let {
+                    if (checkFileExist()) {
+                        InstallActivity.launchInstallActivity(
+                            context,
+                            InstallTag.BROWSER_AD,
+                            it.getApkData(),
+                            null
+                        )
+                    } else {
+                        cc12P.Ab8(context, uri)
+                    }
+                }
+                dismiss()
+            }
+        }.show()
+    }
+
+    private fun checkFileExist(): Boolean {
+        val config = this.config ?: return false
+        return DownloadManager.isApkFileExists(config.key, config.packageName, config.apkMd5)
     }
 
     @JvmStatic
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt
index 46b22ff9..940f5f8e 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt
@@ -373,6 +373,7 @@ class HomeActivityDelegate @SmaliInject constructor(private val homeActivity: Ho
                 sdk.loadFabBootEndAd(mHostActivity)
                 sdk.loadConversationRevokeAd(mHostActivity)
                 sdk.loadCallListBannerAd(mHostActivity)
+                sdk.loadBrowserAd(mHostActivity)
                 sdk.setPushAppInterstitialListener { interstitialAd: PushAppConfig? ->
                     sdk.launchPushAppInterstitialAdActivity(mHostActivity, interstitialAd)
                     if (homeActivity != null) {
diff --git a/app/src/main/java/com/coocoo/widget/InstallMaxBrowserDialog.kt b/app/src/main/java/com/coocoo/widget/InstallMaxBrowserDialog.kt
deleted file mode 100644
index 042fffa8..00000000
--- a/app/src/main/java/com/coocoo/widget/InstallMaxBrowserDialog.kt
+++ /dev/null
@@ -1,29 +0,0 @@
-package com.coocoo.widget
-
-import android.content.Context
-import android.os.Bundle
-import android.widget.Button
-import android.widget.TextView
-import com.coocoo.utils.Constants
-import com.coocoo.utils.ResMgr
-import com.coocoo.utils.StringUtil
-import com.coocoo.utils.dialog.BaseDialog
-
-class InstallMaxBrowserDialog(context: Context) : BaseDialog(context) {
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-
-        setContentView(ResMgr.getLayoutId("cc_dialog_install_max_browser"))
-
-        findViewById<TextView>(ResMgr.getId("cc_btn_cancel")).text = StringUtil.getHtml(
-            ResMgr.getString(
-                "cc_underline_cancel"
-            )
-        )
-
-        findViewById<Button>(ResMgr.getId("cc_btn_cancel")).setOnClickListener {
-            dismiss()
-        }
-    }
-}
\ No newline at end of file
diff --git a/status-traffic-api/src/main/java/com/status/traffic/Constant.kt b/status-traffic-api/src/main/java/com/status/traffic/Constant.kt
index 76464dba..96907449 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/Constant.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/Constant.kt
@@ -61,6 +61,7 @@ object Constant {
             const val CHATWINDOW_TOP_BANNER_AD = "/ChatWindowTopBannerAd"
             const val CALL_LIST_BANNER_AD = "/CallListBannerAd"
             const val HOME_LIST_BANNER_AD = "/HomeListBannerAd"
+            const val BROWSER_AD = "/BrowserAd"
 
             const val STATUS_TRAFFIC_MAXIMUM_DATA_VOLUME = 2
             const val STATUS_TRAFFIC_A_PIECE_OF_DATA = 1
diff --git a/status-traffic-api/src/main/java/com/status/traffic/StatusTrafficSdk.kt b/status-traffic-api/src/main/java/com/status/traffic/StatusTrafficSdk.kt
index 5b4c7457..932724ab 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/StatusTrafficSdk.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/StatusTrafficSdk.kt
@@ -7,11 +7,7 @@ import android.content.Intent
 import android.net.Uri
 import android.os.Bundle
 import android.widget.RelativeLayout
-import com.coocoo.utils.AESCipherUtil
-import com.coocoo.utils.ActivityUtil
-import com.coocoo.utils.DialogUtils
-import com.coocoo.utils.ResMgr
-import com.coocoo.utils.SystemUtil
+import com.coocoo.utils.*
 import com.status.traffic.Constant.Report.Param.KEY_FROM
 import com.status.traffic.ads.AdSdkManager
 import com.status.traffic.ads.AdSdkManager.ISdkController
@@ -19,98 +15,22 @@ import com.status.traffic.ads.ad.OnLoadAdListener
 import com.status.traffic.data.ApiRemoteConfigService
 import com.status.traffic.data.ApiRemoteConfigServiceV2
 import com.status.traffic.data.DownloadManager
-import com.status.traffic.data.config.BannerRequestRemoteConfigV2
-import com.status.traffic.data.config.BaseRemoteConfig
-import com.status.traffic.data.config.CallDestroyAdRemoteConfig
-import com.status.traffic.data.config.CallListBannerAdRemoteConfig
-import com.status.traffic.data.config.ChatWindowTopBannerAdRemoteConfig
-import com.status.traffic.data.config.ConversationDestroyAdRemoteConfig
-import com.status.traffic.data.config.ConversationFabAdRemoteConfig
-import com.status.traffic.data.config.ConversationPickAudioAdRemoteConfig
-import com.status.traffic.data.config.ConversationPickVideoAdRemoteConfig
-import com.status.traffic.data.config.ConversationRevokeAdRemoteConfig
-import com.status.traffic.data.config.DownloadAudioAdRemoteConfig
-import com.status.traffic.data.config.DownloadVideoAdRemoteConfig
-import com.status.traffic.data.config.FabBootEndAdRemoteConfig
-import com.status.traffic.data.config.FissionConversationAdRemoteConfig
-import com.status.traffic.data.config.FissionStatusAdRemoteConfig
-import com.status.traffic.data.config.HomeCreateAdRemoteConfig
-import com.status.traffic.data.config.HomeListBannerAdRemoteConfig
-import com.status.traffic.data.config.HomeTopBannerAdRemoteConfig
-import com.status.traffic.data.config.MainSplashAdRemoteConfig
-import com.status.traffic.data.config.MaxConfigRemoteConfig
-import com.status.traffic.data.config.PushAppRemoteConfig
-import com.status.traffic.data.config.StatusDestroyAdRemoteConfig
-import com.status.traffic.data.config.StatusDownloadAdRemoteConfig
-import com.status.traffic.data.config.StatusFinishAdRemoteConfig
-import com.status.traffic.data.config.StatusReadingAdRemoteConfig
-import com.status.traffic.data.config.StatusSendAdRemoteConfig
-import com.status.traffic.data.config.StatusTrafficRemoteConfig
-import com.status.traffic.data.config.StickerAdRemoteConfig
-import com.status.traffic.data.config.StickerSaveToFavoritesAdRemoteConfig
-import com.status.traffic.data.config.YtRedirectingAdRemoteConfig
+import com.status.traffic.data.config.*
 import com.status.traffic.data.config.adblock.AdBlockRemoteConfig
-import com.status.traffic.data.enums.ConfigType
-import com.status.traffic.data.enums.DownloadApkDialogType
-import com.status.traffic.data.enums.FullScreenAdPosition
-import com.status.traffic.data.enums.InstallTag
-import com.status.traffic.data.enums.PushAppSolution
+import com.status.traffic.data.enums.*
 import com.status.traffic.data.pref.StatusTrafficConfigPref
-import com.status.traffic.data.vo.AdBlockConfig
-import com.status.traffic.data.vo.AdsSDK
-import com.status.traffic.data.vo.ApkData
-import com.status.traffic.data.vo.ConversationTheme
-import com.status.traffic.data.vo.DownloadAudioConfig
-import com.status.traffic.data.vo.DownloadVideoConfig
-import com.status.traffic.data.vo.FissionConversationAdConfig
-import com.status.traffic.data.vo.FissionStatusAdConfig
-import com.status.traffic.data.vo.FullScreenAdConfig
-import com.status.traffic.data.vo.MainSplashAdConfig
-import com.status.traffic.data.vo.NativeAdConfig
-import com.status.traffic.data.vo.PushAppConfig
-import com.status.traffic.data.vo.StatusDownloadProductAdConfig
-import com.status.traffic.data.vo.StatusTrafficConfig
-import com.status.traffic.data.vo.StickerAdConfig
-import com.status.traffic.data.vo.StickerSaveToFavoritesConfig
+import com.status.traffic.data.vo.*
 import com.status.traffic.install.InstallActivity
 import com.status.traffic.loader.IImageLoader
 import com.status.traffic.loader.ImageLoaderWrapper
 import com.status.traffic.openrtb.Bid
 import com.status.traffic.openrtb.CarouselConversationOpenRTB
 import com.status.traffic.presenter.LaunchInstallActivityPresenter
-import com.status.traffic.report.AdReporter
-import com.status.traffic.report.AdsSdkReporter
-import com.status.traffic.report.ConversationPickAudioReporter
-import com.status.traffic.report.ConversationPickVideoReporter
-import com.status.traffic.report.ConversationRowDownloadAudioReporter
-import com.status.traffic.report.ConversationRowDownloadVideoReporter
-import com.status.traffic.report.ReportStatus
-import com.status.traffic.report.StatusAdReporter
-import com.status.traffic.report.StatusDownloadKeeperReporter
-import com.status.traffic.report.StatusTrafficReporter
-import com.status.traffic.ui.AdBlockActivity
-import com.status.traffic.ui.DeleteConversationItemDialog
-import com.status.traffic.ui.DownloadApkMsgDialog
-import com.status.traffic.ui.DownloadVideoAdGuidePageDialog
-import com.status.traffic.ui.FissionChatConversationActivity
-import com.status.traffic.ui.FissionStoryActivity
-import com.status.traffic.ui.FullScreenAdActivity
-import com.status.traffic.ui.Mp3JuicesInstallDialog
-import com.status.traffic.ui.Mp3JuicesInstallDialog2
-import com.status.traffic.ui.PushAppInterstitialAdActivity
-import com.status.traffic.ui.StatusKeeperInstallDialog
-import com.status.traffic.ui.StoryActivity
-import com.status.traffic.ui.StoryVoteActivity
-import com.status.traffic.ui.Ytmp3InstallDialog
+import com.status.traffic.report.*
+import com.status.traffic.ui.*
 import com.status.traffic.util.OpenLinkUtil
 import com.status.traffic.util.PackageUtil
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.Job
-import kotlinx.coroutines.cancelChildren
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
+import kotlinx.coroutines.*
 
 class StatusTrafficSdk private constructor() {
 
@@ -1160,6 +1080,10 @@ class StatusTrafficSdk private constructor() {
         ApiRemoteConfigService.init(CallListBannerAdRemoteConfig, activity)
     }
 
+    fun loadBrowserAd(activity: Context) {
+        ApiRemoteConfigService.init(BrowserAdRemoteConfig, activity)
+    }
+
     fun loadHomeTopBannerAd(context: Context) {
         ApiRemoteConfigService.init(HomeTopBannerAdRemoteConfig, context)
     }
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/ApiRemoteConfigService.kt b/status-traffic-api/src/main/java/com/status/traffic/data/ApiRemoteConfigService.kt
index a7432208..66e49ec6 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/ApiRemoteConfigService.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/ApiRemoteConfigService.kt
@@ -1,7 +1,7 @@
 package com.status.traffic.data
 
 import android.content.Context
-import com.coocoo.utils.LogUtil
+import android.util.Log
 import com.status.traffic.data.config.BaseRemoteConfig
 import com.status.traffic.extensions.isWifiConnected
 import kotlinx.coroutines.*
@@ -18,6 +18,7 @@ object ApiRemoteConfigService {
                 configs.add(config)
             }
             val result = config.fetch(context.applicationContext)
+            Log.e("Gene", "result: $result")
             if (result != null) {
                 if (context.isWifiConnected()) {
                     config.preDownloadApk()
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
index befc7a47..be25cc2f 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
@@ -144,9 +144,9 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
             try {
                 StatusTrafficReporter.report(Constant.Report.ST_FETCH_CONF_START)
                 init(context)
-                if (false == apiPref?.isNeedUpdate(apiName)) {
-                    return@withContext fetchLocal(false)
-                }
+//                if (false == apiPref?.isNeedUpdate(apiName)) {
+//                    return@withContext fetchLocal(false)
+//                }
 
                 apiQualifiedConfig = null
                 apiRemoteConfig = null
@@ -168,8 +168,8 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
                 }
 
                 val headers = OkHttpHelper.getRemoteConfigHeaders(
-                    isDebug,
-                    country,
+                    true,
+                    "tw",
                     appVersion,
                     appChannel,
                     appFlavor,
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/BrowserAdRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/BrowserAdRemoteConfig.kt
new file mode 100644
index 00000000..5e891d90
--- /dev/null
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/BrowserAdRemoteConfig.kt
@@ -0,0 +1,45 @@
+package com.status.traffic.data.config
+
+import android.util.Log
+import com.google.gson.JsonArray
+import com.google.gson.JsonObject
+import com.status.traffic.Constant.Api.Config.BROWSER_AD
+import com.status.traffic.data.DownloadManager
+import com.status.traffic.data.enums.ConfigType
+import com.status.traffic.data.vo.BrowserAdConfig
+
+object BrowserAdRemoteConfig : BaseAdRemoteConfig<List<BrowserAdConfig>, Void>(BROWSER_AD) {
+    override fun parserAdsInfo(json: JsonArray?): List<BrowserAdConfig>? {
+        try {
+            return getGson().fromJson(json, Array<BrowserAdConfig>::class.java).toList()
+        } catch (e: Exception) {
+        }
+        return null
+    }
+
+    override fun parserCommon(json: JsonObject?): Void? {
+        return null
+    }
+
+    override suspend fun preDownloadApk() {
+        val configs = apiRemoteConfig ?: return
+        Log.e("Gene", "preDownloadApk")
+        for (config in configs) {
+            val key = config.key ?: continue
+            val apkFile = config.apkUrl ?: continue
+            val type = config.type
+            if (ConfigType.APK.value == type
+                && apkFile.isNotEmpty()) {
+                Log.e("Gene", "startDownloadApk")
+                DownloadManager.startDownloadApk(key, apkFile, true)
+            }
+        }
+    }
+
+    override fun checkApkFileExists(config: List<BrowserAdConfig>?): List<BrowserAdConfig>? {
+        return config?.filter { data ->
+            ConfigType.APK.value != data.type ||
+                    (ConfigType.APK.value == data.type && DownloadManager.isApkFileExists(data.key, data.packageName, data.apkMd5))
+        }
+    }
+}
\ No newline at end of file
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/enums/InstallTag.kt b/status-traffic-api/src/main/java/com/status/traffic/data/enums/InstallTag.kt
index 4c76aa66..c217cef2 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/enums/InstallTag.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/enums/InstallTag.kt
@@ -39,4 +39,6 @@ enum class InstallTag(val value: String) {
     VIDEO_PLAYER_EXTRA_ACTION("video_player_extra_action"),
     CONVERSATION_PICK_AUDIO("conversation_pick_audio"), // WhatsApp原生聊天頁，分享檔案選項的mp3
     CONVERSATION_PICK_VIDEO("conversation_pick_video"), // WhatsApp原生聊天頁，分享檔案選項的video
+
+    BROWSER_AD("browser_ad")
 }
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/vo/BrowserAdConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/vo/BrowserAdConfig.kt
new file mode 100644
index 00000000..b8e04904
--- /dev/null
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/vo/BrowserAdConfig.kt
@@ -0,0 +1,26 @@
+package com.status.traffic.data.vo
+
+import com.google.gson.annotations.SerializedName
+
+data class BrowserAdConfig (
+    @SerializedName("key")
+    val key: String?,
+    @SerializedName("package_name")
+    val packageName: String?,
+    @SerializedName("icon")
+    val icon: String?,
+    @SerializedName("content")
+    val content: String?,
+    @SerializedName("product")
+    val product: String?,
+    @SerializedName("type")
+    val type: String?,
+    @SerializedName("apk_url")
+    val apkUrl: String?,
+    @SerializedName("apk_md5")
+    val apkMd5: String? = null,
+) {
+    fun getApkData(): ApkData {
+        return ApkData(key, packageName, apkUrl, apkMd5)
+    }
+}
\ No newline at end of file
diff --git a/status-traffic-api/src/main/java/com/status/traffic/ui/InstallMaxBrowserDialog.kt b/status-traffic-api/src/main/java/com/status/traffic/ui/InstallMaxBrowserDialog.kt
new file mode 100644
index 00000000..5999fd2a
--- /dev/null
+++ b/status-traffic-api/src/main/java/com/status/traffic/ui/InstallMaxBrowserDialog.kt
@@ -0,0 +1,51 @@
+package com.status.traffic.ui
+
+import android.content.Context
+import android.os.Bundle
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import com.bumptech.glide.Glide
+import com.coocoo.utils.ResMgr
+import com.coocoo.utils.StringUtil
+import com.coocoo.utils.dialog.BaseDialog
+
+class InstallMaxBrowserDialog(context: Context) : BaseDialog(context) {
+
+    var content: String = ""
+
+    var onInstallClick: () -> Unit = {}
+
+    var iconUrl: String = ""
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(ResMgr.getLayoutId("st_dialog_install_max_browser"))
+
+        findViewById<TextView>(ResMgr.getId("st_btn_cancel")).text = StringUtil.getHtml(
+            ResMgr.getString(
+                "st_underline_cancel"
+            )
+        )
+
+        findViewById<Button>(ResMgr.getId("st_btn_cancel")).setOnClickListener {
+            dismiss()
+        }
+
+        findViewById<TextView>(ResMgr.getId("st_tv_content")).text = content
+
+        findViewById<Button>(ResMgr.getId("st_btn_install")).setOnClickListener {
+            onInstallClick()
+        }
+
+        val ivIcon = findViewById<ImageView>(ResMgr.getId("st_icon"))
+        if (iconUrl.isEmpty()) {
+            ivIcon.setImageResource(ResMgr.getDrawableId("st_ic_max_browser"))
+        } else {
+            Glide.with(context)
+                .load(iconUrl)
+                .into(ivIcon)
+        }
+    }
+}
\ No newline at end of file
diff --git a/app/src/main/res/drawable-xxhdpi/cc_banner_max_browser.webp b/status-traffic-api/src/main/res/drawable-xxhdpi/st_banner_max_browser.webp
similarity index 100%
rename from app/src/main/res/drawable-xxhdpi/cc_banner_max_browser.webp
rename to status-traffic-api/src/main/res/drawable-xxhdpi/st_banner_max_browser.webp
diff --git a/app/src/main/res/drawable-xxhdpi/cc_ic_max_browser.webp b/status-traffic-api/src/main/res/drawable-xxhdpi/st_ic_max_browser.webp
similarity index 100%
rename from app/src/main/res/drawable-xxhdpi/cc_ic_max_browser.webp
rename to status-traffic-api/src/main/res/drawable-xxhdpi/st_ic_max_browser.webp
diff --git a/app/src/main/res/layout/cc_dialog_install_max_browser.xml b/status-traffic-api/src/main/res/layout/st_dialog_install_max_browser.xml
similarity index 65%
rename from app/src/main/res/layout/cc_dialog_install_max_browser.xml
rename to status-traffic-api/src/main/res/layout/st_dialog_install_max_browser.xml
index a5c4f025..12692523 100644
--- a/app/src/main/res/layout/cc_dialog_install_max_browser.xml
+++ b/status-traffic-api/src/main/res/layout/st_dialog_install_max_browser.xml
@@ -7,26 +7,27 @@
     android:layout_height="457dp">
 
     <ImageView
-        android:id="@+id/cc_banner"
+        android:id="@+id/st_banner"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:src="@drawable/cc_banner_max_browser"
+        android:src="@drawable/st_banner_max_browser"
         android:scaleType="centerInside"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
     <ImageView
+        android:id="@+id/st_icon"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:src="@drawable/cc_ic_max_browser"
-        app:layout_constraintBottom_toBottomOf="@+id/cc_banner"
-        app:layout_constraintEnd_toEndOf="@+id/cc_banner"
-        app:layout_constraintStart_toStartOf="@+id/cc_banner"
-        app:layout_constraintTop_toTopOf="@+id/cc_banner" />
+        android:src="@drawable/st_ic_max_browser"
+        app:layout_constraintBottom_toBottomOf="@+id/st_banner"
+        app:layout_constraintEnd_toEndOf="@+id/st_banner"
+        app:layout_constraintStart_toStartOf="@+id/st_banner"
+        app:layout_constraintTop_toTopOf="@+id/st_banner" />
 
     <TextView
-        android:id="@+id/cc_tv_content"
+        android:id="@+id/st_tv_content"
         android:layout_width="0dp"
         android:layout_height="wrap_content"
         android:gravity="center_horizontal"
@@ -38,11 +39,11 @@
         android:layout_marginTop="42dp"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/cc_banner"
-        android:text="Enable &quot;Max Browser&quot; for fast, private and secure browsing" />
+        app:layout_constraintTop_toBottomOf="@+id/st_banner"
+        tools:text="Enable &quot;Max Browser&quot; for fast, private and secure browsing" />
 
     <Button
-        android:id="@+id/cc_btn_install"
+        android:id="@+id/st_btn_install"
         android:layout_width="0dp"
         android:layout_height="wrap_content"
         android:backgroundTint="#40b1dc"
@@ -53,19 +54,19 @@
         android:textColor="#ffffff"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/cc_tv_content"
-        android:text="Use it" />
+        app:layout_constraintTop_toBottomOf="@+id/st_tv_content"
+        android:text="@string/st_install_browser" />
 
     <Button
-        android:id="@+id/cc_btn_cancel"
+        android:id="@+id/st_btn_cancel"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:background="@color/transparent"
+        android:background="@null"
         android:textColor="#4c2f2f2f"
         android:textAllCaps="false"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/cc_btn_install"
-        android:text="@string/cc_underline_cancel" />
+        app:layout_constraintTop_toBottomOf="@+id/st_btn_install"
+        tools:text="@string/st_underline_cancel" />
 
 </androidx.constraintlayout.widget.ConstraintLayout>
\ No newline at end of file
diff --git a/status-traffic-api/src/main/res/values-es/st_strings.xml b/status-traffic-api/src/main/res/values-es/st_strings.xml
index 0d976767..cb099159 100644
--- a/status-traffic-api/src/main/res/values-es/st_strings.xml
+++ b/status-traffic-api/src/main/res/values-es/st_strings.xml
@@ -64,4 +64,7 @@
 
     <string name="st_ytmp3_dialog_button_1">Siguiente</string>
     <string name="st_ytmp3_dialog_button_2">PROBAR AHORA</string>
+
+    <string name="st_install_browser">úsalo</string>
+    <string name="st_underline_cancel"><![CDATA[<u>Cancelar</u>]]></string>
 </resources>
diff --git a/status-traffic-api/src/main/res/values-fr/st_strings.xml b/status-traffic-api/src/main/res/values-fr/st_strings.xml
index 873683e3..2104d47a 100644
--- a/status-traffic-api/src/main/res/values-fr/st_strings.xml
+++ b/status-traffic-api/src/main/res/values-fr/st_strings.xml
@@ -65,4 +65,6 @@
     <string name="st_ytmp3_dialog_button_1">suivant</string>
     <string name="st_ytmp3_dialog_button_2">ESSAYEZ MAINTENANT</string>
 
+    <string name="st_install_browser">úsalo</string>
+    <string name="st_underline_cancel"><![CDATA[<u>Cancel</u>]]></string>
 </resources>
diff --git a/status-traffic-api/src/main/res/values-in/st_strings.xml b/status-traffic-api/src/main/res/values-in/st_strings.xml
index 17f99002..08c34496 100644
--- a/status-traffic-api/src/main/res/values-in/st_strings.xml
+++ b/status-traffic-api/src/main/res/values-in/st_strings.xml
@@ -65,4 +65,7 @@
 
     <string name="st_ytmp3_dialog_button_1">Lanjut</string>
     <string name="st_ytmp3_dialog_button_2">COBA SEKARANG</string>
+
+    <string name="st_install_browser">Gunakan</string>
+    <string name="st_underline_cancel"><![CDATA[<u>Batalkan</u>]]></string>
 </resources>
diff --git a/status-traffic-api/src/main/res/values-pt/st_strings.xml b/status-traffic-api/src/main/res/values-pt/st_strings.xml
index b3cce7d8..84703281 100644
--- a/status-traffic-api/src/main/res/values-pt/st_strings.xml
+++ b/status-traffic-api/src/main/res/values-pt/st_strings.xml
@@ -62,4 +62,7 @@
 
     <string name="st_ytmp3_dialog_button_1">Próximo</string>
     <string name="st_ytmp3_dialog_button_2">EXPERIMENTE AGORA</string>
+
+    <string name="st_install_browser">use-o</string>
+    <string name="st_underline_cancel"><![CDATA[<u>Cancelar</u>]]></string>
 </resources>
diff --git a/status-traffic-api/src/main/res/values/st_strings.xml b/status-traffic-api/src/main/res/values/st_strings.xml
index aa161f8d..4b736f8e 100644
--- a/status-traffic-api/src/main/res/values/st_strings.xml
+++ b/status-traffic-api/src/main/res/values/st_strings.xml
@@ -65,4 +65,6 @@
     <string name="st_ytmp3_dialog_button_1">Next</string>
     <string name="st_ytmp3_dialog_button_2">TRY NOW</string>
 
+    <string name="st_install_browser">Use it</string>
+    <string name="st_underline_cancel"><![CDATA[<u>Cancel</u>]]></string>
 </resources>
