diff --git a/app/build.gradle b/app/build.gradle
index 522dd566f..64d651d1f 100644
--- a/app/build.gradle
+++ b/app/build.gradle
@@ -316,7 +316,7 @@ dependencies {
 
     implementation(name: 'sublive_modsdk', ext: 'aar') // v0.5.0
     implementation 'org.greenrobot:eventbus:3.2.0'
-    implementation 'com.google.android.gms:play-services-ads:18.0.0'
+    implementation 'com.google.android.gms:play-services-ads:20.3.0'
 }
 
 gradle.taskGraph.whenReady {
diff --git a/app/src/main/java/com/coocoo/CoocooRepoContainer.kt b/app/src/main/java/com/coocoo/CoocooRepoContainer.kt
index 93fe511fa..e4f3622dd 100644
--- a/app/src/main/java/com/coocoo/CoocooRepoContainer.kt
+++ b/app/src/main/java/com/coocoo/CoocooRepoContainer.kt
@@ -2,6 +2,8 @@ package com.coocoo
 
 import X.c1_2Oc
 import android.content.Context
+import com.coocoo.admob.AdMobLoader
+import com.coocoo.admob.AdMobRepository
 import com.coocoo.callerid.CallerIdMocker
 import com.coocoo.callerid.CallerIdRepository
 import com.coocoo.callerid.remote.ApiClient
@@ -125,4 +127,11 @@ class CoocooRepoContainer(private val appContext: Context) {
             SPDelegate.getInstance()
         )
     }
+
+    val adMobRepo: AdMobRepository by lazy {
+        AdMobRepository(
+            RemoteConfig.conversationBannerAd,
+            AdMobLoader(appContext)
+        )
+    }
 }
diff --git a/app/src/main/java/com/coocoo/admob/AdMobLoader.kt b/app/src/main/java/com/coocoo/admob/AdMobLoader.kt
new file mode 100644
index 000000000..b7721d98c
--- /dev/null
+++ b/app/src/main/java/com/coocoo/admob/AdMobLoader.kt
@@ -0,0 +1,22 @@
+package com.coocoo.admob
+
+import android.content.Context
+import androidx.lifecycle.CCMutableLiveData
+import com.coocoo.report.Report
+import com.google.android.gms.ads.AdListener
+import com.google.android.gms.ads.AdLoader
+import com.google.android.gms.ads.AdRequest
+import com.google.android.gms.ads.nativead.NativeAd
+
+class AdMobLoader(private val context: Context) {
+
+    fun load(adId: String, cachedNativeAd: CCMutableLiveData<NativeAd>, adListener: AdListener) {
+
+        val adLoader = AdLoader.Builder(context, adId).forNativeAd { nativeAd ->
+            cachedNativeAd.value?.destroy()
+            cachedNativeAd.value = nativeAd
+        }.withAdListener(adListener).build()
+
+        adLoader.loadAds(AdRequest.Builder().build(),5)
+    }
+}
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/admob/AdMobRepository.kt b/app/src/main/java/com/coocoo/admob/AdMobRepository.kt
new file mode 100644
index 000000000..85de574fe
--- /dev/null
+++ b/app/src/main/java/com/coocoo/admob/AdMobRepository.kt
@@ -0,0 +1,55 @@
+package com.coocoo.admob
+
+import androidx.lifecycle.CCLiveData
+import androidx.lifecycle.CCMutableLiveData
+import com.coocoo.report.Report
+import com.google.android.gms.ads.AdListener
+import com.google.android.gms.ads.LoadAdError
+import com.google.android.gms.ads.nativead.NativeAd
+import java.util.concurrent.TimeUnit
+
+class AdMobRepository(
+    private val adId: String,
+    private val adMobLoader: AdMobLoader
+) {
+
+    private var cachedTime: Long = 0
+
+    private val isTimeOut: Boolean get() {
+        return System.currentTimeMillis() - cachedTime > TIME_OUT
+    }
+
+    private val cachedNativeAd: CCMutableLiveData<NativeAd> = CCMutableLiveData()
+    val nativeAd: CCLiveData<NativeAd> get() = cachedNativeAd
+
+    fun reloadNativeAd() {
+        if (adId.isEmpty()) return
+
+        if (isTimeOut || cachedNativeAd.value == null) {
+            adMobLoader.load(adId, cachedNativeAd, object : AdListener() {
+                override fun onAdLoaded() {
+                    Report.reportConversationBannerAdReqOk(adId)
+                }
+
+                override fun onAdImpression() {
+                    cachedTime = System.currentTimeMillis()
+                    Report.reportConversationBannerAdShow(adId)
+                }
+
+                override fun onAdClicked() {
+                    reloadNativeAd()
+                }
+
+                override fun onAdFailedToLoad(loadAdError: LoadAdError) {
+                    reloadNativeAd()
+                    Report.reportConversationBannerAdReqFail(adId, loadAdError.code)
+                }
+            })
+            Report.reportConversationBannerAdReqStart(adId)
+        }
+    }
+
+    companion object {
+        private val TIME_OUT = TimeUnit.SECONDS.toMillis(10)
+    }
+}
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt b/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt
index d9ea2e382..83d7c1cd6 100644
--- a/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt
+++ b/app/src/main/java/com/coocoo/conversation/item/RetentionConversationItem.kt
@@ -5,21 +5,15 @@ import X.c1_3Ul
 import android.content.Context
 import android.content.Intent
 import android.graphics.Color
-import android.graphics.PorterDuff
-import android.graphics.PorterDuffColorFilter
 import android.graphics.drawable.Drawable
 import android.graphics.drawable.GradientDrawable
 import android.os.Handler
 import android.os.Looper
 import android.text.TextUtils
-import android.view.Gravity
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.FrameLayout
-import android.widget.ImageView
-import android.widget.RelativeLayout
-import android.widget.TextView
+import android.widget.*
 import com.bumptech.glide.Glide
 import com.bumptech.glide.load.engine.DiskCacheStrategy
 import com.bumptech.glide.load.model.GlideUrl
@@ -28,7 +22,7 @@ import com.bumptech.glide.load.resource.drawable.GlideDrawable
 import com.bumptech.glide.request.RequestListener
 import com.bumptech.glide.request.target.Target
 import com.coocoo.CooCooApp
-import com.coocoo.applets.AppletsHelper
+import com.coocoo.admob.AdMobRepository
 import com.coocoo.coocoosp.SPDelegate
 import com.coocoo.manager.RoutingManager
 import com.coocoo.newtheme.ThemeManager
@@ -37,7 +31,8 @@ import com.coocoo.transform.CircleTransform
 import com.coocoo.utils.*
 import com.coocoo.wallpapers.model.WallpaperBean
 import com.coocoo.whatsappdelegate.HomeActivityDelegate
-import com.google.android.gms.ads.AdView
+import com.google.android.gms.ads.nativead.NativeAd
+import com.google.android.gms.ads.nativead.NativeAdView
 import com.status.traffic.AdCommonManager
 import com.status.traffic.AdSceneManager
 import com.status.traffic.StatusTrafficSdk
@@ -54,7 +49,6 @@ import com.status.traffic.report.FissionConversationReporter
 import com.status.traffic.ui.DeleteConversationItemDialog
 import com.status.traffic.util.OpenLinkUtil
 import com.sublive.mod.applets.bean.AppletsPushMessage
-import com.sublive.mod.view.LarkMiniAppWebView
 import kotlinx.android.parcel.Parcelize
 import kotlinx.coroutines.*
 import java.lang.ref.SoftReference
@@ -769,45 +763,56 @@ class AppletsConversationItem(val title : String, val des: String, val pushMessa
     }
 }
 
-class ConversationBannerItem(
-    private val adView: AdView?) : RetentionConversationItem() {
+class ConversationBannerItem : RetentionConversationItem() {
 
-    var isAdLoadSuccess = false
+    var adView: NativeAdView? = null
+        private set
 
-    fun resume() {
-        adView?.resume()
-    }
+    var nativeAd: NativeAd? = null
 
     fun pause() {
-        adView?.pause()
     }
 
     fun destroy() {
         adView?.destroy()
     }
 
-    override fun getView(parent: ViewGroup): View {
-        if (adView?.parent != null) {
-            return adView.parent as View
-        }
+    fun refresh(nativeAd: NativeAd, adapter: BaseAdapter?) {
+        this.nativeAd = nativeAd
+        adapter?.notifyDataSetChanged()
+    }
 
-        val frameLayout = FrameLayout(CooCooApp.getAppContext())
-        val params: ViewGroup.LayoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
-        frameLayout.layoutParams = params
-        if (!isAdLoadSuccess) {
-            return frameLayout
+    override fun getView(parent: ViewGroup): View {
+        val ad = nativeAd
+        return if (ad == null) {
+            FrameLayout(parent.context)
+        } else {
+            LayoutInflater
+                .from(parent.context)
+                .inflate(ResMgr.getLayoutId("cc_ad_banner"), parent, false).also { view ->
+                    populateNativeAdView(ad, view as NativeAdView)
+                }
         }
+    }
 
-        if (adView != null) {
-            val childParams: FrameLayout.LayoutParams = FrameLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT)
-            childParams.width = adView.adSize.getWidthInPixels(CooCooApp.getAppContext())
-            childParams.height = adView.adSize.getHeightInPixels(CooCooApp.getAppContext())
-            childParams.gravity = Gravity.CENTER
-            frameLayout.addView(adView, childParams)
+    private fun populateNativeAdView(nativeAd: NativeAd, nativeAdView: NativeAdView) {
+//        nativeAdView.setHeadlineView(nativeAdView.findViewById(ResMgr.getId("cc_tv_title")));
+//        nativeAdView.setBodyView(nativeAdView.findViewById(ResMgr.getId("cc_tv_subtitle")));
+//        nativeAdView.setIconView(nativeAdView.findViewById(ResMgr.getId("cc_iv_ad")));
+//        nativeAdView.setNativeAd(nativeAd);
+        nativeAdView.callToActionView = nativeAdView
+        val textView = nativeAdView.findViewById<TextView>(ResMgr.getId("cc_tv_title"))
+        textView.text = nativeAd.headline
+        nativeAd.icon?.let { icon ->
+            val iconView = nativeAdView.findViewById<ImageView>(ResMgr.getId("cc_iv_ad"))
+            Glide.with(nativeAdView.context)
+                .load(icon.uri)
+                .transform(CircleTransform(nativeAdView.context))
+                .into(iconView)
         }
-
-
-        return frameLayout
+        val body = nativeAdView.findViewById<TextView>(ResMgr.getId("cc_tv_subtitle"))
+        body.text = nativeAd.body
+        nativeAdView.setNativeAd(nativeAd)
     }
 }
 
diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index a818895da..5c8c34e67 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -62,7 +62,6 @@ import com.coocoo.firebase.properties.UserProperties;
 import com.coocoo.firebase.remoteConfig.RemoteConfig;
 import com.coocoo.fm.yo;
 import com.coocoo.manager.AdvertisingIdManager;
-import com.coocoo.manager.ContactTimeLineManager;
 import com.coocoo.manager.ConversationsItemTypeManager;
 import com.coocoo.manager.FrostWireManager;
 import com.coocoo.manager.PrivacyPreferenceManager;
@@ -116,9 +115,9 @@ import javax.crypto.SecretKey;
 
 import X.c1_01f;
 import X.c1_2Oc;
-import X.c1_3Ul;
 import X.c1_2P6;
 import X.c1_2ua;
+import X.c1_3Ul;
 import X.cc_3ck;
 
 public class Coocoo {
@@ -278,6 +277,7 @@ public class Coocoo {
         LarkSdk.INSTANCE.init(appContext, BuildConfig.IS_DEBUG);
         LarkSdk.INSTANCE.setAppletsInterface(new LarkSdkInterfaceImpl());
         AppletsHelper.init();
+        Coocoo.getRepoContainer().getAdMobRepo().reloadNativeAd();
     }
 
     public static void initStatusTrafficAdCommon() {
diff --git a/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfigKeys.kt b/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfigKeys.kt
index affcc597e..d80114505 100644
--- a/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfigKeys.kt
+++ b/app/src/main/java/com/coocoo/firebase/remoteConfig/RemoteConfigKeys.kt
@@ -124,7 +124,7 @@ object RemoteConfigKeys {
     const val THEME_STORE_BANNER = "modad_theme_store_banner"
 
     // Conversation banner ad
-    const val CONVERSATION_BANNER = "modad_homelist_banner"
+    const val CONVERSATION_BANNER = "modad_homelist_banner_native"
     const val APPLETS_PUSH_LAST_CODE = "applets_push_switch_last_code"
     const val APPLETS_BLOCK_PUSH_MESSAGE = "applets_block_push_message"
     const val APPLETS_PUSH_MESSAGE_VERSION = "applets_push_message_version"
diff --git a/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java b/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
index 1e1a9d2b8..c63c9b619 100644
--- a/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
+++ b/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
@@ -11,6 +11,8 @@ import android.widget.Button;
 import android.widget.FrameLayout;
 import android.widget.ImageView;
 
+import androidx.annotation.NonNull;
+
 import com.bumptech.glide.Glide;
 import com.coocoo.android.support.v4.app.Fragment;
 import com.coocoo.android.support.v4.app.FragmentManager;
@@ -32,6 +34,7 @@ import com.google.android.gms.ads.AdListener;
 import com.google.android.gms.ads.AdRequest;
 import com.google.android.gms.ads.AdSize;
 import com.google.android.gms.ads.AdView;
+import com.google.android.gms.ads.LoadAdError;
 import com.status.traffic.ThemeFabAdManager;
 
 import java.io.File;
@@ -125,8 +128,8 @@ public class ThemeStoreActivity extends CCBaseActivity implements View.OnClickLi
             }
 
             @Override
-            public void onAdFailedToLoad(int code) {
-                Report.reportThemeStoreBannerAdReqFail(BANNER_PLACEMENT, bannerId, code);
+            public void onAdFailedToLoad(@NonNull LoadAdError loadAdError) {
+                Report.reportThemeStoreBannerAdReqFail(BANNER_PLACEMENT, bannerId, loadAdError.getCode());
             }
         });
         adView.setAdUnitId(bannerId);
diff --git a/app/src/main/java/com/coocoo/report/Report.java b/app/src/main/java/com/coocoo/report/Report.java
index 5fbd7f0a7..07a8d1dfc 100644
--- a/app/src/main/java/com/coocoo/report/Report.java
+++ b/app/src/main/java/com/coocoo/report/Report.java
@@ -1016,19 +1016,19 @@ public class Report {
     }
 
     public static void reportConversationBannerAdReqStart(String adId) {
-        ReportCore.reportThemeStoreBannerAdReqStart("modad_homelist_banner", adId);
+        ReportCore.reportThemeStoreBannerAdReqStart("modad_homelist_banner_native", adId);
     }
 
     public static void reportConversationBannerAdShow(String adId) {
-        ReportCore.reportThemeStoreBannerAdShow("modad_homelist_banner", adId);
+        ReportCore.reportThemeStoreBannerAdShow("modad_homelist_banner_native", adId);
     }
 
     public static void reportConversationBannerAdReqOk(String adId) {
-        ReportCore.reportBannerAdReqOk("modad_homelist_banner", adId);
+        ReportCore.reportBannerAdReqOk("modad_homelist_banner_native", adId);
     }
 
     public static void reportConversationBannerAdReqFail(String adId, int code) {
-        ReportCore.reportBannerAdReqFail("modad_homelist_banner", adId, code);
+        ReportCore.reportBannerAdReqFail("modad_homelist_banner_native", adId, code);
     }
 
     public static void miniAppMonitor(int type, int status) {
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
index 79a9459a3..1b6be8747 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
@@ -12,7 +12,6 @@ import android.graphics.Point;
 import android.net.Uri;
 import android.os.Bundle;
 import android.os.Handler;
-import android.os.Looper;
 import android.text.TextUtils;
 import android.view.LayoutInflater;
 import android.view.View;
@@ -78,10 +77,6 @@ import com.coocoo.widget.AutoScrollViewPager;
 import com.coocoo.widget.indicator.ViewPagerIndicator;
 import com.coocoowhatsapp.conversationslist.ConversationsFragment;
 import com.coocoowhatsapp.conversationslist.ViewHolder;
-import com.google.android.gms.ads.AdListener;
-import com.google.android.gms.ads.AdRequest;
-import com.google.android.gms.ads.AdSize;
-import com.google.android.gms.ads.AdView;
 import com.status.traffic.StatusTrafficSdk;
 import com.status.traffic.data.enums.DownloadApkDialogType;
 import com.status.traffic.data.enums.InstallTag;
@@ -89,7 +84,6 @@ import com.status.traffic.data.vo.ApkData;
 import com.status.traffic.presenter.ILaunchInstallActivityPresenter;
 import com.status.traffic.presenter.LaunchInstallActivityPresenter;
 import com.status.traffic.report.AdReporter;
-import com.status.traffic.report.AdsFillReporter;
 import com.sublive.mod.LarkSdk;
 import com.sublive.mod.applets.Applets;
 import com.sublive.mod.applets.bean.AppletsPushMessage;
@@ -97,7 +91,6 @@ import com.whatsapp.jid.UserJid;
 
 import org.jetbrains.annotations.NotNull;
 
-import java.io.File;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
@@ -304,46 +297,12 @@ public class ConversationsFragmentDelegate extends FragmentDelegate<Conversation
     }
 
     private void initBannerAd() {
-
-        final String bannerId = RemoteConfig.getConversationBannerAd();
-        if (TextUtils.isEmpty(bannerId)) {
-            return;
-        }
-
-        AdView adView = new AdView(Coocoo.getAppContext());
-        adView.setAdListener(new AdListener() {
-            @Override
-            public void onAdLoaded() {
-                Report.reportConversationBannerAdReqOk(bannerId);
-                new Handler(Looper.getMainLooper()).post(new Runnable() {
-                    @Override
-                    public void run() {
-                        if (conversationBannerItem != null) {
-                            conversationBannerItem.setAdLoadSuccess(true);
-                        }
-                        BaseAdapter adapter = ((ConversationsFragment) mHostFragment).A0r;
-                        if (adapter != null) {
-                            adapter.notifyDataSetChanged();
-                        }
-                    }
-                });
-            }
-
-            @Override
-            public void onAdImpression() {
-                Report.reportConversationBannerAdShow(bannerId);
-            }
-
-            @Override
-            public void onAdFailedToLoad(int code) {
-                Report.reportConversationBannerAdReqFail(bannerId, code);
-            }
+        conversationBannerItem = new ConversationBannerItem();
+        Coocoo.getRepoContainer().getAdMobRepo().getNativeAd().observeForever((nativeAd) -> {
+            LogUtil.d("mod_ads_behavior", nativeAd.getHeadline());
+            BaseAdapter adapter = ((ConversationsFragment) mHostFragment).A0r;
+            conversationBannerItem.refresh(nativeAd, adapter);
         });
-        adView.setAdUnitId(bannerId);
-        adView.setAdSize(AdSize.BANNER);
-        adView.loadAd(new AdRequest.Builder().build());
-        conversationBannerItem = new ConversationBannerItem(adView);
-        Report.reportConversationBannerAdReqStart(bannerId);
     }
 
     public void create(Bundle savedInstanceState) {
@@ -372,9 +331,9 @@ public class ConversationsFragmentDelegate extends FragmentDelegate<Conversation
         if (isCurrentHomePage && isNewcomerPromoteFeatureVisible()) {
             newcomerPromoteFeaturePager.startAutoScroll();
         }
-        if (conversationBannerItem != null) {
-            conversationBannerItem.resume();
-        }
+//        if (conversationBannerItem != null) {
+//            conversationBannerItem.refresh();
+//        }
     }
 
     @Override
diff --git a/app/src/main/res/layout/cc_ad_banner.xml b/app/src/main/res/layout/cc_ad_banner.xml
new file mode 100644
index 000000000..afd4bc67e
--- /dev/null
+++ b/app/src/main/res/layout/cc_ad_banner.xml
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.google.android.gms.ads.nativead.NativeAdView xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="76dp"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    tools:visibility="visible">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <com.coocoowhatsapp.WaImageView
+            android:id="@+id/cc_iv_ad"
+            android:layout_width="48dp"
+            android:layout_height="48dp"
+            android:layout_marginStart="16dp"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            tools:src="@android:drawable/stat_sys_warning"/>
+
+        <com.coocoowhatsapp.WaTextView
+            android:id="@+id/cc_tv_title"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            style="@style/ListItemTitle"
+            app:layout_constraintTop_toTopOf="@+id/cc_iv_ad"
+            app:layout_constraintStart_toEndOf="@+id/cc_iv_ad"
+            tools:text="Adventure Duck"/>
+
+        <com.coocoowhatsapp.WaImageView
+            android:id="@+id/cc_tv_label"
+            android:layout_height="wrap_content"
+            android:layout_width="wrap_content"
+            android:src="@drawable/cc_ic_ad_tag_chatlist"
+            android:layout_marginStart="6dp"
+            app:layout_constraintTop_toTopOf="@+id/cc_tv_title"
+            app:layout_constraintBottom_toBottomOf="@+id/cc_tv_title"
+            app:layout_constraintStart_toEndOf="@+id/cc_tv_title"/>
+
+        <com.coocoowhatsapp.WaTextView
+            android:id="@+id/cc_tv_subtitle"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginEnd="13dp"
+            android:ellipsize="end"
+            style="?listItemSubtitle"
+            app:layout_constraintStart_toStartOf="@+id/cc_tv_title"
+            app:layout_constraintEnd_toStartOf="@+id/btn_download"
+            app:layout_constraintBottom_toBottomOf="@+id/cc_iv_ad"
+            tools:text="Slash &amp; swim your way down…"/>
+
+        <com.coocoowhatsapp.WaTextView
+            android:id="@+id/btn_download"
+            android:layout_width="85dp"
+            android:layout_height="36dp"
+            android:textSize="13sp"
+            android:textColor="#ffffff"
+            android:layout_marginEnd="4dp"
+            android:background="@drawable/cc_btn_green_gradient"
+            android:gravity="center"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            android:text="@string/cc_carousel_conversation_download"/>
+
+        <com.coocoowhatsapp.WaImageView
+            android:layout_width="12dp"
+            android:layout_height="12dp"
+            android:src="@drawable/cc_ic_info_tag"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"/>
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+</com.google.android.gms.ads.nativead.NativeAdView>
\ No newline at end of file
