diff --git a/app/src/main/java/X/c1_017.java b/app/src/main/java/X/c1_017.java
index dc95e1c32..f987a3470 100644
--- a/app/src/main/java/X/c1_017.java
+++ b/app/src/main/java/X/c1_017.java
@@ -6,6 +6,8 @@ import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 
+import androidx.annotation.Nullable;
+
 /**
  * Fragment.
  */
@@ -72,6 +74,7 @@ public class c1_017 {
     }
 
     // getActivity() LX/0LX
+    @Nullable
     public cc_01K A09() {
         // do nothing
         return null;
diff --git a/app/src/main/java/com/coocoo/report/Report.java b/app/src/main/java/com/coocoo/report/Report.java
index 03f3503b7..ccc9b74dc 100644
--- a/app/src/main/java/com/coocoo/report/Report.java
+++ b/app/src/main/java/com/coocoo/report/Report.java
@@ -1108,4 +1108,8 @@ public class Report {
                                                           int feedVideoCount, int exitIndex) {
         ReportCore.reportStatusFeedPlaybackActionBack(feedImgCount, feedVideoCount, exitIndex);
     }
+
+    public static void reportStatusFeedShareDone(String statusId) {
+        ReportCore.reportStatusFeedShareDone(statusId);
+    }
 }
diff --git a/app/src/main/java/com/coocoo/report/ReportConstant.java b/app/src/main/java/com/coocoo/report/ReportConstant.java
index b37efe6df..37afe5971 100644
--- a/app/src/main/java/com/coocoo/report/ReportConstant.java
+++ b/app/src/main/java/com/coocoo/report/ReportConstant.java
@@ -946,6 +946,8 @@ public interface ReportConstant {
     String KEY_FEED_IMAGE = "feed_img";
     String KEY_FEED_VIDEO = "feed_vid";
     String KEY_EXIT_INDEX = "exit_index";
+    String KEY_STATUS_ID = "status_id";
     String ACTION_ENTRY_SHOW = "entry_show";
     String ACTION_BACK = "back";
+    String ACTION_SHARE_DONE = "share_done";
 }
diff --git a/app/src/main/java/com/coocoo/report/ReportCore.java b/app/src/main/java/com/coocoo/report/ReportCore.java
index 7cc2f8367..48cb3012a 100644
--- a/app/src/main/java/com/coocoo/report/ReportCore.java
+++ b/app/src/main/java/com/coocoo/report/ReportCore.java
@@ -1762,4 +1762,11 @@ class ReportCore implements ReportConstant {
         params.put(ReportConstant.KEY_EXIT_INDEX, String.valueOf(exitIndex));
         report(ReportConstant.EVENT_STATUS_FEED, params);
     }
+
+    public static void reportStatusFeedShareDone(String statusId) {
+        Map<String, String> params = new HashMap<>();
+        params.put(ReportConstant.KEY_ACTION, ACTION_SHARE_DONE);
+        params.put(ReportConstant.KEY_STATUS_ID, statusId);
+        report(ReportConstant.EVENT_STATUS_FEED, params);
+    }
 }
diff --git a/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt b/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt
index 24fc257db..32f6cdfa3 100644
--- a/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt
+++ b/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt
@@ -37,6 +37,7 @@ import com.coocoowhatsapp.Conversation
 
 class StatusHeaderViewHolder(
     view: View,
+    private val statusId: String,
     private val fragment: CCBaseFragment,
     private val onSendFile: (to: List<c1_2Ob>, filePath: String) -> Boolean
 ) {
@@ -114,6 +115,7 @@ class StatusHeaderViewHolder(
             A0R = arrayListOf(1)
             A00()
         }
+        intent.putExtra(BUNDLE_KEY_STATUS_ID, statusId)
         fragment.startActivityForResult(intent, REQUEST_CODE)
     }
 
@@ -131,5 +133,6 @@ class StatusHeaderViewHolder(
 
     companion object {
         private const val REQUEST_CODE = 0x2
+        const val BUNDLE_KEY_STATUS_ID = "bundle_key_status_id"
     }
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusPlaybackFragment.kt b/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusPlaybackFragment.kt
index 3f77532de..08bf81aaa 100644
--- a/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusPlaybackFragment.kt
+++ b/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusPlaybackFragment.kt
@@ -9,7 +9,6 @@ import android.content.Intent
 import android.os.Bundle
 import android.view.View
 import android.widget.ImageView
-import android.widget.Toast
 import androidx.lifecycle.CCObserver
 import androidx.lifecycle.CCViewModelProvider
 import com.coocoo.base.CCBaseActivity
@@ -49,6 +48,8 @@ abstract class StatusPlaybackFragment : CCBaseFragment() {
     protected lateinit var ivLike: ImageView
     protected lateinit var headerViewHolder: StatusHeaderViewHolder
 
+    private lateinit var statusId: String
+
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
         activityCreated = true
@@ -57,7 +58,8 @@ abstract class StatusPlaybackFragment : CCBaseFragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        headerViewHolder = StatusHeaderViewHolder(view, this, ::sendFile)
+        statusId = arguments?.getString(EXTRA_STATUS_ID, "") ?: ""
+        headerViewHolder = StatusHeaderViewHolder(view, statusId, this, ::sendFile)
         ivDownload = ResMgr.findViewById<ImageView>(Constants.Res.Id.STATUS_FEED_PLAYBACK_DOWNLOAD, view).apply {
             setOnClickListener {
                 // TODO
@@ -88,7 +90,6 @@ abstract class StatusPlaybackFragment : CCBaseFragment() {
         motionViewModel = CCViewModelProvider(activity.viewModelStore, StatusPlaybackMotionViewModelFactory())[StatusPlaybackMotionViewModel::class.java]
 
         val statusFeedRepo = Coocoo.getRepoContainer().statusFeedRepo
-        val statusId: String = arguments?.getString(EXTRA_STATUS_ID, "") ?: ""
         val isFavorite: Boolean = arguments?.getBoolean(EXTRA_IS_FAVORITE) ?: false
         viewModel = if (isFavorite) {
             CCViewModelProvider(viewModelStore, FavoriteStatusPlaybackPageViewModelFactory(statusFeedRepo, statusId))[FavoriteStatusPlaybackPageViewModel::class.java]
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
new file mode 100644
index 000000000..a9bed3c87
--- /dev/null
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
@@ -0,0 +1,18 @@
+package com.coocoo.whatsappdelegate
+
+import com.coocoo.annotation.SmaliInject
+import com.coocoo.report.Report.reportStatusFeedShareDone
+import com.coocoo.statuses.feed.fragment.StatusHeaderViewHolder
+import com.coocoo.utils.LogUtil
+import com.coocoowhatsapp.contact.picker.ContactPickerFragment
+
+object ContactPickerSendBtnDelegate {
+    @SmaliInject
+    @JvmStatic
+    fun onSendBtnClick(fragment: ContactPickerFragment) {
+        val statusId = fragment.A09()?.intent?.getStringExtra(StatusHeaderViewHolder.BUNDLE_KEY_STATUS_ID)
+        if (statusId.isNullOrEmpty().not()) {
+            reportStatusFeedShareDone(statusId)
+        }
+    }
+}
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoowhatsapp/contact/picker/ContactPickerFragment.java b/app/src/main/java/com/coocoowhatsapp/contact/picker/ContactPickerFragment.java
index 95d2d775a..9682c2023 100644
--- a/app/src/main/java/com/coocoowhatsapp/contact/picker/ContactPickerFragment.java
+++ b/app/src/main/java/com/coocoowhatsapp/contact/picker/ContactPickerFragment.java
@@ -4,9 +4,10 @@ import android.os.Bundle;
 
 import java.util.Map;
 
+import X.c1_017;
 import X.c1_0HQ;
 
-public class ContactPickerFragment {
+public class ContactPickerFragment extends c1_017 {
 
     /**
      * 当前选择的用户数量
diff --git a/whatsapp_modify_module/smali/X/1Fe.1.smali b/whatsapp_modify_module/smali/X/1Fe.1.smali
index eea1a1953..37372812c 100644
--- a/whatsapp_modify_module/smali/X/1Fe.1.smali
+++ b/whatsapp_modify_module/smali/X/1Fe.1.smali
@@ -26,6 +26,9 @@
 
     .line 220959
     iget-object v4, p0, LX/1Fe;->A00:Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;
+#coocoo
+    invoke-static {v4}, Lcom/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate;->onSendBtnClick(Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;)V
+#coocoo
 
 #coocoo
     invoke-virtual {v4}, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->A0A()LX/01K;
