diff --git a/app/src/main/java/com/coocoo/report/Report.java b/app/src/main/java/com/coocoo/report/Report.java
index 69f3672db..38e310271 100644
--- a/app/src/main/java/com/coocoo/report/Report.java
+++ b/app/src/main/java/com/coocoo/report/Report.java
@@ -1159,4 +1159,28 @@ public class Report {
     public static void reportMyMessageForwardDone() {
         ReportCore.reportMyMessageForwardDone();
     }
+
+    public static void reportSingleSelectHomeConversationAction(String action) {
+        ReportCore.reportConversationList(action, ReportConstant.VALUE_SINGLE ,ReportConstant.VALUE_HOME);
+    }
+
+    public static void reportMultiSelectHomeConversationAction(String action) {
+        ReportCore.reportConversationList(action, ReportConstant.VALUE_MUTIPLE ,ReportConstant.VALUE_HOME);
+    }
+
+    public static void reportSingleSelectArchiveConversationAction(String action) {
+        ReportCore.reportConversationList(action, ReportConstant.VALUE_SINGLE ,ReportConstant.VALUE_ARCHIVE);
+    }
+
+    public static void reportMultiSelectArchiveConversationAction(String action) {
+        ReportCore.reportConversationList(action, ReportConstant.VALUE_MUTIPLE ,ReportConstant.VALUE_ARCHIVE);
+    }
+
+    public static void reportSingleSelectHideConversationAction(String action) {
+        ReportCore.reportConversationList(action, ReportConstant.VALUE_SINGLE ,ReportConstant.VALUE_HIDE);
+    }
+
+    public static void reportMultiSelectHideConversationAction(String action) {
+        ReportCore.reportConversationList(action, ReportConstant.VALUE_MUTIPLE ,ReportConstant.VALUE_HIDE);
+    }
 }
diff --git a/app/src/main/java/com/coocoo/report/ReportConstant.java b/app/src/main/java/com/coocoo/report/ReportConstant.java
index d0c833501..16825d371 100644
--- a/app/src/main/java/com/coocoo/report/ReportConstant.java
+++ b/app/src/main/java/com/coocoo/report/ReportConstant.java
@@ -958,4 +958,24 @@ public interface ReportConstant {
     String ACTION_SHOW_DIALOG = "show_dialog";
     String ACTION_FORWARD_DONE = "forward_done";
     String ACTION_OK = "ok";
+
+    String EVENT_CONVERSATION_LIST = "con_list_select";
+    String KEY_CON_NUM = "con_num";
+    String KEY_SCENE = "scene";
+    String VALUE_SINGLE = "1";
+    String VALUE_MUTIPLE = "2";
+    String VALUE_UNPIN = "unpin";
+    String VALUE_DELETE = "delete";
+    String VALUE_MUTE = "mute";
+    String VALUE_UNMUTE = "unmute";
+    String VALUE_ARCHIVE = "archive";
+    String VALUE_UNARCHIV = "unarchive";
+    String VALUE_ADD_SHORT_CUT = "add_shortcut";
+    String VALUE_ADD_TO_CONTACTS = "add_to_contacts";
+    String VALUE_READ = "read";
+    String VALUE_UNREAD = "unread";
+    String VALUE_INVITE = "invite";
+    String VALUE_HIDE = "hide";
+    String VALUE_UNHIDE = "unhide";
+    String VALUE_HOME = "home";
 }
diff --git a/app/src/main/java/com/coocoo/report/ReportCore.java b/app/src/main/java/com/coocoo/report/ReportCore.java
index 4e94a3d7b..62c3e8e6b 100644
--- a/app/src/main/java/com/coocoo/report/ReportCore.java
+++ b/app/src/main/java/com/coocoo/report/ReportCore.java
@@ -1874,4 +1874,12 @@ class ReportCore implements ReportConstant {
         params.put(ReportConstant.KEY_ACTION, ReportConstant.ACTION_FORWARD_DONE);
         report(ReportConstant.EVENT_MY_MESSAGE, params);
     }
+
+    public static void reportConversationList(String action, String num, String scene) {
+        Map<String, String> params = new HashMap<>();
+        params.put(KEY_ACTION, action);
+        params.put(KEY_NUM, num);
+        params.put(KEY_SCENE, scene);
+        report(ReportConstant.EVENT_CONVERSATION_LIST, params);
+    }
 }
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
index 4d7c65397..4ea646351 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
@@ -14,6 +14,7 @@ import android.os.Bundle;
 import android.os.Handler;
 import android.text.TextUtils;
 import android.view.LayoutInflater;
+import android.view.MenuItem;
 import android.view.View;
 import android.widget.BaseAdapter;
 import android.widget.FrameLayout;
@@ -978,4 +979,10 @@ public class ConversationsFragmentDelegate extends FragmentDelegate<Conversation
         }
         notifyConversationsAdapter();
     }
+
+    @SmaliInject
+    public void onMenuItemClick(MenuItem item) {
+        ConversationsFragment fragment = (ConversationsFragment) mHostFragment;
+        LogUtil.d("item.getId:" + item.getItemId() + " A21:" + fragment.A21.size() + " A20:" + fragment.A20.size());
+    }
 }
\ No newline at end of file
diff --git a/whatsapp_modify_module/smali/com/coocoowhatsapp/conversationslist/ConversationsFragment.smali b/whatsapp_modify_module/smali/com/coocoowhatsapp/conversationslist/ConversationsFragment.smali
index d545ee193..f60fabb4b 100644
--- a/whatsapp_modify_module/smali/com/coocoowhatsapp/conversationslist/ConversationsFragment.smali
+++ b/whatsapp_modify_module/smali/com/coocoowhatsapp/conversationslist/ConversationsFragment.smali
@@ -1338,7 +1338,10 @@
 
 .method public A0k(Landroid/view/MenuItem;)Z
     .locals 2
-
+#coocoo
+    iget-object v1, p0, Lcom/coocoowhatsapp/conversationslist/ConversationsFragment;->conversationsFragmentDelegate:Lcom/coocoo/whatsappdelegate/ConversationsFragmentDelegate;
+    invoke-virtual {v1, p1}, Lcom/coocoo/whatsappdelegate/ConversationsFragmentDelegate;->onMenuItemClick(Landroid/view/MenuItem;)V
+#coocoo
     .line 487478
     iget-object v1, p0, Lcom/coocoowhatsapp/conversationslist/ConversationsFragment;->A0u:LX/35c;
 
@@ -1363,6 +1366,11 @@
 .method public A0l(Landroid/view/MenuItem;)Z
     .locals 8
 
+#coocoo
+    iget-object v0, p0, Lcom/coocoowhatsapp/conversationslist/ConversationsFragment;->conversationsFragmentDelegate:Lcom/coocoo/whatsappdelegate/ConversationsFragmentDelegate;
+    invoke-virtual {v0, p1}, Lcom/coocoo/whatsappdelegate/ConversationsFragmentDelegate;->onMenuItemClick(Landroid/view/MenuItem;)V
+#coocoo
+
     .line 487479
     invoke-interface {p1}, Landroid/view/MenuItem;->getItemId()I
 
