diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt
index d4d7176..7a842b5 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/HomeActivityDelegate.kt
@@ -15,6 +15,7 @@ import com.bumptech.glide.request.target.Target
 import com.coocoo.CooCoo
 import com.coocoo.CooCoo.repoContainer
 import com.coocoo.annotation.SmaliInject
+import com.coocoo.business.BusinessController.allowEnabledFullAds
 import com.coocoo.coocoosp.SPDelegate
 import com.coocoo.manager.PerformanceAnalyticsManager.onHomeResume
 import com.coocoo.presenter.HomeActivityPresenter
@@ -67,8 +68,10 @@ class HomeActivityDelegate @SmaliInject constructor(homeActivity: HomeActivity?)
             if (adBlockConfig != null && !isMembershipEffective) {
                 try {
                     AdSceneManager.getInstance().displayAd(AdSceneManager.AdScene.AD_BLOCK)
+                    reportEntrance(AdBlockReporter.EntranceType.FAB, AdBlockReporter.EntranceAction.SHOW)
                     fabAdView.setImageResource(ResMgr.getDrawableId(Constants.Res.Drawable.IC_FAB_AD_BLOCK))
                     fabAdView.setTag(fabAdView.id, adBlockConfig)
+                    fabAdView.visibility = View.VISIBLE
                 } catch (e: Exception) {
                     e.printStackTrace()
                 }
@@ -77,34 +80,38 @@ class HomeActivityDelegate @SmaliInject constructor(homeActivity: HomeActivity?)
             val fabAdConfig = StatusTrafficSdk.getInstance().getConversationFabAdRemoteConfig()
             LogUtil.d("fabAdConfig: $fabAdConfig")
             if (fabAdConfig != null) {
-                Glide.with(fabAdView.context)
-                    .load(fabAdConfig.avatar)
-                    .diskCacheStrategy(DiskCacheStrategy.ALL)
-                    .transform(CircleTransform(fabAdView.context))
-                    .listener(object : RequestListener<String?, GlideDrawable?> {
-                        override fun onException(
-                            e: Exception?,
-                            model: String?,
-                            target: Target<GlideDrawable?>?,
-                            isFirstResource: Boolean
-                        ): Boolean {
-                            return false
-                        }
+                try {
+                    Glide.with(fabAdView.context)
+                        .load(fabAdConfig.avatar)
+                        .diskCacheStrategy(DiskCacheStrategy.ALL)
+                        .transform(CircleTransform(fabAdView.context))
+                        .listener(object : RequestListener<String?, GlideDrawable?> {
+                            override fun onException(
+                                e: Exception?,
+                                model: String?,
+                                target: Target<GlideDrawable?>?,
+                                isFirstResource: Boolean
+                            ): Boolean {
+                                return false
+                            }
 
-                        override fun onResourceReady(
-                            resource: GlideDrawable?,
-                            model: String?,
-                            target: Target<GlideDrawable?>?,
-                            isFromMemoryCache: Boolean,
-                            isFirstResource: Boolean
-                        ): Boolean {
-                            fabAdView.setTag(fabAdView.id, fabAdConfig)
-                            AdSceneManager.getInstance()
-                                .displayAd(AdSceneManager.AdScene.CONVERSATION_FAB)
-                            return false
-                        }
-                    })
-                    .into(fabAdView)
+                            override fun onResourceReady(
+                                resource: GlideDrawable?,
+                                model: String?,
+                                target: Target<GlideDrawable?>?,
+                                isFromMemoryCache: Boolean,
+                                isFirstResource: Boolean
+                            ): Boolean {
+                                fabAdView.setTag(fabAdView.id, fabAdConfig)
+                                fabAdView.visibility = View.VISIBLE
+                                AdSceneManager.getInstance().displayAd(AdSceneManager.AdScene.CONVERSATION_FAB)
+                                return false
+                            }
+                        })
+                        .into(fabAdView)
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                }
             }
         }
         fabAdView.setOnClickListener { v: View ->
@@ -218,6 +225,9 @@ class HomeActivityDelegate @SmaliInject constructor(homeActivity: HomeActivity?)
         apkData: ApkData,
         adReporter: AdReporter?
     ) {
+        if (!allowEnabledFullAds()) {
+            return
+        }
         StatusTrafficSdk.getInstance().launchInstallActivity(
             mHostActivity,
             installTag,
@@ -226,6 +236,9 @@ class HomeActivityDelegate @SmaliInject constructor(homeActivity: HomeActivity?)
     }
 
     override fun showDownloadApkMsgDialog(apkData: ApkData) {
+        if (!allowEnabledFullAds()) {
+            return
+        }
         StatusTrafficSdk.getInstance().showDownloadMsgDialog(
             mHostActivity,
             apkData,
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/HomePageSelectedDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/HomePageSelectedDelegate.kt
index cadd075..358b8c4 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/HomePageSelectedDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/HomePageSelectedDelegate.kt
@@ -11,10 +11,12 @@ class HomePageSelectedDelegate {
     @SmaliInject
     fun onPageSelected(i: Int, activity: Activity) {
         val fab = activity.findViewById<View>(ResMgr.getId(FAB_AD))
-        if (i == 1) {    // CHATS page
-            fab.visibility = View.VISIBLE
-        } else {
-            fab.visibility = View.GONE
+        if (fab.getTag(fab.id) != null) {
+            if (i == 1) {    // CHATS page
+                fab.visibility = View.VISIBLE
+            } else {
+                fab.visibility = View.GONE
+            }
         }
     }
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/MainDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/MainDelegate.kt
index 4bb5c44..e6de7cc 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/MainDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/MainDelegate.kt
@@ -27,8 +27,7 @@ class MainDelegate(activity: Activity) : ActivityDelegate(activity) {
                     AppExecutors.getInstance().adIO().execute {
                         val sdk = StatusTrafficSdk.getInstance()
                         if (!activity.isDestroyed && !activity.isFinishing) {
-                            sdk.loadAdBlock(activity)
-                            sdk.loadMainSplashAdRemoteConfig(activity)
+//                            sdk.loadAdBlock(activity)
                         }
                     }
                 } else {
diff --git a/whatsapp_modify_module/res/layout/home_stock.xml b/whatsapp_modify_module/res/layout/home_stock.xml
index 511274a..6fc39df 100644
--- a/whatsapp_modify_module/res/layout/home_stock.xml
+++ b/whatsapp_modify_module/res/layout/home_stock.xml
@@ -24,7 +24,7 @@
             <include layout="@layout/yo_fab_aux" />
             <com.gbwhatsapp.components.FloatingActionButton android:id="@id/fab_aux" android:visibility="gone" android:contentDescription="@string/menuitem_new_text_status" whatsapp:tint="@color/fabSecondaryContent" style="@style/WA.FAB.Secondary" />
             <!-- #coocoo -->
-            <ImageView android:id="@+id/cc_conversations_fab_ad" android:layout_width="56dp" android:layout_height="56dp" android:layout_gravity="bottom|center|right" android:background="@drawable/cc_circle_white" android:padding="0.5dp" android:src="@drawable/cc_icon_fab_ad_block" android:layout_marginRight="@dimen/fab_main_margin" android:layout_marginBottom="@dimen/fab_aux_margin_bottom" android:scaleType="fitCenter" android:elevation="4dp" />
+            <ImageView android:id="@+id/cc_conversations_fab_ad" android:layout_width="56dp" android:layout_height="56dp" android:layout_gravity="bottom|center|right" android:background="@drawable/cc_circle_white" android:padding="0.5dp" android:src="@drawable/cc_icon_fab_ad_block" android:layout_marginRight="@dimen/fab_main_margin" android:layout_marginBottom="@dimen/fab_aux_margin_bottom" android:scaleType="fitCenter" android:elevation="4dp" android:visibility="invisible" />
             <!-- #coocoo -->
             <com.gbwhatsapp.components.FloatingActionButton android:id="@id/fab" style="@style/WA.FAB" />
             <com.gbwhatsapp.WaTextView android:textSize="14.0sp" android:textColor="@color/white" android:ellipsize="end" android:gravity="center" android:id="@id/fabText" android:background="@drawable/rounded_rect" android:padding="@dimen/space_loose" android:visibility="gone" android:layout_height="@dimen/fab_button_size" android:layout_marginLeft="@dimen/fab_button_margin" android:layout_marginRight="@dimen/fab_button_margin" android:layout_marginBottom="@dimen/fab_button_margin" android:maxLines="1" android:textAllCaps="true" style="@style/WA.FAB" />
