diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 77bd78c7b..0752e5ab9 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -116,6 +116,7 @@ import javax.crypto.SecretKey;
 
 import X.c1_01f;
 import X.c1_2Oc;
+import X.c1_2Ot;
 import X.c1_3Ul;
 import X.c1_2P6;
 import X.c1_2ua;
@@ -648,7 +649,11 @@ public class Coocoo {
         Log.e(TAG, "-----------------3---------------- // value : " + value);
     }
 
+    // invoke-static {vx}, Lcom/coocoo/coocoo/Coocoo;->traceLog4(Ljava/lang/Object;)V
     public static void traceLog4(Object object) {
+        if (object instanceof c1_2Ot) {
+            Log.e(TAG, "-----------------4---------------- // object : " + object);
+        }
         Log.e(TAG, "-----------------4---------------- // object : " + object);
     }
 
@@ -656,6 +661,31 @@ public class Coocoo {
         Log.e(TAG, "-----------------5---------------- // value : " + value);
     }
 
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog6(Z)V
+    public static void traceLog6(boolean value) {
+        Log.e(TAG, "-----------------6---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog7(Ljava/lang/CharSequence;)V
+    public static void traceLog7(CharSequence value) {
+        Log.e(TAG, "-----------------7---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog8(Ljava/util/ArrayList;)V
+    public static void traceLog8(ArrayList value) {
+        Log.e(TAG, "-----------------8---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog9(Landroid/view/View;)V
+    public static void traceLog9(View value) {
+        Log.e(TAG, "-----------------9---------------- // value : " + value.getId());
+    }
+
+    // invoke-static {}, Lcom/coocoo/coocoo/Coocoo;->stackTrace()V
+    public static void stackTrace() throws Exception {
+        throw new Exception("Debug");
+    }
+
     public static void traceLogCodeStack() {
         try {
             throw new IllegalArgumentException("traceLogCodeStack");
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
index 9e41c9f92..4188b0529 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
@@ -156,4 +156,8 @@ public class ContactPickerFragmentDelegate {
         return fragmentBundle.getBoolean(EXTRA_SKIP_CHECK_TEXT);
     }
 
+    @SmaliInject
+    public static boolean isMyMessage(c1_2Ot c12Ot) {
+        return ResMgr.getString(MY_MESSAGE_CONTACT_PICKER_NAME).equals(c12Ot.A0I);
+    }
 }
diff --git a/whatsapp_modify_module/smali/X/0HQ.smali b/whatsapp_modify_module/smali/X/0HQ.smali
index 12bcbbc88..bbf2768db 100644
--- a/whatsapp_modify_module/smali/X/0HQ.smali
+++ b/whatsapp_modify_module/smali/X/0HQ.smali
@@ -706,6 +706,10 @@
 
 .method public A07(Landroid/widget/ImageView;LX/2Ot;Z)V
     .locals 4
+#coocoo
+    const-string v0, "0HQ.A07"
+    invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog3(Ljava/lang/String;)V
+#coocoo
 
     .line 84712
     iget-object v0, p0, LX/0HQ;->A07:LX/06l;
diff --git a/whatsapp_modify_module/smali/X/1Da.1.smali b/whatsapp_modify_module/smali/X/1Da.1.smali
index 631347fc1..8b75cc393 100644
--- a/whatsapp_modify_module/smali/X/1Da.1.smali
+++ b/whatsapp_modify_module/smali/X/1Da.1.smali
@@ -891,7 +891,11 @@
     .line 218640
     :goto_6
     invoke-static/range {v25 .. v25}, LX/0AT;->A06(Landroid/widget/TextView;)V
-
+#coocoo
+    invoke-static {v7}, Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;->isMyMessage(LX/2Ot;)Z
+    move-result v14
+    if-nez v14, :cond_a
+#coocoo
     .line 218641
     iget-object v14, v11, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->A0f:LX/0HQ;
 
