diff --git a/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java b/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
index 9dbc74b34..fcbee8ed3 100644
--- a/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
+++ b/app/src/main/java/com/coocoo/newtheme/store/ThemeStoreActivity.java
@@ -71,7 +71,7 @@ public class ThemeStoreActivity extends CCBaseActivity implements View.OnClickLi
     private void initThemeFragment() {
         String from = getIntent().getStringExtra(Constants.FROM);
         Fragment themeFragment = new CombinedThemeFragment();
-        if (Constants.THEME_STORE.equals(from) && ThemeRemoteConfig.isUseWebStrategy()) {
+        if (true) {
             themeFragment = new LocalThemeFragment();
         }
         FragmentManager fragmentManager = getSupportFragmentManager();
