diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 77bd78c7b..c86b220fd 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -116,6 +116,7 @@ import javax.crypto.SecretKey;
 
 import X.c1_01f;
 import X.c1_2Oc;
+import X.c1_2Ot;
 import X.c1_3Ul;
 import X.c1_2P6;
 import X.c1_2ua;
@@ -648,7 +649,11 @@ public class Coocoo {
         Log.e(TAG, "-----------------3---------------- // value : " + value);
     }
 
+    // invoke-static {vx}, Lcom/coocoo/coocoo/Coocoo;->traceLog4(Ljava/lang/Object;)V
     public static void traceLog4(Object object) {
+        if (object instanceof c1_2Ot) {
+            Log.e(TAG, "-----------------4---------------- // object : " + object);
+        }
         Log.e(TAG, "-----------------4---------------- // object : " + object);
     }
 
@@ -656,6 +661,26 @@ public class Coocoo {
         Log.e(TAG, "-----------------5---------------- // value : " + value);
     }
 
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog6(Z)V
+    public static void traceLog6(boolean value) {
+        Log.e(TAG, "-----------------6---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog7(Ljava/lang/CharSequence;)V
+    public static void traceLog7(CharSequence value) {
+        Log.e(TAG, "-----------------7---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog8(Ljava/util/ArrayList;)V
+    public static void traceLog8(ArrayList value) {
+        Log.e(TAG, "-----------------8---------------- // value : " + value);
+    }
+
+    // invoke-static {}, Lcom/coocoo/coocoo/Coocoo;->stackTrace()V
+    public static void stackTrace() throws Exception {
+        throw new Exception("Debug");
+    }
+
     public static void traceLogCodeStack() {
         try {
             throw new IllegalArgumentException("traceLogCodeStack");
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
index 977d6130a..5e5f0c877 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
@@ -9,6 +9,7 @@ import android.content.Intent;
 import android.os.Bundle;
 import android.os.Parcelable;
 import android.view.View;
+import android.widget.ImageView;
 import android.widget.TextView;
 
 import androidx.annotation.Nullable;
@@ -105,7 +106,9 @@ public class ContactPickerFragmentDelegate {
         String myMessage = ResMgr.getString(MY_MESSAGE_CONTACT_PICKER_NAME);
         if (c12Ot != null && myMessage.equals(c12Ot.A0I)) {
             TextView tv = view.findViewById(ResMgr.getId(TV_MY_MESSAGE_ROW_STATUS));
+            ImageView iv = view.findViewById(ResMgr.getId("contactpicker_row_ephemeral_status"));
             tv.setText(ResMgr.getString(MY_MESSAGE_CONTACT_PICKER_STATUS));
+            iv.setImageResource(ResMgr.getDrawableId("cc_ic_avarta_mymessage_contactpicker"));
         }
     }
 
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
index a42ff3bbe..6d0a22336 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
@@ -2,6 +2,7 @@ package com.coocoo.whatsappdelegate
 
 import com.coocoo.annotation.SmaliInject
 import com.coocoo.report.Report
+import com.coocoo.utils.LogUtil
 import com.coocoowhatsapp.contact.picker.ContactPickerFragment
 
 object ContactPickerSendBtnDelegate {
@@ -12,6 +13,7 @@ object ContactPickerSendBtnDelegate {
     @SmaliInject
     @JvmStatic
     fun onSendBtnClick(fragment: ContactPickerFragment) {
+        LogUtil.d("onSendBtnClick")
         fragment.A09()?.intent?.getStringExtra(BUNDLE_KEY_STATUS_FEED_ID)?.let { statusId ->
             if (statusId.isNotEmpty()) {
                 Report.reportStatusFeedPlaybackShareDone(statusId)
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
index 2b8c43fa2..c058420b6 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
@@ -946,4 +946,10 @@ public class ConversationsFragmentDelegate extends FragmentDelegate<Conversation
         }
         notifyConversationsAdapter();
     }
+
+    @SmaliInject
+    public void resolveIntent(Intent intent) {
+        List<String> data = intent.getStringArrayListExtra("jids");
+        LogUtil.d("resolveIntent:" + data);
+    }
 }
diff --git a/app/src/main/res/drawable-xxhdpi/cc_ic_avarta_mymessage_contactpicker.webp b/app/src/main/res/drawable-xxhdpi/cc_ic_avarta_mymessage_contactpicker.webp
new file mode 100644
index 000000000..62024ea94
Binary files /dev/null and b/app/src/main/res/drawable-xxhdpi/cc_ic_avarta_mymessage_contactpicker.webp differ
diff --git a/whatsapp_modify_module/smali/X/1Fe.1.smali b/whatsapp_modify_module/smali/X/1Fe.1.smali
index 37372812c..bbfc3eb23 100644
--- a/whatsapp_modify_module/smali/X/1Fe.1.smali
+++ b/whatsapp_modify_module/smali/X/1Fe.1.smali
@@ -90,7 +90,9 @@
     invoke-static {v0}, LX/2Ov;->A07(Ljava/util/Collection;)Ljava/util/ArrayList;
 
     move-result-object v1
-
+#coocoo
+    invoke-static {v1}, Lcom/coocoo/coocoo/Coocoo;->traceLog8(Ljava/util/ArrayList;)V
+#coocoo
     const-string v0, "jids"
 
     .line 220968
