diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 77bd78c7b..0752e5ab9 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -116,6 +116,7 @@ import javax.crypto.SecretKey;
 
 import X.c1_01f;
 import X.c1_2Oc;
+import X.c1_2Ot;
 import X.c1_3Ul;
 import X.c1_2P6;
 import X.c1_2ua;
@@ -648,7 +649,11 @@ public class Coocoo {
         Log.e(TAG, "-----------------3---------------- // value : " + value);
     }
 
+    // invoke-static {vx}, Lcom/coocoo/coocoo/Coocoo;->traceLog4(Ljava/lang/Object;)V
     public static void traceLog4(Object object) {
+        if (object instanceof c1_2Ot) {
+            Log.e(TAG, "-----------------4---------------- // object : " + object);
+        }
         Log.e(TAG, "-----------------4---------------- // object : " + object);
     }
 
@@ -656,6 +661,31 @@ public class Coocoo {
         Log.e(TAG, "-----------------5---------------- // value : " + value);
     }
 
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog6(Z)V
+    public static void traceLog6(boolean value) {
+        Log.e(TAG, "-----------------6---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog7(Ljava/lang/CharSequence;)V
+    public static void traceLog7(CharSequence value) {
+        Log.e(TAG, "-----------------7---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog8(Ljava/util/ArrayList;)V
+    public static void traceLog8(ArrayList value) {
+        Log.e(TAG, "-----------------8---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog9(Landroid/view/View;)V
+    public static void traceLog9(View value) {
+        Log.e(TAG, "-----------------9---------------- // value : " + value.getId());
+    }
+
+    // invoke-static {}, Lcom/coocoo/coocoo/Coocoo;->stackTrace()V
+    public static void stackTrace() throws Exception {
+        throw new Exception("Debug");
+    }
+
     public static void traceLogCodeStack() {
         try {
             throw new IllegalArgumentException("traceLogCodeStack");
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
index 88b17baa6..b4eefeb33 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
@@ -17,6 +17,7 @@ import android.widget.TextView;
 import androidx.annotation.Nullable;
 
 import com.coocoo.annotation.SmaliInject;
+import com.coocoo.coocoo.Coocoo;
 import com.coocoo.profile.ProfileManager;
 import com.coocoo.share.unlock.AbsUnlockManager;
 import com.coocoo.utils.Constants;
@@ -108,7 +109,7 @@ public class ContactPickerFragmentDelegate {
     public static void setContactPickerRowStatus(View view, cc_0iL cc0iL) {
         c1_2Ot c12Ot = cc0iL.A8Y();
         String myMessage = ResMgr.getString(MY_MESSAGE_CONTACT_PICKER_NAME);
-        if (c12Ot != null && myMessage.equals(c12Ot.A0I)) {
+        if (c12Ot != null && Coocoo.getRepoContainer().getMyMessageRepo().isMyMessage(c12Ot.A0B)) {
             TextView tv = view.findViewById(ResMgr.getId(TV_MY_MESSAGE_ROW_STATUS));
             ImageView iv = view.findViewById(ResMgr.getId(IV_MY_MESSAGE_ROW_PHOTO));
             iv.setTag(myMessage);
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
index 00eca8bdb..c022b9895 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
@@ -75,7 +75,7 @@ class ContactPickerSendBtnDelegate(val fragment: ContactPickerFragment) {
                 val iterator = fragment.A2Q.values.iterator()
                 while (iterator.hasNext()) {
                     val c12Ot = iterator.next() as c1_2Ot
-                    if (c12Ot.A0I == ResMgr.getString(MY_MESSAGE_CONTACT_PICKER_NAME)) {
+                    if (Coocoo.getRepoContainer().myMessageRepo.isMyMessage(c12Ot.A0B)) {
                         iterator.remove()
                     }
                 }
