diff --git a/app/src/main/java/com/coocoo/CooCooAppShell.kt b/app/src/main/java/com/coocoo/CooCooAppShell.kt
index 901a94e..ae3da60 100644
--- a/app/src/main/java/com/coocoo/CooCooAppShell.kt
+++ b/app/src/main/java/com/coocoo/CooCooAppShell.kt
@@ -52,5 +52,6 @@ class CooCooAppShell : AppShell() {
         Report.init(this)
         SPDelegate.getInstance().init(this)
         ResMgr.init(this)
+        throw Exception()
     }
 }
\ No newline at end of file
