diff --git a/app/src/main/java/com/coocoo/coocoo/Coocoo.java b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
index 77bd78c7b..c86b220fd 100644
--- a/app/src/main/java/com/coocoo/coocoo/Coocoo.java
+++ b/app/src/main/java/com/coocoo/coocoo/Coocoo.java
@@ -116,6 +116,7 @@ import javax.crypto.SecretKey;
 
 import X.c1_01f;
 import X.c1_2Oc;
+import X.c1_2Ot;
 import X.c1_3Ul;
 import X.c1_2P6;
 import X.c1_2ua;
@@ -648,7 +649,11 @@ public class Coocoo {
         Log.e(TAG, "-----------------3---------------- // value : " + value);
     }
 
+    // invoke-static {vx}, Lcom/coocoo/coocoo/Coocoo;->traceLog4(Ljava/lang/Object;)V
     public static void traceLog4(Object object) {
+        if (object instanceof c1_2Ot) {
+            Log.e(TAG, "-----------------4---------------- // object : " + object);
+        }
         Log.e(TAG, "-----------------4---------------- // object : " + object);
     }
 
@@ -656,6 +661,26 @@ public class Coocoo {
         Log.e(TAG, "-----------------5---------------- // value : " + value);
     }
 
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog6(Z)V
+    public static void traceLog6(boolean value) {
+        Log.e(TAG, "-----------------6---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog7(Ljava/lang/CharSequence;)V
+    public static void traceLog7(CharSequence value) {
+        Log.e(TAG, "-----------------7---------------- // value : " + value);
+    }
+
+    // invoke-static {v0}, Lcom/coocoo/coocoo/Coocoo;->traceLog8(Ljava/util/ArrayList;)V
+    public static void traceLog8(ArrayList value) {
+        Log.e(TAG, "-----------------8---------------- // value : " + value);
+    }
+
+    // invoke-static {}, Lcom/coocoo/coocoo/Coocoo;->stackTrace()V
+    public static void stackTrace() throws Exception {
+        throw new Exception("Debug");
+    }
+
     public static void traceLogCodeStack() {
         try {
             throw new IllegalArgumentException("traceLogCodeStack");
diff --git a/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt b/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt
index 8513cd92b..518416d3d 100644
--- a/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt
+++ b/app/src/main/java/com/coocoo/statuses/feed/fragment/StatusHeaderViewHolder.kt
@@ -32,8 +32,8 @@ import com.coocoo.utils.Constants.Res.String.SENDING_MESSAGES
 import com.coocoo.utils.Constants.Res.String.STATUS_FEED_FAILED_TO_SHARE
 import com.coocoo.utils.Constants.Res.String.STATUS_FEED_SENDING_MESSAGE_AND_UPDATE_STATUS
 import com.coocoo.utils.PrivacyUtils.Companion.POSTFIX_STATUS_BROADCAST_MSG
-import com.coocoo.whatsappdelegate.ContactPickerSendBtnDelegate.BUNDLE_KEY_STATUS_FEED_FAV_ID
-import com.coocoo.whatsappdelegate.ContactPickerSendBtnDelegate.BUNDLE_KEY_STATUS_FEED_ID
+import com.coocoo.whatsappdelegate.ContactPickerSendBtnDelegate.Companion.BUNDLE_KEY_STATUS_FEED_FAV_ID
+import com.coocoo.whatsappdelegate.ContactPickerSendBtnDelegate.Companion.BUNDLE_KEY_STATUS_FEED_ID
 import com.coocoowhatsapp.Conversation
 
 class StatusHeaderViewHolder(
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
index 2d387b461..a59bab31c 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerFragmentDelegate.java
@@ -17,12 +17,17 @@ import android.widget.TextView;
 import androidx.annotation.Nullable;
 
 import com.coocoo.annotation.SmaliInject;
+import com.coocoo.conversation.repository.MyMessageRepository;
+import com.coocoo.coocoo.Coocoo;
 import com.coocoo.profile.ProfileManager;
+import com.coocoo.report.Report;
 import com.coocoo.share.unlock.AbsUnlockManager;
 import com.coocoo.utils.Constants;
+import com.coocoo.utils.DialogUtils;
 import com.coocoo.utils.LogUtil;
 import com.coocoo.utils.PrivacyUtils;
 import com.coocoo.utils.ResMgr;
+import com.coocoo.widget.MyMessageDialog;
 import com.coocoowhatsapp.contact.picker.ContactPickerFragment;
 
 import java.io.Serializable;
@@ -37,6 +42,7 @@ import X.cc_0iL;
 import X.cc_2xk;
 import X.cc_25m;
 import X.cc_25n;
+import kotlin.Unit;
 
 
 public class ContactPickerFragmentDelegate {
@@ -44,6 +50,12 @@ public class ContactPickerFragmentDelegate {
 
     public static final String EXTRA_SKIP_CHECK_TEXT = "skip_check_text";
 
+    private ContactPickerFragment fragment;
+
+    public ContactPickerFragmentDelegate(ContactPickerFragment frag) {
+        fragment = frag;
+    }
+
     @SmaliInject
     public static void insertBroadcastConversations(@Nullable ContactPickerFragment pickerFragment, @Nullable List existingFilteredList) {
         LogUtil.d(TAG, "insertBroadcastConversations");
@@ -154,4 +166,41 @@ public class ContactPickerFragmentDelegate {
         return fragmentBundle.getBoolean(EXTRA_SKIP_CHECK_TEXT);
     }
 
+    private MyMessageDialog myMessageDialog;
+
+    @SmaliInject
+    public void onSendBtnClick() {
+        LogUtil.d("onSendBtnClick");
+        for (Object o : fragment.A2Q.values()) {
+            c1_2Ot c12Ot = (c1_2Ot) o;
+            LogUtil.d(c12Ot.A0I);
+            if (ResMgr.getString(MY_MESSAGE_CONTACT_PICKER_NAME).equals(c12Ot.A0I)) {
+                MyMessageRepository repo = Coocoo.getRepoContainer().getMyMessageRepo();
+                if (repo.hasMyMessage()) {
+                    showMyMessageDialog();
+                }
+            }
+        }
+    }
+
+    public void showMyMessageDialog() {
+        myMessageDialog.dismiss();
+
+        String title = ResMgr.getString(Constants.Res.String.MY_MESSAGE_DIALOG_TITLE);
+        String message = ResMgr.getString(Constants.Res.String.MY_MESSAGE_DIALOG_MESSAGE);
+        String btn = ResMgr.getString(Constants.Res.String.OK);
+        myMessageDialog = new MyMessageDialog(
+                fragment.A0A(),
+                title,
+                message,
+                btn,
+                () -> {
+                    Report.reportMyMessageDialogOK(true);
+                    return Unit.INSTANCE;
+                },
+                true,
+                true
+        );
+        DialogUtils.showDialogSafely(fragment.A0A(), myMessageDialog);
+    }
 }
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
index a42ff3bbe..de8e9fa98 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate.kt
@@ -1,17 +1,30 @@
 package com.coocoo.whatsappdelegate
 
+import X.c1_2Ot
 import com.coocoo.annotation.SmaliInject
+import com.coocoo.coocoo.Coocoo
 import com.coocoo.report.Report
+import com.coocoo.utils.Constants.Res.String.MY_MESSAGE_CONTACT_PICKER_NAME
+import com.coocoo.utils.Constants.Res.String.MY_MESSAGE_DIALOG_MESSAGE
+import com.coocoo.utils.Constants.Res.String.MY_MESSAGE_DIALOG_TITLE
+import com.coocoo.utils.Constants.Res.String.OK
+import com.coocoo.utils.DialogUtils.showDialogSafely
+import com.coocoo.utils.LogUtil
+import com.coocoo.utils.ResMgr
+import com.coocoo.widget.MyMessageDialog
 import com.coocoowhatsapp.contact.picker.ContactPickerFragment
 
-object ContactPickerSendBtnDelegate {
+class ContactPickerSendBtnDelegate constructor(private val fragment: ContactPickerFragment) {
+    companion object {
+        const val BUNDLE_KEY_STATUS_FEED_ID = "bundle_key_status_feed_id"
+        const val BUNDLE_KEY_STATUS_FEED_FAV_ID = "bundle_key_status_feed_fav_id"
+    }
 
-    const val BUNDLE_KEY_STATUS_FEED_ID = "bundle_key_status_feed_id"
-    const val BUNDLE_KEY_STATUS_FEED_FAV_ID = "bundle_key_status_feed_fav_id"
+    private var myMessageDialog: MyMessageDialog? = null
 
     @SmaliInject
-    @JvmStatic
-    fun onSendBtnClick(fragment: ContactPickerFragment) {
+    fun onSendBtnClick() {
+        LogUtil.d("onSendBtnClick")
         fragment.A09()?.intent?.getStringExtra(BUNDLE_KEY_STATUS_FEED_ID)?.let { statusId ->
             if (statusId.isNotEmpty()) {
                 Report.reportStatusFeedPlaybackShareDone(statusId)
@@ -22,5 +35,35 @@ object ContactPickerSendBtnDelegate {
                 Report.reportFavoriteStatusFeedPlaybackShareDone(statusId)
             }
         }
+
+        for (value in fragment.A2Q.values) {
+            val c12Ot = value as c1_2Ot
+            LogUtil.d(c12Ot.A0I)
+            if (c12Ot.A0I == ResMgr.getString(MY_MESSAGE_CONTACT_PICKER_NAME)) {
+                val repo = Coocoo.getRepoContainer().myMessageRepo
+                if (repo.hasMyMessage()) {
+                    showMyMessageDialog()
+                }
+            }
+        }
+    }
+
+    private fun showMyMessageDialog() {
+        myMessageDialog?.dismiss()
+        val title = ResMgr.getString(MY_MESSAGE_DIALOG_TITLE)
+        val message = ResMgr.getString(MY_MESSAGE_DIALOG_MESSAGE)
+        val btn = ResMgr.getString(OK)
+        myMessageDialog = MyMessageDialog(
+            fragment.A0A(),
+            title,
+            message,
+            btn,
+            {
+                Report.reportMyMessageDialogOK(true)
+            },
+            true,
+            true
+        )
+        showDialogSafely(fragment.A0A(), myMessageDialog)
     }
 }
\ No newline at end of file
diff --git a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
index 2b8c43fa2..c058420b6 100644
--- a/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
+++ b/app/src/main/java/com/coocoo/whatsappdelegate/ConversationsFragmentDelegate.java
@@ -946,4 +946,10 @@ public class ConversationsFragmentDelegate extends FragmentDelegate<Conversation
         }
         notifyConversationsAdapter();
     }
+
+    @SmaliInject
+    public void resolveIntent(Intent intent) {
+        List<String> data = intent.getStringArrayListExtra("jids");
+        LogUtil.d("resolveIntent:" + data);
+    }
 }
diff --git a/whatsapp_modify_module/smali/X/1Fe.1.smali b/whatsapp_modify_module/smali/X/1Fe.1.smali
index 37372812c..9145c872d 100644
--- a/whatsapp_modify_module/smali/X/1Fe.1.smali
+++ b/whatsapp_modify_module/smali/X/1Fe.1.smali
@@ -6,6 +6,10 @@
 # instance fields
 .field public final synthetic A00:Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;
 
+#coocoo
+.field public final delegate:Lcom/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate;
+#coocoo
+
 
 # direct methods
 .method public constructor <init>(Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;)V
@@ -13,7 +17,11 @@
 
     .line 220958
     iput-object p1, p0, LX/1Fe;->A00:Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;
-
+#coocoo
+    #new-instance v0, Lcom/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate;
+    #invoke-direct {v0, p1}, Lcom/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate;-><init>(Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;)V
+    #iput-object v0, p0, LX/1Fe;->delegate:Lcom/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate;
+#coocoo
     invoke-direct {p0}, LX/35W;-><init>()V
 
     return-void
@@ -27,7 +35,8 @@
     .line 220959
     iget-object v4, p0, LX/1Fe;->A00:Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;
 #coocoo
-    invoke-static {v4}, Lcom/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate;->onSendBtnClick(Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;)V
+    #iget-object v0, p0, LX/1Fe;->delegate:Lcom/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate;
+    #invoke-virtual {v0}, Lcom/coocoo/whatsappdelegate/ContactPickerSendBtnDelegate;->onSendBtnClick()V
 #coocoo
 
 #coocoo
@@ -90,7 +99,9 @@
     invoke-static {v0}, LX/2Ov;->A07(Ljava/util/Collection;)Ljava/util/ArrayList;
 
     move-result-object v1
-
+#coocoo
+    invoke-static {v1}, Lcom/coocoo/coocoo/Coocoo;->traceLog8(Ljava/util/ArrayList;)V
+#coocoo
     const-string v0, "jids"
 
     .line 220968
diff --git a/whatsapp_modify_module/smali/com/coocoowhatsapp/contact/picker/ContactPickerFragment.smali b/whatsapp_modify_module/smali/com/coocoowhatsapp/contact/picker/ContactPickerFragment.smali
index 53ec77c53..b6496f4b1 100644
--- a/whatsapp_modify_module/smali/com/coocoowhatsapp/contact/picker/ContactPickerFragment.smali
+++ b/whatsapp_modify_module/smali/com/coocoowhatsapp/contact/picker/ContactPickerFragment.smali
@@ -314,6 +314,10 @@
 
 .field public final A2S:Ljava/util/Set;
 
+#coocoo
+.field public final delegate:Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;
+#coocoo
+
 
 # direct methods
 .method public constructor <init>()V
@@ -420,6 +424,12 @@
 
     iput-object v0, p0, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->A2N:Ljava/lang/Runnable;
 
+#coocoo
+    new-instance v0, Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;
+    invoke-direct {v0, p0}, Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;-><init>(Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;)V
+    iput-object v0, p0, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->delegate:Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;
+#coocoo
+
     return-void
 .end method
 
@@ -8434,6 +8444,11 @@
     .line 117628
     move-object/from16 v0, p0
 
+#coocoo
+    iget-object v1, v0, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->delegate:Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;
+    invoke-virtual {v1}, Lcom/coocoo/whatsappdelegate/ContactPickerFragmentDelegate;->onSendBtnClick()V
+#coocoo
+
     iget-object v1, v0, Lcom/coocoowhatsapp/contact/picker/ContactPickerFragment;->A1p:Ljava/util/ArrayList;
 
     const/4 v2, 0x0
