diff --git a/app/src/main/java/com/coocoo/CooCoo.kt b/app/src/main/java/com/coocoo/CooCoo.kt
index 6859204e..bf5ca0a6 100644
--- a/app/src/main/java/com/coocoo/CooCoo.kt
+++ b/app/src/main/java/com/coocoo/CooCoo.kt
@@ -137,7 +137,7 @@ object CooCoo {
             appFlavor,
             com.coocoo.config.BuildConfig.COOCOO_VERSION_CODE,
             appChannel,
-            com.coocoo.config.BuildConfig.IS_DEBUG,
+            false,
         )
         AdCommonManager.injectReporter(ReportStatusVideoTraffic)
     }
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
index befc7a47..f6fb965e 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
@@ -144,9 +144,9 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
             try {
                 StatusTrafficReporter.report(Constant.Report.ST_FETCH_CONF_START)
                 init(context)
-                if (false == apiPref?.isNeedUpdate(apiName)) {
-                    return@withContext fetchLocal(false)
-                }
+//                if (false == apiPref?.isNeedUpdate(apiName)) {
+//                    return@withContext fetchLocal(false)
+//                }
 
                 apiQualifiedConfig = null
                 apiRemoteConfig = null
@@ -169,7 +169,7 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
 
                 val headers = OkHttpHelper.getRemoteConfigHeaders(
                     isDebug,
-                    country,
+                    "tw",
                     appVersion,
                     appChannel,
                     appFlavor,
