diff --git a/app/src/main/java/com/coocoo/CooCoo.kt b/app/src/main/java/com/coocoo/CooCoo.kt
index 6859204e..bf5ca0a6 100644
--- a/app/src/main/java/com/coocoo/CooCoo.kt
+++ b/app/src/main/java/com/coocoo/CooCoo.kt
@@ -137,7 +137,7 @@ object CooCoo {
             appFlavor,
             com.coocoo.config.BuildConfig.COOCOO_VERSION_CODE,
             appChannel,
-            com.coocoo.config.BuildConfig.IS_DEBUG,
+            false,
         )
         AdCommonManager.injectReporter(ReportStatusVideoTraffic)
     }
diff --git a/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt b/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
index 6eecf980..a455a962 100644
--- a/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
+++ b/app/src/main/java/com/coocoo/callhistory/CallHistoryDelegate.kt
@@ -1,6 +1,7 @@
 package com.coocoo.callhistory
 
 import X.cc_5WG
+import android.util.Log
 import android.view.Gravity
 import android.view.View
 import android.view.ViewGroup
@@ -23,9 +24,9 @@ object CallHistoryDelegate {
                     gravity = Gravity.CENTER
                 }
                 val vh = CallHistoryViewHolder(itemView, fragment)
+                vh.bindAd()
                 itemView.tag = vh
             }
-            (itemView.tag as CallHistoryViewHolder).bindAd()
             itemView
         } else {
             null
@@ -35,6 +36,7 @@ object CallHistoryDelegate {
     @JvmStatic
     @SmaliInject
     fun addCallListBannerItem(arr: ArrayList<cc_5WG>) {
+        Log.e("Gene", "addCallListBannerItem: ${AdSdkManager.isCallListBannerAdReady()}")
         if (AdSdkManager.isCallListBannerAdReady()) {
             if (arr.size > 4) {
                 arr.add(4, CallHistoryItem())
@@ -44,6 +46,10 @@ object CallHistoryDelegate {
         }
     }
 
+    private fun ArrayList<cc_5WG>.containsCallListBanner(): Boolean {
+        return find { it.ADd() == CallHistoryItem.VIEW_TYPE } != null
+    }
+
     @JvmStatic
     @SmaliInject
     fun onResume(fragment: CallsHistoryFragment) {
@@ -51,6 +57,7 @@ object CallHistoryDelegate {
             val adsSdk = config.waterFall
             if (adsSdk != null) {
                 AdSdkManager.loadCallListBannerAd(fragment.A0C(), adsSdk) {
+                    Log.e("Gene", "onLoadAdListener: ${fragment.A0f}")
                     addCallListBannerItem(fragment.A0f)
                     fragment.A07.notifyDataSetChanged()
                 }
diff --git a/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt b/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt
index 1b85ca04..4fbdf43a 100644
--- a/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt
+++ b/app/src/main/java/com/coocoo/callhistory/CallHistoryViewHolder.kt
@@ -2,6 +2,7 @@ package com.coocoo.callhistory
 
 import X.cc_4VA
 import X.cc_5WG
+import android.util.Log
 import android.widget.RelativeLayout
 import com.gbwhatsapp.calling.callhistory.CallsHistoryFragment
 import com.status.traffic.ads.AdSdkManager
@@ -16,6 +17,7 @@ class CallHistoryViewHolder(
     }
 
     fun bindAd() {
+        Log.e("Gene", "targetView.childCount: ${targetView.childCount}")
         AdSdkManager.showCallListBannerAd(targetView) {
             val iterable = fragment.A0f.iterator()
             while (iterable.hasNext()) {
diff --git a/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseBannerAd.kt b/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseBannerAd.kt
index 053c2982..4c666c7d 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseBannerAd.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/ads/ad/BaseBannerAd.kt
@@ -1,5 +1,6 @@
 package com.status.traffic.ads.ad
 
+import android.util.Log
 import android.view.View
 import android.view.ViewGroup
 import android.widget.RelativeLayout
@@ -53,6 +54,7 @@ open class BaseBannerAd (
                         setUpView(bannerContainer, nativeHandler.adViewGroup)?.run {
                             bannerContainer.addView(this)
                         }
+                        Log.e("Report", "nativeHandler.onResume()")
                         nativeHandler.onResume()
                     }
                 }
diff --git a/status-traffic-api/src/main/java/com/status/traffic/ads/ad/CallListBannerAdController.kt b/status-traffic-api/src/main/java/com/status/traffic/ads/ad/CallListBannerAdController.kt
index 75289422..ac05f9dc 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/ads/ad/CallListBannerAdController.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/ads/ad/CallListBannerAdController.kt
@@ -1,3 +1,8 @@
 package com.status.traffic.ads.ad
 
-class CallListBannerAdController: BaseBannerAdController()
\ No newline at end of file
+class CallListBannerAdController: BaseBannerAdController() {
+    override fun release() {
+        super.release()
+        java.lang.Exception().printStackTrace()
+    }
+}
\ No newline at end of file
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
index befc7a47..7c353e6c 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/BaseRemoteConfig.kt
@@ -144,9 +144,9 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
             try {
                 StatusTrafficReporter.report(Constant.Report.ST_FETCH_CONF_START)
                 init(context)
-                if (false == apiPref?.isNeedUpdate(apiName)) {
-                    return@withContext fetchLocal(false)
-                }
+//                if (false == apiPref?.isNeedUpdate(apiName)) {
+//                    return@withContext fetchLocal(false)
+//                }
 
                 apiQualifiedConfig = null
                 apiRemoteConfig = null
@@ -169,7 +169,7 @@ abstract class BaseRemoteConfig<T, S>(protected val apiName: String) {
 
                 val headers = OkHttpHelper.getRemoteConfigHeaders(
                     isDebug,
-                    country,
+                    "cn",
                     appVersion,
                     appChannel,
                     appFlavor,
diff --git a/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt b/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
index 1d430ca3..5361c3ec 100644
--- a/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
+++ b/status-traffic-api/src/main/java/com/status/traffic/data/config/CallListBannerAdRemoteConfig.kt
@@ -3,9 +3,11 @@ package com.status.traffic.data.config
 import com.google.gson.JsonArray
 import com.google.gson.JsonObject
 import com.status.traffic.Constant.Api.Config.CALL_LIST_BANNER_AD
+import com.status.traffic.data.vo.AdsSDK
 import com.status.traffic.data.vo.CallListBannerAdConfig
 
-object CallListBannerAdRemoteConfig : BaseAdRemoteConfig<List<CallListBannerAdConfig>, Void>(CALL_LIST_BANNER_AD) {
+object CallListBannerAdRemoteConfig :
+    BaseAdRemoteConfig<List<CallListBannerAdConfig>, Void>(CALL_LIST_BANNER_AD) {
     override fun parserAdsInfo(json: JsonArray?): List<CallListBannerAdConfig>? {
         try {
             return getGson().fromJson(json, Array<CallListBannerAdConfig>::class.java).toList()
@@ -22,6 +24,15 @@ object CallListBannerAdRemoteConfig : BaseAdRemoteConfig<List<CallListBannerAdCo
     }
 
     fun getCallListBannerAd(): CallListBannerAdConfig? {
-        return apiRemoteConfig?.get(0)
+//        return apiRemoteConfig?.get(0)
+        return CallListBannerAdConfig(
+            waterFall = arrayListOf(
+                AdsSDK(
+                    sdk = "mv",
+                    adId = "dj9WUBLYwepIKQMlEa7LzaWR4EozNehoLzdIrVkA7Lo=",
+                    format = "native"
+                )
+            )
+        )
     }
 }
\ No newline at end of file
